{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/poltorashka-s-BMa/course-os-linux/blob/main/%D0%9A%D0%BE%D0%BF%D0%B8%D1%8F_%D0%B1%D0%BB%D0%BE%D0%BA%D0%BD%D0%BE%D1%82%D0%B0_%22Genome_assembly_de_novo_2025_spring_ipynb%22.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "pMK2HUxP5LMt"
      },
      "source": [
        "qc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Pk8vMQpR5FNK"
      },
      "outputs": [],
      "source": [
        "import os"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4EPp6qz65NuK",
        "outputId": "355c2979-6f05-4c16-8708-5f79e5bdeec8"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "2304"
            ]
          },
          "execution_count": 3,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"unzip QC_workshop.zip\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kpadQxDR5WPK",
        "outputId": "dba39a02-2aef-4fa5-815f-15c8f6619e18"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "execution_count": 4,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"mkdir QC_Reports\")\n",
        "os.system(\"wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.12.1.zip\")\n",
        "os.system(\"unzip fastqc_v0.12.1.zip\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yf6QmBP05aiu",
        "outputId": "82d49e0d-788f-4587-9a67-b27af72f05d6"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"wget https://github.com/usadellab/Trimmomatic/files/5854859/Trimmomatic-0.39.zip\")\n",
        "os.system(\"unzip Trimmomatic-0.39.zip\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ptDZmmJ25ydj",
        "outputId": "b5a85adc-60e1-4971-fc3d-9bdf221aa6da"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "256"
            ]
          },
          "execution_count": 6,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"pip install h5py\")\n",
        "os.system(\"pip install pysam\")\n",
        "os.system(\"pip install edlib\")\n",
        "os.system(\"pip install python-edlib\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "so-EeLC7575I",
        "outputId": "96d041e3-947e-46ba-a981-9c7827cdc7b6"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"git clone https://github.com/yfukasawa/LongQC.git\")\n",
        "os.system(\"cd LongQC/minimap2-coverage && make\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "oQP5DidT6FrQ",
        "outputId": "62530d5f-52ce-4167-9a25-7a97d5d04cd0"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"wget https://github.com/richardmleggett/nextclip/archive/refs/tags/NextClip_v1.3.1.tar.gz\")\n",
        "os.system(\"tar -xzvf NextClip_v1.3.1.tar.gz\")\n",
        "os.system(\"cd nextclip-NextClip_v1.3.1 && make\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "07wiarM96OL5",
        "outputId": "6b6ba69e-b283-4338-eee2-4073ae8a0e15"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "execution_count": 9,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"wget https://github.com/ebiggers/flash/releases/download/v1.2.11/FLASH-1.2.11.tar.gz\")\n",
        "os.system(\"tar -xzvf FLASH-1.2.11.tar.gz\")\n",
        "os.system(\"cd FLASH-1.2.11 && make\")\n",
        "#"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "23wIiAQs6Y1w"
      },
      "source": [
        "анализ"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "S0A5Irb-6c15",
        "outputId": "04c158a8-ce9a-4d39-90f8-38272aeed1de"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "execution_count": 9,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"FastQC/fastqc -o QC_Reports /content/7_S4_L001_R1_001.fastq\")\n",
        "os.system(\"FastQC/fastqc -o QC_Reports /content/7_S4_L001_R1_002.fastq\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "jjIsKYGe8fTL"
      },
      "source": [
        "тримируем"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BxLyL8J47yOP",
        "outputId": "8712700c-fb30-4d0d-a0e2-85a68f0d8dde"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"mkdir trimmed\")\n",
        "os.system(\"java -jar Trimmomatic-0.39/trimmomatic-0.39.jar PE -threads 1 -phred33 /content/7_S4_L001_R1_001.fastq /content/7_S4_L001_R2_001.fastq trimmed/7_forward_paired.trimmed.fastq trimmed/7_forward_unpaired.trimmed.fastq trimmed/7_reverse_paired.trimmed.fastq trimmed/7_reverse_unpaired.trimmed.fastq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:30 MINLEN:36\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "DNPfKXCi9GPX"
      },
      "source": [
        "смотрим"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VRJLhbwX9Hwf",
        "outputId": "ab937cc0-cd56-4990-cf26-5eabdc1c6733"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "execution_count": 11,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "os.system(\"FastQC/fastqc -o QC_Reports trimmed/7_forward_paired.trimmed.fastq\")\n",
        "os.system(\"FastQC/fastqc -o QC_Reports trimmed/7_forward_unpaired.trimmed.fastq\")\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vtdbh08edWAR"
      },
      "source": [
        "# 1. *de novo* genome assembly"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "pRtVv8sUdjq8"
      },
      "source": [
        "## 1.1 Prepare environment for genome assembly *de novo*"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2opUgW_mdVar"
      },
      "outputs": [],
      "source": [
        "! mkdir genome_assembly\n",
        "! mkdir genome_assembly/k11\n",
        "#! mkdir genome_assembly/k21\n",
        "#! mkdir genome_assembly/k25"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zDbbNzSKdNaQ",
        "outputId": "82b1fa16-88aa-4c16-e055-1fefbda63d90"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2025-07-12 10:11:28--  https://github.com/ablab/spades/releases/download/v4.1.0/SPAdes-4.1.0-Linux.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.114.3\n",
            "Connecting to github.com (github.com)|140.82.114.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/60187186/ec2c3b64-90aa-4933-9cf7-3101d3f9fdec?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250712%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250712T101128Z&X-Amz-Expires=1800&X-Amz-Signature=61a8c4679fe3666f6c24dfee401895d0c81727da2c7d0948190e28c583beef4c&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3DSPAdes-4.1.0-Linux.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2025-07-12 10:11:28--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/60187186/ec2c3b64-90aa-4933-9cf7-3101d3f9fdec?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250712%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250712T101128Z&X-Amz-Expires=1800&X-Amz-Signature=61a8c4679fe3666f6c24dfee401895d0c81727da2c7d0948190e28c583beef4c&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3DSPAdes-4.1.0-Linux.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 179131491 (171M) [application/octet-stream]\n",
            "Saving to: ‘SPAdes-4.1.0-Linux.tar.gz’\n",
            "\n",
            "SPAdes-4.1.0-Linux. 100%[===================>] 170.83M   312MB/s    in 0.5s    \n",
            "\n",
            "2025-07-12 10:11:29 (312 MB/s) - ‘SPAdes-4.1.0-Linux.tar.gz’ saved [179131491/179131491]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! wget https://github.com/ablab/spades/releases/download/v4.1.0/SPAdes-4.1.0-Linux.tar.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "19Tf-JD7fFMe",
        "outputId": "61a2ad29-466e-49bc-f507-6ea4c97a3802"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "SPAdes-4.1.0-Linux/bin/\n",
            "SPAdes-4.1.0-Linux/bin/spades-gmapper\n",
            "SPAdes-4.1.0-Linux/bin/spades-gsimplifier\n",
            "SPAdes-4.1.0-Linux/bin/pathracer\n",
            "SPAdes-4.1.0-Linux/bin/metaspades.py\n",
            "SPAdes-4.1.0-Linux/bin/spades-ionhammer\n",
            "SPAdes-4.1.0-Linux/bin/coronaspades.py\n",
            "SPAdes-4.1.0-Linux/bin/spades-bwa\n",
            "SPAdes-4.1.0-Linux/bin/spaligner\n",
            "SPAdes-4.1.0-Linux/bin/plasmidspades.py\n",
            "SPAdes-4.1.0-Linux/bin/spades-gfa-split\n",
            "SPAdes-4.1.0-Linux/bin/spades-gbuilder\n",
            "SPAdes-4.1.0-Linux/bin/spades_init.py\n",
            "SPAdes-4.1.0-Linux/bin/pathracer-seq-fs\n",
            "SPAdes-4.1.0-Linux/bin/metaplasmidspades.py\n",
            "SPAdes-4.1.0-Linux/bin/spades-kmer-estimating\n",
            "SPAdes-4.1.0-Linux/bin/rnaspades.py\n",
            "SPAdes-4.1.0-Linux/bin/spades-convert-bin-to-fasta\n",
            "SPAdes-4.1.0-Linux/bin/spades-hammer\n",
            "SPAdes-4.1.0-Linux/bin/binspreader\n",
            "SPAdes-4.1.0-Linux/bin/spades-read-filter\n",
            "SPAdes-4.1.0-Linux/bin/rnaviralspades.py\n",
            "SPAdes-4.1.0-Linux/bin/metaviralspades.py\n",
            "SPAdes-4.1.0-Linux/bin/spades-core\n",
            "SPAdes-4.1.0-Linux/bin/spades-corrector-core\n",
            "SPAdes-4.1.0-Linux/bin/spades.py\n",
            "SPAdes-4.1.0-Linux/bin/spades-kmercount\n",
            "SPAdes-4.1.0-Linux/share/\n",
            "SPAdes-4.1.0-Linux/share/spaligner/\n",
            "SPAdes-4.1.0-Linux/share/spaligner/spaligner_config.yaml\n",
            "SPAdes-4.1.0-Linux/share/spades/\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset/\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset/genes_1K.gff\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset/ecoli_1K_2.fq.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset/ecoli_1K_1.fq.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset/operons_1K.txt\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset/operons_1K.gff\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset/reference_1K.fa.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset/genes_1K.txt\n",
            "SPAdes-4.1.0-Linux/share/spades/coronaspades_hmms/\n",
            "SPAdes-4.1.0-Linux/share/spades/coronaspades_hmms/Phan-CoV.hmm.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/coronaspades_hmms/Pfam-A.SARS-CoV-2.hmm.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/LICENSE\n",
            "SPAdes-4.1.0-Linux/share/spades/VERSION\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset_plasmid/\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset_plasmid/pl2.fq.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/test_dataset_plasmid/pl1.fq.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/func_inspect.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/__init__.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/_multiprocessing_helpers.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/_compat.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/disk.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/numpy_pickle.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/pool.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/testing.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/my_exceptions.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/hashing.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/parallel.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/memory.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/format_stack.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/logger.py\n",
            "SPAdes-4.1.0-Linux/share/spades/joblib3/_memory_helpers.py\n",
            "SPAdes-4.1.0-Linux/share/spades/GPLv2.txt\n",
            "SPAdes-4.1.0-Linux/share/spades/sewage/\n",
            "SPAdes-4.1.0-Linux/share/spades/sewage/usher_barcodes.csv\n",
            "SPAdes-4.1.0-Linux/share/spades/biosynthetic_spades_hmms/\n",
            "SPAdes-4.1.0-Linux/share/spades/biosynthetic_spades_hmms/TE.hmm.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/biosynthetic_spades_hmms/AT.hmm.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/biosynthetic_spades_hmms/AMP.hmm.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/biosynthetic_spades_hmms/KR.hmm.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/biosynthetic_spades_hmms/KS.hmm.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/biosynthetic_spades_hmms/CStart.hmm.gz\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/__init__.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/tokens.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/resolver.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/loader.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/composer.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/serializer.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/nodes.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/cyaml.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/error.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/emitter.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/scanner.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/constructor.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/representer.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/parser.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/events.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/dumper.py\n",
            "SPAdes-4.1.0-Linux/share/spades/pyyaml3/reader.py\n",
            "SPAdes-4.1.0-Linux/share/spades/README.md\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/ionhammer/\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/ionhammer/ionhammer.cfg\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/sewage_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/simplification.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/careful_mda_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/large_genome_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/careful_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/construction.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/rna_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/pe_params.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/mda_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/metaviral_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/rnaviral_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/metaplasmid_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/distance_estimation.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/detail_info_printer.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/config.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/plasmid_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/toy.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/hmm_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/isolate_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/debruijn/meta_mode.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/corrector/\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/corrector/corrector.info\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/hammer/\n",
            "SPAdes-4.1.0-Linux/share/spades/configs/hammer/config.info\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/support.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/__init__.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/spades_iteration_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/preprocess_reads_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/error_correction_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/breaking_scaffolds_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/pipeline.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/before_start_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/terminating_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/check_test_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/scaffold_correction_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/spades_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/stages/correction_stage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/executors/\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/executors/executor_save_yaml.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/executors/executor_local.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/executors/executors.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/plasmid_glue.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/check_test_script.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/preprocess_interlaced_reads.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/copy_files.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/correction_iteration_script.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/compress_all.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/scripts/preprocess_contigs.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/process_cfg.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/options_storage.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/common/\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/common/__init__.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/common/SeqIO.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/common/sam_parser.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/common/alignment.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/common/parallel_launcher.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/commands_parser.py\n",
            "SPAdes-4.1.0-Linux/share/spades/spades_pipeline/options_parser.py\n"
          ]
        }
      ],
      "source": [
        "! tar -xvzf SPAdes-4.1.0-Linux.tar.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "qCjbbauUkiWZ"
      },
      "outputs": [],
      "source": [
        "! mv SPAdes-4.1.0-Linux spades"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "RTjlBIU1fQZP",
        "outputId": "396e9c9a-3a97-4a9c-d133-d4909605be26"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "\n",
            "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
            "\n",
            "\n",
            "Command line: spades/bin/spades.py\t--test\t\n",
            "\n",
            "System information:\n",
            "  SPAdes version: 4.1.0\n",
            "  Python version: 3.11.13\n",
            "  OS: Linux-6.1.123+-x86_64-with-glibc2.35\n",
            "\n",
            "Output dir: /content/spades_test\n",
            "Mode: read error correction and assembling\n",
            "Debug mode is turned OFF\n",
            "\n",
            "Dataset parameters:\n",
            "  Standard mode\n",
            "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
            "  Reads:\n",
            "    Library number: 1, library type: paired-end\n",
            "      orientation: fr\n",
            "      left reads: ['/content/spades/share/spades/test_dataset/ecoli_1K_1.fq.gz']\n",
            "      right reads: ['/content/spades/share/spades/test_dataset/ecoli_1K_2.fq.gz']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "Read error correction parameters:\n",
            "  Iterations: 1\n",
            "  PHRED offset will be auto-detected\n",
            "  Corrected reads will be compressed\n",
            "Assembly parameters:\n",
            "  k: automatic selection based on read length\n",
            "  Repeat resolution is enabled\n",
            "  Mismatch careful mode is turned OFF\n",
            "  MismatchCorrector will be SKIPPED\n",
            "  Coverage cutoff is turned OFF\n",
            "  Assembly graph output will use GFA v1.2 format\n",
            "Other parameters:\n",
            "  Dir for temp files: /content/spades_test/tmp\n",
            "  Threads: 16\n",
            "  Memory limit (in Gb): 12\n",
            "\n",
            "\n",
            "======= SPAdes pipeline started. Log can be found here: /content/spades_test/spades.log\n",
            "\n",
            "/content/spades/share/spades/test_dataset/ecoli_1K_1.fq.gz: max reads length: 100\n",
            "/content/spades/share/spades/test_dataset/ecoli_1K_2.fq.gz: max reads length: 100\n",
            "\n",
            "Reads length: 100\n",
            "\n",
            "\n",
            "===== Before start started. \n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-hammer /content/spades_test/corrected/configs/config.info\n",
            "\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  :  76)   Starting BayesHammer, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  :  77)   Loading config from \"/content/spades_test/corrected/configs/config.info\"\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (main.cpp                  :  79)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (main.cpp                  :  87)   Trying to determine PHRED offset\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (main.cpp                  :  93)   Determined value is 33\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (hammer_tools.cpp          :  40)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (main.cpp                  : 114)   Size of aux. kmer data 24 bytes\n",
            "     === ITERATION 0 begins ===\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 16 files using 2 threads. This might take a while.\n",
            "  0:00:00.003     1M / 21M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.003     1M / 21M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99998 Gb\n",
            "  0:00:00.003     1M / 21M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 4194304\n",
            "  0:00:00.003  1153M / 1153M INFO   K-mer Splitting          (kmer_data.cpp             :  98)   Processing \"/content/spades/share/spades/test_dataset/ecoli_1K_1.fq.gz\"\n",
            "  0:00:00.040  1153M / 1153M INFO   K-mer Splitting          (kmer_data.cpp             :  98)   Processing \"/content/spades/share/spades/test_dataset/ecoli_1K_2.fq.gz\"\n",
            "  0:00:00.076  1153M / 1153M INFO   K-mer Splitting          (kmer_data.cpp             : 113)   Total 4108 reads processed\n",
            "  0:00:00.077     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.079     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 1974 kmers in total.\n",
            "  0:00:00.080     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.083     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 1974 kmers, 13944 bytes occupied (56.5106 bits per kmer).\n",
            "  0:00:00.083     1M / 21M   INFO   K-mer Counting           (kmer_data.cpp             : 355)   Arranging kmers in hash map order\n",
            "  0:00:00.085     1M / 21M   INFO    General                 (main.cpp                  : 149)   Clustering Hamming graph.\n",
            "  0:00:00.090     1M / 21M   INFO    General                 (main.cpp                  : 156)   Extracting clusters:\n",
            "  0:00:00.091     1M / 21M   INFO    General                 (concurrent_dsu.cpp        :  19)   Connecting to root\n",
            "  0:00:00.091     1M / 21M   INFO    General                 (concurrent_dsu.cpp        :  35)   Calculating counts\n",
            "  0:00:00.091     1M / 21M   INFO    General                 (concurrent_dsu.cpp        :  64)   Writing down entries\n",
            "  0:00:00.091     1M / 21M   INFO    General                 (main.cpp                  : 168)   Clustering done. Total clusters: 1960\n",
            "  0:00:00.092     1M / 21M   INFO   K-mer Counting           (kmer_data.cpp             : 372)   Collecting K-mer information, this takes a while.\n",
            "  0:00:00.092     1M / 21M   INFO   K-mer Counting           (kmer_data.cpp             : 378)   Processing \"/content/spades/share/spades/test_dataset/ecoli_1K_1.fq.gz\"\n",
            "  0:00:00.219     1M / 21M   INFO   K-mer Counting           (kmer_data.cpp             : 378)   Processing \"/content/spades/share/spades/test_dataset/ecoli_1K_2.fq.gz\"\n",
            "  0:00:00.361     1M / 21M   INFO   K-mer Counting           (kmer_data.cpp             : 385)   Collection done, postprocessing.\n",
            "  0:00:00.362     1M / 21M   INFO   K-mer Counting           (kmer_data.cpp             : 398)   There are 1974 kmers in total. Among them 0 (0%) are singletons.\n",
            "  0:00:00.362     1M / 21M   INFO    General                 (main.cpp                  : 174)   Subclustering Hamming graph\n",
            "  0:00:00.362     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 637)   Subclustering done. Total 0 non-read kmers were generated.\n",
            "  0:00:00.362     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 638)   Subclustering statistics:\n",
            "  0:00:00.363     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 639)     Total singleton hamming clusters: 1946. Among them 1946 (100%) are good\n",
            "  0:00:00.363     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 640)     Total singleton subclusters: 8. Among them 8 (100%) are good\n",
            "  0:00:00.363     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 641)     Total non-singleton subcluster centers: 10. Among them 10 (100%) are good\n",
            "  0:00:00.363     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 642)     Average size of non-trivial subcluster: 2.8 kmers\n",
            "  0:00:00.363     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 643)     Average number of sub-clusters per non-singleton cluster: 1.28571\n",
            "  0:00:00.363     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 644)     Total solid k-mers: 1964\n",
            "  0:00:00.363     1M / 21M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 645)     Substitution probabilities: (        0.875        0.125            0            0 )\n",
            "(            0            1            0            0 )\n",
            "(            0            0            1            0 )\n",
            "(            0            0        0.125        0.875 )\n",
            "  0:00:00.363     1M / 21M   INFO    General                 (main.cpp                  : 179)   Finished clustering.\n",
            "  0:00:00.363     1M / 21M   INFO    General                 (main.cpp                  : 198)   Starting solid k-mers expansion in 2 threads.\n",
            "  0:00:00.408     1M / 21M   INFO    General                 (main.cpp                  : 219)   Solid k-mers iteration 0 produced 0 new k-mers.\n",
            "  0:00:00.408     1M / 21M   INFO    General                 (main.cpp                  : 223)   Solid k-mers finalized\n",
            "  0:00:00.408     1M / 21M   INFO    General                 (hammer_tools.cpp          : 226)   Starting read correction in 2 threads.\n",
            "  0:00:00.408     1M / 21M   INFO    General                 (hammer_tools.cpp          : 239)   Correcting pair of reads: \"/content/spades/share/spades/test_dataset/ecoli_1K_1.fq.gz\" and \"/content/spades/share/spades/test_dataset/ecoli_1K_2.fq.gz\"\n",
            "  0:00:00.453    46M / 50M   INFO    General                 (hammer_tools.cpp          : 173)   Prepared batch 0 of 2054 reads.\n",
            "  0:00:00.468    46M / 50M   INFO    General                 (hammer_tools.cpp          : 180)   Processed batch 0\n",
            "  0:00:00.472    46M / 50M   INFO    General                 (hammer_tools.cpp          : 190)   Written batch 0\n",
            "  0:00:00.479     1M / 50M   INFO    General                 (hammer_tools.cpp          : 280)   Correction done. Changed 4 bases in 4 reads.\n",
            "  0:00:00.479     1M / 50M   INFO    General                 (hammer_tools.cpp          : 281)   Failed to correct 0 bases out of 353915.\n",
            "  0:00:00.480     1M / 50M   INFO    General                 (main.cpp                  : 256)   Saving corrected dataset description to /content/spades_test/corrected/corrected.yaml\n",
            "  0:00:00.480     1M / 50M   INFO    General                 (main.cpp                  : 263)   All done. Exiting.\n",
            "\n",
            "===== Read error correction finished. \n",
            "\n",
            "\n",
            "===== corrected reads compression started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/compress_all.py --input_file /content/spades_test/corrected/corrected.yaml --ext_python_modules_home /content/spades/share/spades --max_threads 16 --output_dir /content/spades_test/corrected --gzip_output\n",
            "\n",
            "== Compressing corrected reads (with pigz)\n",
            "== Files to compress: ['/content/spades_test/corrected/ecoli_1K_1.fq00.0_0.cor.fastq', '/content/spades_test/corrected/ecoli_1K_2.fq00.0_0.cor.fastq', '/content/spades_test/corrected/ecoli_1K__unpaired00.0_0.cor.fastq']\n",
            "== Files compression is finished\n",
            "== Dataset yaml file is updated\n",
            "\n",
            "===== corrected reads compression finished. \n",
            "\n",
            "\n",
            "===== Read error correction finished. \n",
            "\n",
            "\n",
            "===== Assembling started. \n",
            "\n",
            "\n",
            "===== K21 started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-core /content/spades_test/K21/configs/config.info\n",
            "\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/spades_test/K21/configs/config.info\"\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 102)   Starting SPAdes, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 103)   Maximum k-mer length: 128\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 104)   Assembling dataset (\"/content/spades_test/dataset.info\") with K=21\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 105)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (pipeline.cpp              : 234)   Two-step repeat resolution disabled\n",
            "  0:00:00.001     1M / 21M   INFO   GraphCore                (graph_core.hpp            : 689)   Graph created, vertex min_id: 3, edge min_id: 3\n",
            "  0:00:00.001     1M / 21M   INFO   GraphCore                (graph_core.hpp            : 690)   Vertex size: 48, edge size: 40\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.001     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Binary Read Conversion (id: read_conversion)\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (read_converter.cpp        :  78)   Converting reads to binary format for library #0 (takes a while)\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (read_converter.cpp        :  99)   Converting paired reads\n",
            "  0:00:00.035    81M / 81M   INFO    General                 (binary_converter.cpp      : 143)   2054 reads written\n",
            "  0:00:00.039    52M / 52M   INFO    General                 (read_converter.cpp        : 113)   Converting single reads\n",
            "  0:00:00.051    91M / 91M   INFO    General                 (binary_converter.cpp      : 143)   0 reads written\n",
            "  0:00:00.053    78M / 78M   INFO    General                 (read_converter.cpp        : 119)   Converting merged reads\n",
            "  0:00:00.056    91M / 91M   INFO    General                 (binary_converter.cpp      : 143)   0 reads written\n",
            "  0:00:00.058     1M / 36M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == de Bruijn graph construction (id: construction)\n",
            "  0:00:00.059     1M / 36M   INFO    General                 (construction.cpp          : 150)   Max read length 100\n",
            "  0:00:00.059     1M / 36M   INFO    General                 (construction.cpp          : 156)   Average read length 86.1526\n",
            "  0:00:00.059     1M / 36M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
            "  0:00:00.060     1M / 36M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.060     1M / 36M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.060     1M / 36M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99997 Gb\n",
            "  0:00:00.060     1M / 36M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 3355443\n",
            "  0:00:00.081     1M / 36M   INFO    General                 (kmer_splitters.hpp        : 134)   Used 8216 reads\n",
            "  0:00:00.081     1M / 36M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.084     1M / 36M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 984 kmers in total.\n",
            "  0:00:00.084     1M / 36M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
            "  0:00:00.085     1M / 36M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 453)   Building kmer index\n",
            "  0:00:00.085     1M / 36M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.085     1M / 36M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.085     1M / 36M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99997 Gb\n",
            "  0:00:00.085     1M / 36M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 3355443\n",
            "  0:00:00.088  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 202)   Used 984 kmers.\n",
            "  0:00:00.089     1M / 36M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.091     1M / 36M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 985 kmers in total.\n",
            "  0:00:00.091     1M / 36M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.094     1M / 36M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 985 kmers, 16424 bytes occupied (133.393 bits per kmer).\n",
            "  0:00:00.094     1M / 36M   INFO    General                 (kmer_index_builder.hpp    : 168)   Merging final buckets.\n",
            "  0:00:00.095     1M / 36M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 101)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.096     1M / 36M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 106)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.096     1M / 36M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
            "  0:00:00.096     1M / 36M   INFO    General                 (construction.cpp          : 298)   Early tip clipper length bound set as (RL - K)\n",
            "  0:00:00.096     1M / 36M   INFO   Early tip clipping       (early_simplification.hpp  :  48)   Early tip clipping\n",
            "  0:00:00.097     1M / 36M   INFO   Early tip clipping       (early_simplification.hpp  :  83)   #tipped junctions: 2\n",
            "  0:00:00.097     1M / 36M   INFO   Early tip clipping       (early_simplification.hpp  :  94)   Clipped tips: 2\n",
            "  0:00:00.097     1M / 36M   INFO   Early tip clipping       (early_simplification.hpp  :  50)   5 22-mers were removed by early tip clipper\n",
            "  0:00:00.097     1M / 36M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
            "  0:00:00.097     1M / 36M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 381)   Extracting unbranching paths\n",
            "  0:00:00.097     1M / 36M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 400)   Extracting unbranching paths finished. 1 sequences extracted\n",
            "  0:00:00.098     1M / 36M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 336)   Collecting perfect loops\n",
            "  0:00:00.098     1M / 36M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 369)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:00:00.098     1M / 36M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 586)   Sorting edges...\n",
            "  0:00:00.098     1M / 36M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 588)   Edges sorted\n",
            "  0:00:00.098     1M / 36M   INFO    General                 (debruijn_graph_constructor: 516)   Total 2 edges to create\n",
            "  0:00:00.098     1M / 36M   INFO    General                 (debruijn_graph_constructor: 519)   Collecting link records\n",
            "  0:00:00.098     1M / 36M   INFO    General                 (debruijn_graph_constructor: 521)   Ordering link records\n",
            "  0:00:00.098     1M / 36M   INFO    General                 (debruijn_graph_constructor: 524)   Sorting done\n",
            "  0:00:00.098     1M / 36M   INFO    General                 (debruijn_graph_constructor: 537)   Sorting LinkRecords...\n",
            "  0:00:00.098     1M / 36M   INFO    General                 (debruijn_graph_constructor: 540)   LinkRecords sorted\n",
            "  0:00:00.098     1M / 36M   INFO    General                 (debruijn_graph_constructor: 542)   Total 2 vertices to create\n",
            "  0:00:00.098     1M / 36M   INFO    General                 (debruijn_graph_constructor: 545)   Connecting the graph\n",
            "  0:00:00.099     1M / 36M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
            "  0:00:00.099     1M / 36M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.101     1M / 36M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 984 kmers, 16392 bytes occupied (133.268 bits per kmer).\n",
            "  0:00:00.101     1M / 36M   INFO    General                 (coverage_hash_map_builder.:  49)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:00.120     1M / 36M   INFO    General                 (construction.cpp          : 427)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:00.121     1M / 36M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
            "  0:00:00.121     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 182)   Kmer coverage valley at: 0\n",
            "  0:00:00.121     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 202)   K-mer histogram maximum: 348\n",
            "  0:00:00.122     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 238)   Estimated median coverage: 348. Coverage mad: 62.2692\n",
            "  0:00:00.122     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 260)   Fitting coverage model\n",
            "  0:00:00.184     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 2\n",
            "  0:00:00.322     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 4\n",
            "  0:00:00.541     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 8\n",
            "  0:00:00.790     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 16\n",
            "  0:00:01.239     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 32\n",
            "  0:00:02.365     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 310)   Fitted mean coverage: 342.845. Fitted coverage std. dev: 68.3997\n",
            "  0:00:02.366     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 335)   Probability of erroneous kmer at valley: 0.999994\n",
            "  0:00:02.366     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 359)   Preliminary threshold calculated as: 355\n",
            "  0:00:02.366     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 363)   Threshold adjusted to: 171\n",
            "  0:00:02.366     1M / 36M   INFO    General                 (kmer_coverage_model.cpp   : 376)   Estimated genome size (ignoring repeats): 749\n",
            "  0:00:02.366     1M / 36M   INFO    General                 (genomic_info_filler.cpp   :  56)   Mean coverage was calculated as 342.845\n",
            "  0:00:02.366     1M / 36M   INFO    General                 (genomic_info_filler.cpp   :  71)   EC coverage threshold value was calculated as 171\n",
            "  0:00:02.366     1M / 36M   INFO    General                 (genomic_info_filler.cpp   :  72)   Trusted kmer low bound: 0\n",
            "  0:00:02.367     1M / 36M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Raw Simplification (id: raw_simplification)\n",
            "  0:00:02.367     1M / 36M   INFO    General                 (simplification.cpp        : 129)   PROCEDURE == Initial cleaning\n",
            "  0:00:02.367     1M / 36M   INFO    General                 (graph_simplification.hpp  : 674)   Flanking coverage based disconnection disabled\n",
            "  0:00:02.367     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Self conjugate edge remover\n",
            "  0:00:02.367     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:02.367     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial tip clipper\n",
            "  0:00:02.367     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial tip clipper triggered 0 times\n",
            "  0:00:02.367     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial ec remover\n",
            "  0:00:02.367     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial ec remover triggered 0 times\n",
            "  0:00:02.367     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial isolated edge remover\n",
            "  0:00:02.367     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial isolated edge remover triggered 0 times\n",
            "  0:00:02.367     1M / 36M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification (id: simplification)\n",
            "  0:00:02.367     1M / 36M   INFO    General                 (simplification.cpp        : 397)   Graph simplification started\n",
            "  0:00:02.367     1M / 36M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.368     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.368     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.368     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:02.368     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.369     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.369     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.369     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.370     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.370     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.370     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.371     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.371     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.371     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.371     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 189)   PROCEDURE == Post simplification\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (graph_simplification.hpp  : 455)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 348)   Disrupting self-conjugate edges\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Removing isolated edges\n",
            "  0:00:02.372     1M / 36M   INFO   Simplification           (parallel_processing.hpp   : 171)   Removing isolated edges triggered 0 times\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 508)   After simplification:\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 509)     Average coverage = 273.35\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 510)     Total length = 979\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 511)     Median edge length: 979\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 512)     Edges: 2\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (simplification.cpp        : 513)     Vertices: 4\n",
            "  0:00:02.372     1M / 36M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:02.372     1M / 36M   INFO    General                 (read_converter.cpp        : 135)   Outputting contigs to \"/content/spades_test/K21/simplified_contigs\"\n",
            "  0:00:02.377     5M / 36M   INFO    General                 (binary_converter.cpp      : 143)   1 reads written\n",
            "  0:00:02.380     1M / 36M   INFO    General                 (pipeline.cpp              : 292)   SPAdes finished\n",
            "  0:00:02.380     1M / 36M   INFO    General                 (main.cpp                  : 131)   Assembling time: 0 hours 0 minutes 2 seconds\n",
            "\n",
            "===== K21 finished. \n",
            "\n",
            "\n",
            "===== K33 started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-core /content/spades_test/K33/configs/config.info\n",
            "\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/spades_test/K33/configs/config.info\"\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 102)   Starting SPAdes, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 103)   Maximum k-mer length: 128\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 104)   Assembling dataset (\"/content/spades_test/dataset.info\") with K=33\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 105)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (pipeline.cpp              : 234)   Two-step repeat resolution disabled\n",
            "  0:00:00.001     1M / 21M   INFO   GraphCore                (graph_core.hpp            : 689)   Graph created, vertex min_id: 3, edge min_id: 3\n",
            "  0:00:00.001     1M / 21M   INFO   GraphCore                (graph_core.hpp            : 690)   Vertex size: 48, edge size: 40\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.001     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Binary Read Conversion (id: read_conversion)\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (read_converter.cpp        :  57)   Binary reads detected\n",
            "  0:00:00.001     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == de Bruijn graph construction (id: construction)\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (construction.cpp          : 115)   Contigs from previous K will be used: /content/spades_test/K21/simplified_contigs\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (construction.cpp          : 150)   Max read length 100\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (construction.cpp          : 156)   Average read length 86.1526\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99997 Gb\n",
            "  0:00:00.002     1M / 21M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1677721\n",
            "  0:00:00.022     1M / 21M   INFO    General                 (kmer_splitters.hpp        : 134)   Used 8218 reads\n",
            "  0:00:00.022     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.024     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 972 kmers in total.\n",
            "  0:00:00.024     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
            "  0:00:00.025     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 453)   Building kmer index\n",
            "  0:00:00.025     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.025     1M / 21M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.025     1M / 21M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99996 Gb\n",
            "  0:00:00.025     1M / 21M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1677721\n",
            "  0:00:00.028  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 202)   Used 972 kmers.\n",
            "  0:00:00.028     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.031     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 973 kmers in total.\n",
            "  0:00:00.031     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.033     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 973 kmers, 16392 bytes occupied (134.775 bits per kmer).\n",
            "  0:00:00.033     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 168)   Merging final buckets.\n",
            "  0:00:00.034     1M / 21M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 101)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.035     1M / 21M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 106)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.035     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
            "  0:00:00.035     1M / 21M   INFO    General                 (construction.cpp          : 298)   Early tip clipper length bound set as (RL - K)\n",
            "  0:00:00.035     1M / 21M   INFO   Early tip clipping       (early_simplification.hpp  :  48)   Early tip clipping\n",
            "  0:00:00.035     1M / 21M   INFO   Early tip clipping       (early_simplification.hpp  :  83)   #tipped junctions: 2\n",
            "  0:00:00.035     1M / 21M   INFO   Early tip clipping       (early_simplification.hpp  :  94)   Clipped tips: 2\n",
            "  0:00:00.035     1M / 21M   INFO   Early tip clipping       (early_simplification.hpp  :  50)   5 34-mers were removed by early tip clipper\n",
            "  0:00:00.035     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
            "  0:00:00.035     1M / 21M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 381)   Extracting unbranching paths\n",
            "  0:00:00.036     1M / 21M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 400)   Extracting unbranching paths finished. 1 sequences extracted\n",
            "  0:00:00.036     1M / 21M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 336)   Collecting perfect loops\n",
            "  0:00:00.036     1M / 21M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 369)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:00:00.036     1M / 21M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 586)   Sorting edges...\n",
            "  0:00:00.036     1M / 21M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 588)   Edges sorted\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (debruijn_graph_constructor: 516)   Total 2 edges to create\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (debruijn_graph_constructor: 519)   Collecting link records\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (debruijn_graph_constructor: 521)   Ordering link records\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (debruijn_graph_constructor: 524)   Sorting done\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (debruijn_graph_constructor: 537)   Sorting LinkRecords...\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (debruijn_graph_constructor: 540)   LinkRecords sorted\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (debruijn_graph_constructor: 542)   Total 2 vertices to create\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (debruijn_graph_constructor: 545)   Connecting the graph\n",
            "  0:00:00.036     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
            "  0:00:00.036     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.038     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 972 kmers, 16392 bytes occupied (134.914 bits per kmer).\n",
            "  0:00:00.039     1M / 21M   INFO    General                 (coverage_hash_map_builder.:  49)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:00.055     1M / 21M   INFO    General                 (construction.cpp          : 427)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:00.056     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
            "  0:00:00.056     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 182)   Kmer coverage valley at: 0\n",
            "  0:00:00.056     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 202)   K-mer histogram maximum: 17\n",
            "  0:00:00.056     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 238)   Estimated median coverage: 17. Coverage mad: 11.8608\n",
            "  0:00:00.056     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 260)   Fitting coverage model\n",
            "  0:00:00.084     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 2\n",
            "  0:00:00.133     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 4\n",
            "  0:00:00.215     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 8\n",
            "  0:00:00.300     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 16\n",
            "  0:00:00.452     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 32\n",
            "  0:00:00.750     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 64\n",
            "  0:00:01.340     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 128\n",
            "  0:00:02.073     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 310)   Fitted mean coverage: 108.181. Fitted coverage std. dev: 65.0594\n",
            "  0:00:02.073     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 335)   Probability of erroneous kmer at valley: 0.0555142\n",
            "  0:00:02.073     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 359)   Preliminary threshold calculated as: 3\n",
            "  0:00:02.073     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 363)   Threshold adjusted to: 3\n",
            "  0:00:02.073     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 376)   Estimated genome size (ignoring repeats): 400\n",
            "  0:00:02.074     1M / 21M   INFO    General                 (genomic_info_filler.cpp   :  56)   Mean coverage was calculated as 108.181\n",
            "  0:00:02.074     1M / 21M   INFO    General                 (genomic_info_filler.cpp   :  71)   EC coverage threshold value was calculated as 3\n",
            "  0:00:02.074     1M / 21M   INFO    General                 (genomic_info_filler.cpp   :  72)   Trusted kmer low bound: 0\n",
            "  0:00:02.074     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Raw Simplification (id: raw_simplification)\n",
            "  0:00:02.074     1M / 21M   INFO    General                 (simplification.cpp        : 129)   PROCEDURE == Initial cleaning\n",
            "  0:00:02.074     1M / 21M   INFO    General                 (graph_simplification.hpp  : 674)   Flanking coverage based disconnection disabled\n",
            "  0:00:02.074     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Self conjugate edge remover\n",
            "  0:00:02.074     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial tip clipper\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial tip clipper triggered 0 times\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial ec remover\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial ec remover triggered 0 times\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial isolated edge remover\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial isolated edge remover triggered 0 times\n",
            "  0:00:02.075     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification (id: simplification)\n",
            "  0:00:02.075     1M / 21M   INFO    General                 (simplification.cpp        : 397)   Graph simplification started\n",
            "  0:00:02.075     1M / 21M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.075     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.075     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.076     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.076     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.076     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.077     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.077     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.077     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.078     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.078     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.078     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.079     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.079     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:02.079     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.079     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.079     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.079     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.079     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.079     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.079     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:02.079     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.080     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.080     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.080     1M / 21M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.081     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
            "  0:00:02.081     1M / 21M   INFO    General                 (simplification.cpp        : 189)   PROCEDURE == Post simplification\n",
            "  0:00:02.081     1M / 21M   INFO    General                 (graph_simplification.hpp  : 455)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:02.081     1M / 21M   INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:02.081     1M / 21M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.081     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.082     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.082     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.082     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.082     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.082     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.082     1M / 21M   INFO    General                 (simplification.cpp        : 348)   Disrupting self-conjugate edges\n",
            "  0:00:02.082     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Removing isolated edges\n",
            "  0:00:02.082     1M / 21M   INFO   Simplification           (parallel_processing.hpp   : 171)   Removing isolated edges triggered 0 times\n",
            "  0:00:02.082     1M / 21M   INFO    General                 (simplification.cpp        : 508)   After simplification:\n",
            "  0:00:02.082     1M / 21M   INFO    General                 (simplification.cpp        : 509)     Average coverage = 225.839\n",
            "  0:00:02.082     1M / 21M   INFO    General                 (simplification.cpp        : 510)     Total length = 967\n",
            "  0:00:02.082     1M / 21M   INFO    General                 (simplification.cpp        : 511)     Median edge length: 967\n",
            "  0:00:02.082     1M / 21M   INFO    General                 (simplification.cpp        : 512)     Edges: 2\n",
            "  0:00:02.082     1M / 21M   INFO    General                 (simplification.cpp        : 513)     Vertices: 4\n",
            "  0:00:02.082     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:02.083     1M / 21M   INFO    General                 (read_converter.cpp        : 135)   Outputting contigs to \"/content/spades_test/K33/simplified_contigs\"\n",
            "  0:00:02.087     5M / 21M   INFO    General                 (binary_converter.cpp      : 143)   1 reads written\n",
            "  0:00:02.090     1M / 21M   INFO    General                 (pipeline.cpp              : 292)   SPAdes finished\n",
            "  0:00:02.090     1M / 21M   INFO    General                 (main.cpp                  : 131)   Assembling time: 0 hours 0 minutes 2 seconds\n",
            "\n",
            "===== K33 finished. \n",
            "\n",
            "\n",
            "===== K55 started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-core /content/spades_test/K55/configs/config.info\n",
            "\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/spades_test/K55/configs/config.info\"\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 102)   Starting SPAdes, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 103)   Maximum k-mer length: 128\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 104)   Assembling dataset (\"/content/spades_test/dataset.info\") with K=55\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (main.cpp                  : 105)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
            "  0:00:00.000     1M / 21M   INFO    General                 (pipeline.cpp              : 234)   Two-step repeat resolution disabled\n",
            "  0:00:00.000     1M / 21M   INFO   GraphCore                (graph_core.hpp            : 689)   Graph created, vertex min_id: 3, edge min_id: 3\n",
            "  0:00:00.001     1M / 21M   INFO   GraphCore                (graph_core.hpp            : 690)   Vertex size: 48, edge size: 40\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (pipeline.cpp              : 245)   Will need read mapping, kmer mapper will be attached\n",
            "  0:00:00.001     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Binary Read Conversion (id: read_conversion)\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (read_converter.cpp        :  57)   Binary reads detected\n",
            "  0:00:00.001     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == de Bruijn graph construction (id: construction)\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (construction.cpp          : 115)   Contigs from previous K will be used: /content/spades_test/K33/simplified_contigs\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (construction.cpp          : 150)   Max read length 100\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (construction.cpp          : 156)   Average read length 86.1526\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99997 Gb\n",
            "  0:00:00.001     1M / 21M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1677721\n",
            "  0:00:00.016     1M / 21M   INFO    General                 (kmer_splitters.hpp        : 134)   Used 8218 reads\n",
            "  0:00:00.016     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.018     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 949 kmers in total.\n",
            "  0:00:00.018     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
            "  0:00:00.019     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 453)   Building kmer index\n",
            "  0:00:00.019     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.019     1M / 21M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.019     1M / 21M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99996 Gb\n",
            "  0:00:00.019     1M / 21M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1677721\n",
            "  0:00:00.022  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 202)   Used 949 kmers.\n",
            "  0:00:00.022     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.024     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 951 kmers in total.\n",
            "  0:00:00.024     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.027     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 951 kmers, 16384 bytes occupied (137.825 bits per kmer).\n",
            "  0:00:00.027     1M / 21M   INFO    General                 (kmer_index_builder.hpp    : 168)   Merging final buckets.\n",
            "  0:00:00.028     1M / 21M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 101)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.029     1M / 21M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 106)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.029     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
            "  0:00:00.029     1M / 21M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 381)   Extracting unbranching paths\n",
            "  0:00:00.029     1M / 21M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 400)   Extracting unbranching paths finished. 4 sequences extracted\n",
            "  0:00:00.030     1M / 21M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 336)   Collecting perfect loops\n",
            "  0:00:00.030     1M / 21M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 369)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:00:00.030     1M / 21M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 586)   Sorting edges...\n",
            "  0:00:00.030     1M / 21M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 588)   Edges sorted\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (debruijn_graph_constructor: 516)   Total 8 edges to create\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (debruijn_graph_constructor: 519)   Collecting link records\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (debruijn_graph_constructor: 521)   Ordering link records\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (debruijn_graph_constructor: 524)   Sorting done\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (debruijn_graph_constructor: 537)   Sorting LinkRecords...\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (debruijn_graph_constructor: 540)   LinkRecords sorted\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (debruijn_graph_constructor: 542)   Total 6 vertices to create\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (debruijn_graph_constructor: 545)   Connecting the graph\n",
            "  0:00:00.030     1M / 21M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
            "  0:00:00.030     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.032     1M / 21M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 949 kmers, 16392 bytes occupied (138.183 bits per kmer).\n",
            "  0:00:00.033     1M / 21M   INFO    General                 (coverage_hash_map_builder.:  49)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:00.044     1M / 21M   INFO    General                 (construction.cpp          : 427)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:00.045     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
            "  0:00:00.045     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 182)   Kmer coverage valley at: 0\n",
            "  0:00:00.045     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 202)   K-mer histogram maximum: 6\n",
            "  0:00:00.045     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 238)   Estimated median coverage: 6. Coverage mad: 4.4478\n",
            "  0:00:00.045     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 260)   Fitting coverage model\n",
            "  0:00:00.051     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 2\n",
            "  0:00:00.071     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 4\n",
            "  0:00:00.135     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 8\n",
            "  0:00:00.222     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 16\n",
            "  0:00:00.300     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 310)   Fitted mean coverage: 13.7375. Fitted coverage std. dev: 9.60863\n",
            "  0:00:00.300     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 335)   Probability of erroneous kmer at valley: 2.02732e-07\n",
            "  0:00:00.300     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 359)   Preliminary threshold calculated as: 1\n",
            "  0:00:00.300     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 363)   Threshold adjusted to: 1\n",
            "  0:00:00.300     1M / 21M   INFO    General                 (kmer_coverage_model.cpp   : 376)   Estimated genome size (ignoring repeats): 238\n",
            "  0:00:00.300     1M / 21M   INFO    General                 (genomic_info_filler.cpp   :  56)   Mean coverage was calculated as 13.7375\n",
            "  0:00:00.301     1M / 21M   INFO    General                 (genomic_info_filler.cpp   :  71)   EC coverage threshold value was calculated as 1\n",
            "  0:00:00.301     1M / 21M   INFO    General                 (genomic_info_filler.cpp   :  72)   Trusted kmer low bound: 0\n",
            "  0:00:00.301     1M / 21M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Gap Closer (id: early_gapcloser)\n",
            "  0:00:00.314    14M / 23M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.314    14M / 23M   INFO    General                 (gap_closer.cpp            : 102)   Total edges in tip neighborhood: 7 out of 8, length: 1520\n",
            "  0:00:00.314    14M / 23M   INFO    General                 (edge_index.hpp            : 196)   Using small index (max_id = 11)\n",
            "  0:00:00.314    14M / 23M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.315    14M / 23M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 949 kmers, 1456 bytes occupied (12.274 bits per kmer).\n",
            "  0:00:00.315    14M / 23M   INFO    General                 (edge_index_builders.hpp   : 266)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:00.315    14M / 23M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:00.322    14M / 23M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 2054 reads processed\n",
            "  0:00:00.327    14M / 23M   INFO    General                 (gap_closer.cpp            : 491)   Initializing gap closer\n",
            "  0:00:00.328    14M / 23M   INFO   GapCloser                (gap_closer.cpp            : 406)   Collecting gap candidates\n",
            "  0:00:00.328    14M / 23M   INFO   GapCloser                (gap_closer.cpp            : 410)   Total 3 tips collected, total 4 connection candidates\n",
            "  0:00:00.328    14M / 23M   INFO   GapCloser                (gap_closer.cpp            : 431)   Closing short gaps complete: filled 0 gaps after checking 4 candidates\n",
            "  0:00:00.328    14M / 23M   INFO    General                 (gap_closer.cpp            : 495)   Gap closer done\n",
            "  0:00:00.329     1M / 23M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Raw Simplification (id: raw_simplification)\n",
            "  0:00:00.330     1M / 23M   INFO    General                 (simplification.cpp        : 129)   PROCEDURE == Initial cleaning\n",
            "  0:00:00.330     1M / 23M   INFO    General                 (graph_simplification.hpp  : 674)   Flanking coverage based disconnection disabled\n",
            "  0:00:00.330     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Self conjugate edge remover\n",
            "  0:00:00.330     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 171)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:00.330     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial tip clipper\n",
            "  0:00:00.330     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial tip clipper triggered 0 times\n",
            "  0:00:00.330     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial ec remover\n",
            "  0:00:00.330     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial ec remover triggered 0 times\n",
            "  0:00:00.330     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Initial isolated edge remover\n",
            "  0:00:00.330     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 171)   Initial isolated edge remover triggered 1 times\n",
            "  0:00:00.330     1M / 23M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification (id: simplification)\n",
            "  0:00:00.330     1M / 23M   INFO    General                 (simplification.cpp        : 397)   Graph simplification started\n",
            "  0:00:00.331     1M / 23M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:00.331     1M / 23M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:00.331     1M / 23M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 1 times\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.331     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.331     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.332     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.332     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.332     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:00.332     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.333     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.333     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.333     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.334     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.334     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.334     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.335     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.335     1M / 24M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:00.335     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.335     1M / 24M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Gap Closer (id: late_gapcloser)\n",
            "  0:00:00.340    14M / 24M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.340    14M / 24M   INFO    General                 (gap_closer.cpp            : 102)   Total edges in tip neighborhood: 2 out of 2, length: 1890\n",
            "  0:00:00.340    14M / 24M   INFO    General                 (edge_index.hpp            : 196)   Using small index (max_id = 11)\n",
            "  0:00:00.340    14M / 24M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.341    14M / 24M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 945 kmers, 1456 bytes occupied (12.3259 bits per kmer).\n",
            "  0:00:00.341    14M / 24M   INFO    General                 (edge_index_builders.hpp   : 266)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:00.341    14M / 24M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:00.348    14M / 24M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 2054 reads processed\n",
            "  0:00:00.354    14M / 24M   INFO    General                 (gap_closer.cpp            : 491)   Initializing gap closer\n",
            "  0:00:00.354    14M / 24M   INFO   GapCloser                (gap_closer.cpp            : 406)   Collecting gap candidates\n",
            "  0:00:00.354    14M / 24M   INFO   GapCloser                (gap_closer.cpp            : 410)   Total 0 tips collected, total 0 connection candidates\n",
            "  0:00:00.354    14M / 24M   INFO   GapCloser                (gap_closer.cpp            : 431)   Closing short gaps complete: filled 0 gaps after checking 0 candidates\n",
            "  0:00:00.354    14M / 24M   INFO    General                 (gap_closer.cpp            : 495)   Gap closer done\n",
            "  0:00:00.356     1M / 24M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
            "  0:00:00.356     1M / 24M   INFO    General                 (simplification.cpp        : 189)   PROCEDURE == Post simplification\n",
            "  0:00:00.356     1M / 24M   INFO    General                 (graph_simplification.hpp  : 455)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:00.356     1M / 24M   INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:00.356     1M / 24M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:00.356     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.356     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.356     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.356     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.357     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:00.357     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:00.357     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:00.357     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (simplification.cpp        : 348)   Disrupting self-conjugate edges\n",
            "  0:00:00.357     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Removing isolated edges\n",
            "  0:00:00.357     1M / 24M   INFO   Simplification           (parallel_processing.hpp   : 171)   Removing isolated edges triggered 0 times\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (simplification.cpp        : 508)   After simplification:\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (simplification.cpp        : 509)     Average coverage = 140.62\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (simplification.cpp        : 510)     Total length = 945\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (simplification.cpp        : 511)     Median edge length: 945\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (simplification.cpp        : 512)     Edges: 2\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (simplification.cpp        : 513)     Vertices: 4\n",
            "  0:00:00.357     1M / 24M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (graph_pack_helpers.cpp    :  44)   Index refill\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (edge_index.hpp            : 175)   Using small index (max_id = 11)\n",
            "  0:00:00.357     1M / 24M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.357     1M / 24M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 945 kmers, 1456 bytes occupied (12.3259 bits per kmer).\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (edge_index_builders.hpp   : 253)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (graph_pack_helpers.cpp    :  54)   Normalizing k-mer map. Total 0 kmers to process\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (kmer_mapper.hpp           :  87)   Total 0 kmers with non-trivial roots\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (graph_pack_helpers.cpp    :  56)   Normalizing done\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (mismatch_correction.cpp   : 392)   Collect potential mismatches\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (mismatch_correction.cpp   : 193)   Total 0 edges (out of 2) with 0 potential mismatch positions (-nan positions per edge)\n",
            "  0:00:00.357     1M / 24M   INFO    General                 (mismatch_correction.cpp   : 394)   Potential mismatches collected\n",
            "  0:00:00.358     1M / 24M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 8216 reads processed\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (mismatch_correction.cpp   : 387)   All edges processed\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (mismatch_correction.cpp   : 442)   Corrected 0 nucleotides\n",
            "  0:00:00.370     1M / 24M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (contig_output.hpp         :  20)   Outputting contigs to \"/content/spades_test/K55/before_rr.fasta\"\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (contig_output_stage.cpp   : 155)   Writing GFA graph to \"/content/spades_test/K55/assembly_graph_after_simplification.gfa\"\n",
            "  0:00:00.370     1M / 24M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (graph_pack_helpers.cpp    :  54)   Normalizing k-mer map. Total 0 kmers to process\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (graph_pack_helpers.cpp    :  56)   Normalizing done\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (pair_info_count.cpp       : 157)   Min edge length for estimation: 945\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (pair_info_count.cpp       : 168)   Estimating insert size for library #0\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (pair_info_count.cpp       :  41)   Selecting usual mapper\n",
            "  0:00:00.370     1M / 24M   INFO    General                 (paired_info_utils.cpp     :  87)   Estimating insert size (takes a while)\n",
            "  0:00:00.406    49M / 60M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:00.467    49M / 60M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 2054 reads processed\n",
            "  0:00:00.687    49M / 60M   INFO    General                 (paired_info_utils.cpp     : 104)   Edge pairs: 2\n",
            "  0:00:00.688    49M / 60M   INFO    General                 (paired_info_utils.cpp     : 106)   1636 paired reads (79.6495% of all) aligned to long edges\n",
            "  0:00:00.688     1M / 60M   INFO    General                 (pair_info_count.cpp       : 188)     Insert size = 214.696, deviation = 10.4821, left quantile = 201, right quantile = 228, read length = 100\n",
            "  0:00:00.688     1M / 60M   INFO    General                 (pair_info_count.cpp       : 203)   Filtering data for library #0\n",
            "  0:00:00.688     1M / 60M   INFO    General                 (pair_info_count.cpp       :  41)   Selecting usual mapper\n",
            "  0:00:00.688     1M / 60M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:00.699     1M / 60M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 2054 reads processed\n",
            "  0:00:00.700     1M / 60M   INFO    General                 (pair_info_count.cpp       : 207)   Mapping library #0\n",
            "  0:00:00.700     1M / 60M   INFO    General                 (pair_info_count.cpp       : 209)   Mapping paired reads (takes a while)\n",
            "  0:00:00.700     1M / 60M   INFO    General                 (pair_info_count.cpp       :  41)   Selecting usual mapper\n",
            "  0:00:00.700     1M / 60M   INFO    General                 (paired_info_utils.cpp     : 138)   Left insert size quantile 201, right insert size quantile 228, filtering threshold 2, rounding threshold 0\n",
            "  0:00:00.706    14M / 60M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:00.725    14M / 60M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 2054 reads processed\n",
            "  0:00:00.734     1M / 60M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Distance Estimation (id: distance_estimation)\n",
            "  0:00:00.735     1M / 60M   INFO    General                 (distance_estimation.cpp   :  42)   Processing library #0\n",
            "  0:00:00.740     1M / 60M   INFO    General                 (distance_estimation_utils.:  23)   Estimating distances\n",
            "  0:00:00.740     1M / 60M   INFO   DistanceEstimator        (distance_estimation.hpp   : 132)   Using SIMPLE distance estimator\n",
            "  0:00:00.761     1M / 60M   INFO    General                 (distance_estimation_utils.: 131)   Refining clustered pair information\n",
            "  0:00:00.761     1M / 60M   INFO    General                 (distance_estimation_utils.: 133)   The refining of clustered pair information has been finished\n",
            "  0:00:00.761     1M / 60M   INFO    General                 (distance_estimation_utils.: 135)   Improving paired information\n",
            "  0:00:00.780     1M / 60M   INFO   PairInfoImprover         (pair_info_improver.hpp    :  78)   Paired info stats: missing = 0; contradictional = 0\n",
            "  0:00:00.795     1M / 60M   INFO   PairInfoImprover         (pair_info_improver.hpp    :  78)   Paired info stats: missing = 0; contradictional = 0\n",
            "  0:00:00.795     1M / 60M   INFO    General                 (distance_estimation_utils.:  80)   Filling scaffolding index\n",
            "  0:00:00.795     1M / 60M   INFO    General                 (distance_estimation_utils.:  23)   Estimating distances\n",
            "  0:00:00.795     1M / 60M   INFO   DistanceEstimator        (distance_estimation.hpp   : 132)   Using SMOOTHING distance estimator\n",
            "  0:00:00.805     1M / 60M   INFO    General                 (distance_estimation.cpp   :  51)   Clearing raw paired index\n",
            "  0:00:00.805     1M / 60M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
            "  0:00:00.805     1M / 60M   INFO    General                 (repeat_resolving.cpp      :  88)   Using Path-Extend repeat resolving\n",
            "  0:00:00.805     1M / 60M   INFO    General                 (launcher.cpp              : 603)   ExSPAnder repeat resolving tool started\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 140.62 calculated of length 945\n",
            "  0:00:00.806     1M / 60M   WARN    General                 (launcher.cpp              : 180)   Your data seems to have high uniform coverage depth. It is strongly recommended to use --isolate option.\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (launcher.cpp              : 420)   Creating main extenders, unique edge length = 2000\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (extenders_logic.cpp       : 341)   Estimated coverage of library #0 is 140.62\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (extenders_logic.cpp       : 352)   Creating extender; library index size: 2\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (extenders_logic.cpp       : 341)   Estimated coverage of library #0 is 140.62\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (extenders_logic.cpp       : 352)   Creating extender; library index size: 2\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (extenders_logic.cpp       : 550)   Using 1 paired-end library\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (extenders_logic.cpp       : 551)   Using 1 paired-end scaffolding library\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (extenders_logic.cpp       : 552)   Using 0 single read libraries\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (launcher.cpp              : 449)   Total number of extenders is 3\n",
            "  0:00:00.806     1M / 60M   INFO    General                 (launcher.cpp              : 252)   Finalizing paths\n",
            "  0:00:00.807     1M / 60M   INFO    General                 (launcher.cpp              : 254)   Deduplicating paths\n",
            "  0:00:00.807     1M / 60M   INFO    General                 (launcher.cpp              : 258)   Paths deduplicated\n",
            "  0:00:00.807     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
            "  0:00:00.807     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
            "  0:00:00.807     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
            "  0:00:00.807     1M / 60M   INFO   OverlapRemover           (overlap_remover.hpp       : 117)   Marking start/end overlaps\n",
            "  0:00:00.807     1M / 60M   INFO   OverlapRemover           (overlap_remover.hpp       : 120)   Marking remaining overlaps\n",
            "  0:00:00.807     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
            "  0:00:00.807     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
            "  0:00:00.807     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
            "  0:00:00.807     1M / 60M   INFO    General                 (launcher.cpp              : 275)   Paths finalized\n",
            "  0:00:00.807     1M / 60M   INFO    General                 (launcher.cpp              : 456)   Closing gaps in paths\n",
            "  0:00:00.807     1M / 60M   INFO    General                 (launcher.cpp              : 486)   Gap closing completed\n",
            "  0:00:00.807     1M / 60M   INFO    General                 (launcher.cpp              : 304)   Traversing tandem repeats\n",
            "  0:00:00.807     1M / 60M   INFO    General                 (launcher.cpp              : 314)   Traversed 0 loops\n",
            "  0:00:00.815     1M / 60M   INFO    General                 (launcher.cpp              : 252)   Finalizing paths\n",
            "  0:00:00.815     1M / 60M   INFO    General                 (launcher.cpp              : 254)   Deduplicating paths\n",
            "  0:00:00.815     1M / 60M   INFO    General                 (launcher.cpp              : 258)   Paths deduplicated\n",
            "  0:00:00.815     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
            "  0:00:00.815     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
            "  0:00:00.815     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
            "  0:00:00.815     1M / 60M   INFO   OverlapRemover           (overlap_remover.hpp       : 117)   Marking start/end overlaps\n",
            "  0:00:00.815     1M / 60M   INFO   OverlapRemover           (overlap_remover.hpp       : 120)   Marking remaining overlaps\n",
            "  0:00:00.815     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
            "  0:00:00.815     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
            "  0:00:00.815     1M / 60M   INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
            "  0:00:00.815     1M / 60M   INFO    General                 (launcher.cpp              : 275)   Paths finalized\n",
            "  0:00:00.815     1M / 60M   INFO    General                 (launcher.cpp              : 666)   ExSPAnder repeat resolving tool finished\n",
            "  0:00:00.816     1M / 60M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:00.823     1M / 60M   INFO    General                 (contig_output.hpp         :  20)   Outputting contigs to \"/content/spades_test/K55/before_rr.fasta\"\n",
            "  0:00:00.824     1M / 60M   INFO    General                 (contig_output_stage.cpp   : 155)   Writing GFA graph to \"/content/spades_test/K55/assembly_graph_with_scaffolds.gfa\"\n",
            "  0:00:00.824     1M / 60M   INFO    General                 (contig_output_stage.cpp   : 169)   Outputting FastG graph to \"/content/spades_test/K55/assembly_graph.fastg\"\n",
            "  0:00:00.824     1M / 60M   INFO    General                 (contig_output_stage.cpp   : 200)   Breaking scaffolds\n",
            "  0:00:00.824     1M / 60M   INFO    General                 (contig_output_stage.cpp   : 101)   Outputting contigs to /content/spades_test/K55/final_contigs.fasta\n",
            "  0:00:00.824     1M / 60M   INFO    General                 (contig_output_stage.cpp   : 107)   Outputting FastG paths to /content/spades_test/K55/final_contigs.paths\n",
            "  0:00:00.825     1M / 60M   INFO    General                 (contig_output_stage.cpp   : 101)   Outputting contigs to /content/spades_test/K55/scaffolds.fasta\n",
            "  0:00:00.825     1M / 60M   INFO    General                 (contig_output_stage.cpp   : 107)   Outputting FastG paths to /content/spades_test/K55/scaffolds.paths\n",
            "  0:00:00.825     1M / 60M   INFO    General                 (contig_output_stage.cpp   : 114)   Populating GFA with scaffold paths\n",
            "  0:00:00.825     1M / 60M   INFO    General                 (pipeline.cpp              : 292)   SPAdes finished\n",
            "  0:00:00.825     1M / 60M   INFO    General                 (main.cpp                  : 131)   Assembling time: 0 hours 0 minutes 0 seconds\n",
            "\n",
            "===== K55 finished. \n",
            "\n",
            "\n",
            "===== Copy files started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/copy_files.py /content/spades_test/K55/before_rr.fasta /content/spades_test/before_rr.fasta /content/spades_test/K55/assembly_graph_after_simplification.gfa /content/spades_test/assembly_graph_after_simplification.gfa /content/spades_test/K55/final_contigs.fasta /content/spades_test/contigs.fasta /content/spades_test/K55/first_pe_contigs.fasta /content/spades_test/first_pe_contigs.fasta /content/spades_test/K55/strain_graph.gfa /content/spades_test/strain_graph.gfa /content/spades_test/K55/scaffolds.fasta /content/spades_test/scaffolds.fasta /content/spades_test/K55/scaffolds.paths /content/spades_test/scaffolds.paths /content/spades_test/K55/assembly_graph_with_scaffolds.gfa /content/spades_test/assembly_graph_with_scaffolds.gfa /content/spades_test/K55/assembly_graph.fastg /content/spades_test/assembly_graph.fastg /content/spades_test/K55/final_contigs.paths /content/spades_test/contigs.paths\n",
            "\n",
            "\n",
            "===== Copy files finished. \n",
            "\n",
            "\n",
            "===== Assembling finished. \n",
            "\n",
            "\n",
            "===== Check test started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/check_test_script.py --mode common --result_contigs_filename /content/spades_test/contigs.fasta --result_scaffolds_filename /content/spades_test/scaffolds.fasta\n",
            "\n",
            "========= TEST PASSED CORRECTLY.\n",
            "\n",
            "===== Check test finished. \n",
            "\n",
            "\n",
            "===== Breaking scaffolds started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /content/spades_test/scaffolds.fasta --misc_dir /content/spades_test/misc --threshold_for_breaking_scaffolds 3\n",
            "\n",
            "\n",
            "===== Breaking scaffolds finished. \n",
            "\n",
            "\n",
            "===== Terminate started. \n",
            "\n",
            "\n",
            "===== Terminate finished. \n",
            "\n",
            " * Corrected reads are in /content/spades_test/corrected/\n",
            " * Assembled contigs are in /content/spades_test/contigs.fasta\n",
            " * Assembled scaffolds are in /content/spades_test/scaffolds.fasta\n",
            " * Paths in the assembly graph corresponding to the contigs are in /content/spades_test/contigs.paths\n",
            " * Paths in the assembly graph corresponding to the scaffolds are in /content/spades_test/scaffolds.paths\n",
            " * Assembly graph is in /content/spades_test/assembly_graph.fastg\n",
            " * Assembly graph in GFA format is in /content/spades_test/assembly_graph_with_scaffolds.gfa\n",
            "\n",
            "======= SPAdes pipeline finished WITH WARNINGS!\n",
            "\n",
            "=== Error correction and assembling warnings:\n",
            " * 0:00:00.806     1M / 60M   WARN    General                 (launcher.cpp              : 180)   Your data seems to have high uniform coverage depth. It is strongly recommended to use --isolate option.\n",
            "======= Warnings saved to /content/spades_test/warnings.log\n",
            "\n",
            "SPAdes log can be found here: /content/spades_test/spades.log\n",
            "\n",
            "Thank you for using SPAdes! If you use it in your research, please cite:\n",
            "\n",
            "  Prjibelski, A., Antipov, D., Meleshko, D., Lapidus, A. and Korobeynikov, A., 2020. Using SPAdes de novo assembler. Current protocols in bioinformatics, 70(1), p.e102.\n",
            "  doi.org/10.1002/cpbi.102\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! spades/bin/spades.py --test"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "eBnHFqJTpMHo"
      },
      "outputs": [],
      "source": [
        "! chmod +x platanus"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "q5gc1S5cpUio",
        "outputId": "b297091d-3432-4559-ad58-2943dc9f6161"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Platanus version: 1.2.4\n",
            "./platanus assemble \n",
            "\n",
            "Error: not specified input file!!\n",
            "\n",
            "Usage platanus assemble [Options]\n",
            "Options:\n",
            "    -o STR               : prefix of output files (default out, length <= 200)\n",
            "    -f FILE1 [FILE2 ...] : reads file (fasta or fastq, number <= 100)\n",
            "    -k INT               : initial k-mer size (default 32)\n",
            "    -s INT               : step size of k-mer extension (>= 1, default 10)\n",
            "    -n INT               : initial k-mer coverage cutoff (default 0, 0 means auto)\n",
            "    -c INT               : minimun k-mer coverage (default 2)\n",
            "    -a FLOAT             : k-mer extension safety level (default 10.0)\n",
            "    -u FLOAT             : maximum difference for bubble crush (identity, default 0.1)\n",
            "    -d FLOAT             : maximum difference for branch cutting (coverage ratio, default 0.5)\n",
            "    -e FLOAT             : k-mer coverage depth (k = initial k-mer size specified by -k) of homozygous region (default auto)\n",
            "    -t INT               : number of threads (<= 100, default 1)\n",
            "    -m INT               : memory limit for making kmer distribution (GB, >=1, default 16)\n",
            "    -tmp DIR             : directory for temporary files (default .)\n",
            "\n",
            "\n",
            "Outputs:\n",
            "    PREFIX_contig.fa\n",
            "    PREFIX_contigBubble.fa\n",
            "    PREFIX_kmerFrq.tsv\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! ./platanus assemble"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "hD6nS_xbpglF",
        "outputId": "8db988e3-9e81-4d39-e708-3e3575cc2dad"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Platanus version: 1.2.4\n",
            "./platanus assemble -s 0 -k 11 -o genome_assembly/k11/platanus_k11 -f 7_S4_L001_R1_001.fastq 7_S4_L001_R2_001.fastq \n",
            "\n",
            "K = 11, saving kmers from reads...\n",
            "AVE_READ_LEN=68.7405\n",
            "\n",
            "KMER_EXTENSION:\n",
            "K=11, KMER_COVERAGE=833.303 (>= 29), COVERAGE_CUTOFF=29\n",
            "loading kmers...\n",
            "connecting kmers...\n",
            "removing branches...\n",
            "BRANCH_DELETE_THRESHOLD=0.5\n",
            "NUM_CUT=43\n",
            "NUM_CUT=0\n",
            "TOTAL_NUM_CUT=43\n",
            "LENGTH_CUTOFF = 22\n",
            "COVERAGE_CUTOFF = 29\n",
            "removing erroneous nodes...\n",
            "NUM_REMOVED_NODES=0\n",
            "TOTAL_NUM_REMOVED_NODES=0\n",
            "AVE_KMER_COV_REMOVING_BUBBLE=849.148\n",
            "removing bubbles...\n",
            "BUBBLE_IDENTITY_THRESHOLD=0.1\n",
            "NUM_REMOVED_BUBBLES=3\n",
            "NUM_REMOVED_BUBBLES=0\n",
            "TOTAL_NUM_REMOVED_BUBBLES=3\n",
            "mapping reads on de Bruijn Graph nodes...\n",
            "TOTAL_MAPPED_READS=103735\n",
            "TOTAL_UNMAPPED_READS=29601\n",
            "TOTAL_SHORT_READS(<11)=0\n",
            "NUM_DELETE_NODE(reads are unmapped)=120\n",
            "NUM_CUT_NODE=10\n",
            "assemble completed!\n",
            "\n",
            "#### PROCESS INFORMATION ####\n",
            "VmPeak:           8.012 GByte\n",
            "VmHWM:            8.009 GByte\n"
          ]
        }
      ],
      "source": [
        "! ./platanus assemble -s 0 -k 11 -o genome_assembly/k11/platanus_k11 -f 7_S4_L001_R1_001.fastq 7_S4_L001_R2_001.fastq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "N4UwXPBygUFm",
        "outputId": "22628d56-8b66-4ab6-ac51-7e253bd99972"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2025-07-12 10:20:03--  https://github.com/ablab/quast/releases/download/quast_5.3.0/quast-5.3.0.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.112.4\n",
            "Connecting to github.com (github.com)|140.82.112.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://release-assets.githubusercontent.com/github-production-release-asset/4779828/61558e1f-1f39-4291-bfa9-53eba925f193?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-07-12T11%3A09%3A39Z&rscd=attachment%3B+filename%3Dquast-5.3.0.tar.gz&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-07-12T10%3A09%3A06Z&ske=2025-07-12T11%3A09%3A39Z&sks=b&skv=2018-11-09&sig=0L9l1WRybJVW2JbMscLrTommoMetCorFlXL45tQUFA0%3D&jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc1MjMxNTkwMywibmJmIjoxNzUyMzE1NjAzLCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.DJxXbfQyxKXhGEJ1OboCGggwGIAE7z1B5QzeLR0jZQk&response-content-disposition=attachment%3B%20filename%3Dquast-5.3.0.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2025-07-12 10:20:03--  https://release-assets.githubusercontent.com/github-production-release-asset/4779828/61558e1f-1f39-4291-bfa9-53eba925f193?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-07-12T11%3A09%3A39Z&rscd=attachment%3B+filename%3Dquast-5.3.0.tar.gz&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-07-12T10%3A09%3A06Z&ske=2025-07-12T11%3A09%3A39Z&sks=b&skv=2018-11-09&sig=0L9l1WRybJVW2JbMscLrTommoMetCorFlXL45tQUFA0%3D&jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc1MjMxNTkwMywibmJmIjoxNzUyMzE1NjAzLCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.DJxXbfQyxKXhGEJ1OboCGggwGIAE7z1B5QzeLR0jZQk&response-content-disposition=attachment%3B%20filename%3Dquast-5.3.0.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving release-assets.githubusercontent.com (release-assets.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to release-assets.githubusercontent.com (release-assets.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 29021488 (28M) [application/octet-stream]\n",
            "Saving to: ‘quast-5.3.0.tar.gz’\n",
            "\n",
            "quast-5.3.0.tar.gz  100%[===================>]  27.68M  --.-KB/s    in 0.08s   \n",
            "\n",
            "2025-07-12 10:20:03 (327 MB/s) - ‘quast-5.3.0.tar.gz’ saved [29021488/29021488]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! wget https://github.com/ablab/quast/releases/download/quast_5.3.0/quast-5.3.0.tar.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "NQDHPpGYiTT-",
        "outputId": "a17b4820-8d9b-4924-a1c5-cff452862485"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "quast-5.3.0/\n",
            "quast-5.3.0/CHANGES.txt\n",
            "quast-5.3.0/CODE_OF_CONDUCT.md\n",
            "quast-5.3.0/CODING_GUIDELINES.md\n",
            "quast-5.3.0/Dockerfile\n",
            "quast-5.3.0/LICENSE.txt\n",
            "quast-5.3.0/MANIFEST.in\n",
            "quast-5.3.0/PKG-INFO\n",
            "quast-5.3.0/README.md\n",
            "quast-5.3.0/VERSION.txt\n",
            "quast-5.3.0/icarus.py\n",
            "quast-5.3.0/install.sh\n",
            "quast-5.3.0/install_full.sh\n",
            "quast-5.3.0/manual.html\n",
            "quast-5.3.0/metaquast.py\n",
            "quast-5.3.0/quast-lg.py\n",
            "quast-5.3.0/quast.egg-info/\n",
            "quast-5.3.0/quast.egg-info/PKG-INFO\n",
            "quast-5.3.0/quast.egg-info/SOURCES.txt\n",
            "quast-5.3.0/quast.egg-info/dependency_links.txt\n",
            "quast-5.3.0/quast.egg-info/not-zip-safe\n",
            "quast-5.3.0/quast.egg-info/requires.txt\n",
            "quast-5.3.0/quast.egg-info/top_level.txt\n",
            "quast-5.3.0/quast.py\n",
            "quast-5.3.0/quast_libs/\n",
            "quast-5.3.0/quast_libs/N50.py\n",
            "quast-5.3.0/quast_libs/__init__.py\n",
            "quast-5.3.0/quast_libs/aligned_stats.py\n",
            "quast-5.3.0/quast_libs/barrnap/\n",
            "quast-5.3.0/quast_libs/barrnap/LICENSE\n",
            "quast-5.3.0/quast_libs/barrnap/LICENSE.Rfam\n",
            "quast-5.3.0/quast_libs/barrnap/LICENSE.SILVA\n",
            "quast-5.3.0/quast_libs/barrnap/Makefile\n",
            "quast-5.3.0/quast_libs/barrnap/README.md\n",
            "quast-5.3.0/quast_libs/barrnap/bin/\n",
            "quast-5.3.0/quast_libs/barrnap/bin/barrnap\n",
            "quast-5.3.0/quast_libs/barrnap/binaries/\n",
            "quast-5.3.0/quast_libs/barrnap/binaries/darwin/\n",
            "quast-5.3.0/quast_libs/barrnap/binaries/darwin/nhmmer\n",
            "quast-5.3.0/quast_libs/barrnap/binaries/linux/\n",
            "quast-5.3.0/quast_libs/barrnap/binaries/linux/nhmmer\n",
            "quast-5.3.0/quast_libs/barrnap/build/\n",
            "quast-5.3.0/quast_libs/barrnap/build/12S.mito.aln\n",
            "quast-5.3.0/quast_libs/barrnap/build/16S.mito.aln\n",
            "quast-5.3.0/quast_libs/barrnap/build/build_HMMs.sh\n",
            "quast-5.3.0/quast_libs/barrnap/build/fix-SILVA.pl\n",
            "quast-5.3.0/quast_libs/barrnap/db/\n",
            "quast-5.3.0/quast_libs/barrnap/db/arc.hmm\n",
            "quast-5.3.0/quast_libs/barrnap/db/bac.hmm\n",
            "quast-5.3.0/quast_libs/barrnap/db/euk.hmm\n",
            "quast-5.3.0/quast_libs/barrnap/db/mito.hmm\n",
            "quast-5.3.0/quast_libs/basic_stats.py\n",
            "quast-5.3.0/quast_libs/bedtools/\n",
            "quast-5.3.0/quast_libs/bedtools/.gitignore\n",
            "quast-5.3.0/quast_libs/bedtools/.travis.yml\n",
            "quast-5.3.0/quast_libs/bedtools/LICENSE\n",
            "quast-5.3.0/quast_libs/bedtools/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/README.md\n",
            "quast-5.3.0/quast_libs/bedtools/RELEASE_HISTORY\n",
            "quast-5.3.0/quast_libs/bedtools/scripts/\n",
            "quast-5.3.0/quast_libs/bedtools/scripts/makeBashScripts.py\n",
            "quast-5.3.0/quast_libs/bedtools/scripts/vcfsort.sh\n",
            "quast-5.3.0/quast_libs/bedtools/src/\n",
            "quast-5.3.0/quast_libs/bedtools/src/annotateBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/annotateBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/annotateBed/annotateBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/annotateBed/annotateBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/annotateBed/annotateMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/bamToBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/bamToBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/bamToBed/bamToBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/bamToFastq/\n",
            "quast-5.3.0/quast_libs/bedtools/src/bamToFastq/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/bamToFastq/bamToFastq.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/bamToFastq/bamToFastq.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/bamToFastq/bamToFastqMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/bed12ToBed6/\n",
            "quast-5.3.0/quast_libs/bedtools/src/bed12ToBed6/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/bed12ToBed6/bed12ToBed6.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedToBam/\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedToBam/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedToBam/bedToBam.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedToIgv/\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedToIgv/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedToIgv/bedToIgv.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedpeToBam/\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedpeToBam/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedpeToBam/bedpeToBam.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedpeToBam/notes.txt\n",
            "quast-5.3.0/quast_libs/bedtools/src/bedtools.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/closestFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/closestFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/closestFile/closestFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/closestFile/closestFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/closestFile/closestHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/clusterBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/clusterBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/clusterBed/clusterBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/clusterBed/clusterBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/clusterBed/clusterMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/complementFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/complementFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/complementFile/complementFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/complementFile/complementFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/complementFile/complementHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/coverageFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/coverageFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/coverageFile/coverageFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/coverageFile/coverageFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/coverageFile/coverageHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/expand/\n",
            "quast-5.3.0/quast_libs/bedtools/src/expand/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/expand/expand.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/fastaFromBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/fastaFromBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/fastaFromBed/fastaFromBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/fastaFromBed/fastaFromBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/fastaFromBed/fastaFromBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/fisher/\n",
            "quast-5.3.0/quast_libs/bedtools/src/fisher/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/fisher/fisher.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/fisher/fisher.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/fisher/fisherHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/fisher/kfunc.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/fisher/kfunc.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/flankBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/flankBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/flankBed/flankBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/flankBed/flankBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/flankBed/flankBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/genomeCoverageBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/genomeCoverageBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/genomeCoverageBed/genomeCoverageBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/getOverlap/\n",
            "quast-5.3.0/quast_libs/bedtools/src/getOverlap/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/getOverlap/getOverlap.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/groupBy/\n",
            "quast-5.3.0/quast_libs/bedtools/src/groupBy/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/groupBy/groupBy.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/groupBy/groupBy.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/groupBy/groupByHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/intersectFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/intersectFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/intersectFile/intersectFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/intersectFile/intersectFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/intersectFile/intersectHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/jaccard/\n",
            "quast-5.3.0/quast_libs/bedtools/src/jaccard/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/jaccard/jaccard.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/jaccard/jaccard.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/jaccard/jaccardHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/linksBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/linksBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/linksBed/linksBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/linksBed/linksBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/linksBed/linksMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/mapFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/mapFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/mapFile/mapFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/mapFile/mapFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/mapFile/mapHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/maskFastaFromBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/maskFastaFromBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/maskFastaFromBed/maskFastaFromBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/mergeFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/mergeFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/mergeFile/mergeFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/mergeFile/mergeFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/mergeFile/mergeHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiBamCov/\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiBamCov/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiBamCov/multiBamCov.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiBamCov/multiBamCov.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiBamCov/multiBamCovMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiIntersectBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiIntersectBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiIntersectBed/multiIntersectBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiIntersectBed/multiIntersectBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/nekSandbox1/\n",
            "quast-5.3.0/quast_libs/bedtools/src/nekSandbox1/FormatGuess.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/nekSandbox1/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/nekSandbox1/nekSandboxMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/nucBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/nucBed/LargeFileSupport.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/nucBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/nucBed/nucBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/nucBed/nucBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/nucBed/nucBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToBed/pairToBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToBed/pairToBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToBed/pairToBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToPair/\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToPair/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToPair/pairToPair.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToPair/pairToPair.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/pairToPair/pairToPairMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/randomBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/randomBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/randomBed/randomBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/randomBed/randomBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/randomBed/randomBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/regressTest/\n",
            "quast-5.3.0/quast_libs/bedtools/src/regressTest/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/regressTest/RegressTest.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/regressTest/RegressTest.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/regressTest/regressTestMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/reldist/\n",
            "quast-5.3.0/quast_libs/bedtools/src/reldist/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/reldist/reldist.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/reldist/reldist.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/reldist/reldistMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/sampleFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/sampleFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/sampleFile/sampleFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/sampleFile/sampleFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/sampleFile/sampleHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/shiftBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/shiftBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/shiftBed/shiftBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/shiftBed/shiftBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/shiftBed/shiftBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/shuffleBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/shuffleBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/shuffleBed/shuffleBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/shuffleBed/shuffleBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/shuffleBed/shuffleBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/slopBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/slopBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/slopBed/slopBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/slopBed/slopBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/slopBed/slopBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/sortBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/sortBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/sortBed/sortBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/sortBed/sortBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/sortBed/sortMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/spacingFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/spacingFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/spacingFile/spacingFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/spacingFile/spacingFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/spacingFile/spacingHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/split/\n",
            "quast-5.3.0/quast_libs/bedtools/src/split/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/split/splitBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/split/splitBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/split/splitBedMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/subtractFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/subtractFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/subtractFile/subtractFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/subtractFile/subtractFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/subtractFile/subtractHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/tagBam/\n",
            "quast-5.3.0/quast_libs/bedtools/src/tagBam/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/tagBam/tagBam.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/tagBam/tagBam.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/tagBam/tagBamMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/unionBedGraphs/\n",
            "quast-5.3.0/quast_libs/bedtools/src/unionBedGraphs/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/unionBedGraphs/unionBedGraphs.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/unionBedGraphs/unionBedGraphs.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/.gitignore\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/LICENSE\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamAlgorithms.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamAlignment.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamAlignment.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamAux.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamConstants.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamIndex.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamMultiReader.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamMultiReader.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamReader.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamReader.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamWriter.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/BamWriter.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/IBamIODevice.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamConstants.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamHeader.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamHeader.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamProgram.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamProgram.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamProgramChain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamProgramChain.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamReadGroup.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamReadGroup.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamReadGroupDictionary.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamReadGroupDictionary.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamSequence.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamSequence.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamSequenceDictionary.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/SamSequenceDictionary.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/algorithms/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/algorithms/Sort.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/api_global.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamHeader_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamHeader_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamMultiMerger_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamMultiReader_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamMultiReader_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamRandomAccessController_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamRandomAccessController_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamReader_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamReader_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamWriter_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/bam/BamWriter_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/index/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/index/BamIndexFactory_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/index/BamIndexFactory_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/index/BamStandardIndex_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/index/BamStandardIndex_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/index/BamToolsIndex_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/index/BamToolsIndex_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamDeviceFactory_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamDeviceFactory_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamFile_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamFile_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamFtp_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamFtp_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamHttp_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamHttp_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamPipe_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BamPipe_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BgzfStream_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/BgzfStream_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/ByteArray_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/ByteArray_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/HostAddress_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/HostAddress_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/HostInfo_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/HostInfo_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/HttpHeader_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/HttpHeader_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/ILocalIODevice_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/ILocalIODevice_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/NetUnix_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/NetWin_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/RollingBuffer_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/RollingBuffer_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/StdStreamDevice_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/TcpSocketEngine_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/TcpSocketEngine_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/TcpSocketEngine_unix_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/TcpSocketEngine_win_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/TcpSocket_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/io/TcpSocket_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/sam/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/sam/SamFormatParser_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/sam/SamFormatParser_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/sam/SamFormatPrinter_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/sam/SamFormatPrinter_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/sam/SamHeaderValidator_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/sam/SamHeaderValidator_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/sam/SamHeaderVersion_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/utils/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/utils/BamException_p.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/api/internal/utils/BamException_p.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/shared/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools/src/shared/bamtools_global.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools-Ancillary/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools-Ancillary/BamAncillary.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BamTools-Ancillary/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BinTree/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BinTree/BinTree.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BinTree/BinTree.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BinTree/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BlockedIntervals/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BlockedIntervals/BlockedIntervals.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/BlockedIntervals/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextBase.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextBase.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextClosest.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextClosest.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextComplement.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextComplement.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextCoverage.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextCoverage.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextFisher.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextFisher.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextGroupBy.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextGroupBy.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextIntersect.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextIntersect.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextJaccard.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextJaccard.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextMap.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextMap.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextMerge.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextMerge.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextSample.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextSample.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextSpacing.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextSpacing.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextSubtract.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/ContextSubtract.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Contexts/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Fasta/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Fasta/Fasta.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Fasta/Fasta.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Fasta/LargeFileSupport.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Fasta/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Fasta/split.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Fasta/split.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/BamFileReader.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/FileReader.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/InputStreamMgr.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileRecordMergeMgr.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/FileRecordMgr.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/BamRecord.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed12Interval.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed3Interval.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed4Interval.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed5Interval.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Bed6Interval.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/BedGraphInterval.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/BedPlusInterval.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/BlockMgr.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/EmptyRecord.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/GffPlusRecord.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/GffRecord.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/NoPosPlusRecord.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/PlusFields.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Record.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/Record.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/RecordKeyList.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/RecordKeyVector.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/RecordList.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/RecordMgr.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/StrandQueue.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/FileRecordTools/Records/VcfRecord.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/GenomeFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/GenomeFile/GenomeFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/GenomeFile/GenomeFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/GenomeFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/GenomeFile/NewGenomeFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/KeyListOps/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/KeyListOps/KeyListOps.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/KeyListOps/KeyListOps.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/KeyListOps/KeyListOpsMethods.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/KeyListOps/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/NewChromsweep/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/NewChromsweep/CloseSweep.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/NewChromsweep/CloseSweep.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/NewChromsweep/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/NewChromsweep/NewChromsweep.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Point/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/Point/Point.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/RecordOutputMgr/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/RecordOutputMgr/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/RecordOutputMgr/RecordOutputMgr.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/ToolBase/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/ToolBase/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/ToolBase/ToolBase.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/ToolBase/ToolBase.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/VectorOps/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/VectorOps/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/VectorOps/VectorOps.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/VectorOps/VectorOps.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedFile/bedFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedFile/bedFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedFilePE/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedFilePE/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedFilePE/bedFilePE.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedFilePE/bedFilePE.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedGraphFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedGraphFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/bedGraphFile/bedGraphFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/chromsweep/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/chromsweep/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/chromsweep/chromsweep.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/chromsweep/chromsweep.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/driver/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/driver/BedtoolsDriver.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/driver/BedtoolsDriver.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/driver/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/fileType/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/fileType/FileRecordTypeChecker.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/fileType/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/fileType/fileType.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/fileType/fileType.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/BTlist.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/BedtoolsTypes.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/CommonHelp.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/CommonHelp.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/CompressionTools.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/CompressionTools.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/ErrorMsg.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/ErrorMsg.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/FreeList.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/InflateStreamBuf.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/ParseTools.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/ParseTools.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/PushBackStreamBuf.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/PushBackStreamBuf.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/QuickString.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/QuickString.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/Tokenizer.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/general/Tokenizer.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/gzstream/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/gzstream/COPYING.LIB\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/gzstream/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/gzstream/README\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/gzstream/gzstream.C\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/gzstream/gzstream.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/gzstream/version\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/lineFileUtilities/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/lineFileUtilities/lineFileUtilities.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/sequenceUtilities/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/sequenceUtilities/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/sequenceUtilities/sequenceUtils.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/stringUtilities/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/stringUtilities/stringUtilities.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/tabFile/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/tabFile/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/tabFile/tabFile.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/tabFile/tabFile.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/version/\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/version/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/version/version.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/version/version.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/utils/version/version_release.txt\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowBed/\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowBed/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowBed/windowBed.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowBed/windowBed.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowBed/windowMain.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowMaker/\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowMaker/Makefile\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowMaker/windowMaker.cpp\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowMaker/windowMaker.h\n",
            "quast-5.3.0/quast_libs/bedtools/src/windowMaker/windowMakerMain.cpp\n",
            "quast-5.3.0/quast_libs/busco/\n",
            "quast-5.3.0/quast_libs/busco/BuscoAnalysis.py\n",
            "quast-5.3.0/quast_libs/busco/BuscoConfig.py\n",
            "quast-5.3.0/quast_libs/busco/GeneSetAnalysis.py\n",
            "quast-5.3.0/quast_libs/busco/GenomeAnalysis.py\n",
            "quast-5.3.0/quast_libs/busco/TranscriptomeAnalysis.py\n",
            "quast-5.3.0/quast_libs/busco/__init__.py\n",
            "quast-5.3.0/quast_libs/busco/_version.py\n",
            "quast-5.3.0/quast_libs/busco/busco.py\n",
            "quast-5.3.0/quast_libs/busco/config.ini.default\n",
            "quast-5.3.0/quast_libs/busco/hmmsearch\n",
            "quast-5.3.0/quast_libs/busco/pipebricks/\n",
            "quast-5.3.0/quast_libs/busco/pipebricks/PipeConfig.py\n",
            "quast-5.3.0/quast_libs/busco/pipebricks/PipeHelper.py\n",
            "quast-5.3.0/quast_libs/busco/pipebricks/PipeLogger.py\n",
            "quast-5.3.0/quast_libs/busco/pipebricks/Toolset.py\n",
            "quast-5.3.0/quast_libs/busco/pipebricks/__init__.py\n",
            "quast-5.3.0/quast_libs/bwa/\n",
            "quast-5.3.0/quast_libs/bwa/.gitignore\n",
            "quast-5.3.0/quast_libs/bwa/.travis.yml\n",
            "quast-5.3.0/quast_libs/bwa/COPYING\n",
            "quast-5.3.0/quast_libs/bwa/ChangeLog\n",
            "quast-5.3.0/quast_libs/bwa/Makefile\n",
            "quast-5.3.0/quast_libs/bwa/NEWS.md\n",
            "quast-5.3.0/quast_libs/bwa/QSufSort.c\n",
            "quast-5.3.0/quast_libs/bwa/QSufSort.h\n",
            "quast-5.3.0/quast_libs/bwa/README-alt.md\n",
            "quast-5.3.0/quast_libs/bwa/README.md\n",
            "quast-5.3.0/quast_libs/bwa/bamlite.c\n",
            "quast-5.3.0/quast_libs/bwa/bamlite.h\n",
            "quast-5.3.0/quast_libs/bwa/bntseq.c\n",
            "quast-5.3.0/quast_libs/bwa/bntseq.h\n",
            "quast-5.3.0/quast_libs/bwa/bwa.1\n",
            "quast-5.3.0/quast_libs/bwa/bwa.c\n",
            "quast-5.3.0/quast_libs/bwa/bwa.h\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/README.md\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/bwa-postalt.js\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/run-HLA\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/run-bwamem\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/run-gen-ref\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/typeHLA-selctg.js\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/typeHLA.js\n",
            "quast-5.3.0/quast_libs/bwa/bwakit/typeHLA.sh\n",
            "quast-5.3.0/quast_libs/bwa/bwamem.c\n",
            "quast-5.3.0/quast_libs/bwa/bwamem.h\n",
            "quast-5.3.0/quast_libs/bwa/bwamem_extra.c\n",
            "quast-5.3.0/quast_libs/bwa/bwamem_pair.c\n",
            "quast-5.3.0/quast_libs/bwa/bwape.c\n",
            "quast-5.3.0/quast_libs/bwa/bwase.c\n",
            "quast-5.3.0/quast_libs/bwa/bwase.h\n",
            "quast-5.3.0/quast_libs/bwa/bwaseqio.c\n",
            "quast-5.3.0/quast_libs/bwa/bwashm.c\n",
            "quast-5.3.0/quast_libs/bwa/bwt.c\n",
            "quast-5.3.0/quast_libs/bwa/bwt.h\n",
            "quast-5.3.0/quast_libs/bwa/bwt_gen.c\n",
            "quast-5.3.0/quast_libs/bwa/bwt_lite.c\n",
            "quast-5.3.0/quast_libs/bwa/bwt_lite.h\n",
            "quast-5.3.0/quast_libs/bwa/bwtaln.c\n",
            "quast-5.3.0/quast_libs/bwa/bwtaln.h\n",
            "quast-5.3.0/quast_libs/bwa/bwtgap.c\n",
            "quast-5.3.0/quast_libs/bwa/bwtgap.h\n",
            "quast-5.3.0/quast_libs/bwa/bwtindex.c\n",
            "quast-5.3.0/quast_libs/bwa/bwtsw2.h\n",
            "quast-5.3.0/quast_libs/bwa/bwtsw2_aux.c\n",
            "quast-5.3.0/quast_libs/bwa/bwtsw2_chain.c\n",
            "quast-5.3.0/quast_libs/bwa/bwtsw2_core.c\n",
            "quast-5.3.0/quast_libs/bwa/bwtsw2_main.c\n",
            "quast-5.3.0/quast_libs/bwa/bwtsw2_pair.c\n",
            "quast-5.3.0/quast_libs/bwa/example.c\n",
            "quast-5.3.0/quast_libs/bwa/fastmap.c\n",
            "quast-5.3.0/quast_libs/bwa/is.c\n",
            "quast-5.3.0/quast_libs/bwa/kbtree.h\n",
            "quast-5.3.0/quast_libs/bwa/khash.h\n",
            "quast-5.3.0/quast_libs/bwa/kopen.c\n",
            "quast-5.3.0/quast_libs/bwa/kseq.h\n",
            "quast-5.3.0/quast_libs/bwa/ksort.h\n",
            "quast-5.3.0/quast_libs/bwa/kstring.c\n",
            "quast-5.3.0/quast_libs/bwa/kstring.h\n",
            "quast-5.3.0/quast_libs/bwa/ksw.c\n",
            "quast-5.3.0/quast_libs/bwa/ksw.h\n",
            "quast-5.3.0/quast_libs/bwa/kthread.c\n",
            "quast-5.3.0/quast_libs/bwa/kvec.h\n",
            "quast-5.3.0/quast_libs/bwa/main.c\n",
            "quast-5.3.0/quast_libs/bwa/malloc_wrap.c\n",
            "quast-5.3.0/quast_libs/bwa/malloc_wrap.h\n",
            "quast-5.3.0/quast_libs/bwa/maxk.c\n",
            "quast-5.3.0/quast_libs/bwa/pemerge.c\n",
            "quast-5.3.0/quast_libs/bwa/qualfa2fq.pl\n",
            "quast-5.3.0/quast_libs/bwa/rle.c\n",
            "quast-5.3.0/quast_libs/bwa/rle.h\n",
            "quast-5.3.0/quast_libs/bwa/rope.c\n",
            "quast-5.3.0/quast_libs/bwa/rope.h\n",
            "quast-5.3.0/quast_libs/bwa/utils.c\n",
            "quast-5.3.0/quast_libs/bwa/utils.h\n",
            "quast-5.3.0/quast_libs/bwa/xa2multi.pl\n",
            "quast-5.3.0/quast_libs/ca_utils/\n",
            "quast-5.3.0/quast_libs/ca_utils/__init__.py\n",
            "quast-5.3.0/quast_libs/ca_utils/align_contigs.py\n",
            "quast-5.3.0/quast_libs/ca_utils/analyze_contigs.py\n",
            "quast-5.3.0/quast_libs/ca_utils/analyze_misassemblies.py\n",
            "quast-5.3.0/quast_libs/ca_utils/best_set_selection.py\n",
            "quast-5.3.0/quast_libs/ca_utils/misc.py\n",
            "quast-5.3.0/quast_libs/ca_utils/save_results.py\n",
            "quast-5.3.0/quast_libs/circos.py\n",
            "quast-5.3.0/quast_libs/contigs_analyzer.py\n",
            "quast-5.3.0/quast_libs/create_meta_summary.py\n",
            "quast-5.3.0/quast_libs/fastaparser.py\n",
            "quast-5.3.0/quast_libs/genemark/\n",
            "quast-5.3.0/quast_libs/genemark/MetaGeneMark_v1.mod\n",
            "quast-5.3.0/quast_libs/genemark/gm_keys/\n",
            "quast-5.3.0/quast_libs/genemark/gm_keys/gm_key\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/Gibbs3\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/INSTALL\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/LICENSE\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/README.GeneMarkSuite\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/README.MetaGeneMark\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/README.heuristic\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gc\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gibbs\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gibbs-3.1.tar.gz\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gibbs9_95.tar.gz\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gm\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gm.pl\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gm_1.tbl\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gm_25.tbl\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gm_4.tbl\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gmhmmp\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gmhmmp.pl\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gmhmmp_heuristic.pl\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/gmsn.pl\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heu_1.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heu_11.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heu_4.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_30.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_31.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_32.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_33.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_34.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_35.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_36.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_37.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_38.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_39.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_40.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_41.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_42.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_43.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_44.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_45.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_46.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_47.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_48.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_49.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_50.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_51.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_52.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_53.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_54.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_55.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_56.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_57.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_58.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_59.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_60.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_61.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_62.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_63.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_64.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_65.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_66.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_67.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_68.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_69.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_11_70.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_30.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_31.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_32.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_33.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_34.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_35.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_36.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_37.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_38.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_39.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_40.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_41.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_42.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_43.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_44.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_45.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_46.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_47.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_48.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_49.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_50.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_51.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_52.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_53.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_54.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_55.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_56.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_57.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_58.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_59.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_60.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_61.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_62.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_63.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_64.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_65.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_66.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_67.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_68.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_69.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_1_70.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_30.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_31.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_32.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_33.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_34.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_35.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_36.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_37.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_38.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_39.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_40.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_41.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_42.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_43.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_44.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_45.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_46.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_47.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_48.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_49.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_50.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_51.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_52.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_53.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_54.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_55.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_56.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_57.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_58.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_59.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_60.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_61.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_62.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_63.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_64.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_65.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_66.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_67.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_68.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_69.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/heuristic_mod/heu_4_70.mod\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/matinfo\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/mkmat\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/par_1.default\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/par_11.default\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/par_4.default\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/par_EST.default\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/probuild\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/ps_gm_overlay.pl\n",
            "quast-5.3.0/quast_libs/genemark/linux_64/viewmat\n",
            "quast-5.3.0/quast_libs/genemark/macosx/\n",
            "quast-5.3.0/quast_libs/genemark/macosx/Gibbs3\n",
            "quast-5.3.0/quast_libs/genemark/macosx/INSTALL\n",
            "quast-5.3.0/quast_libs/genemark/macosx/LICENSE\n",
            "quast-5.3.0/quast_libs/genemark/macosx/README.GeneMarkSuite\n",
            "quast-5.3.0/quast_libs/genemark/macosx/README.MetaGeneMark\n",
            "quast-5.3.0/quast_libs/genemark/macosx/README.heuristic\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gc\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gibbs\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gibbs-3.1.tar.gz\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gibbs9_95.tar.gz\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gm\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gm.pl\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gm_1.tbl\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gm_25.tbl\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gm_4.tbl\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gmhmmp\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gmhmmp.pl\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gmhmmp_heuristic.pl\n",
            "quast-5.3.0/quast_libs/genemark/macosx/gmsn.pl\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heu_1.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heu_11.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heu_4.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_30.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_31.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_32.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_33.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_34.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_35.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_36.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_37.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_38.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_39.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_40.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_41.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_42.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_43.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_44.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_45.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_46.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_47.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_48.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_49.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_50.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_51.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_52.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_53.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_54.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_55.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_56.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_57.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_58.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_59.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_60.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_61.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_62.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_63.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_64.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_65.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_66.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_67.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_68.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_69.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_11_70.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_30.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_31.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_32.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_33.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_34.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_35.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_36.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_37.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_38.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_39.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_40.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_41.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_42.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_43.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_44.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_45.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_46.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_47.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_48.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_49.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_50.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_51.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_52.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_53.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_54.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_55.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_56.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_57.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_58.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_59.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_60.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_61.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_62.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_63.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_64.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_65.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_66.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_67.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_68.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_69.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_1_70.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_30.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_31.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_32.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_33.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_34.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_35.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_36.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_37.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_38.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_39.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_40.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_41.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_42.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_43.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_44.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_45.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_46.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_47.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_48.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_49.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_50.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_51.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_52.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_53.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_54.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_55.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_56.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_57.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_58.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_59.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_60.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_61.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_62.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_63.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_64.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_65.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_66.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_67.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_68.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_69.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/heuristic_mod/heu_4_70.mod\n",
            "quast-5.3.0/quast_libs/genemark/macosx/matinfo\n",
            "quast-5.3.0/quast_libs/genemark/macosx/mkmat\n",
            "quast-5.3.0/quast_libs/genemark/macosx/par_1.default\n",
            "quast-5.3.0/quast_libs/genemark/macosx/par_11.default\n",
            "quast-5.3.0/quast_libs/genemark/macosx/par_4.default\n",
            "quast-5.3.0/quast_libs/genemark/macosx/par_EST.default\n",
            "quast-5.3.0/quast_libs/genemark/macosx/probuild\n",
            "quast-5.3.0/quast_libs/genemark/macosx/ps_gm_overlay.pl\n",
            "quast-5.3.0/quast_libs/genemark/macosx/viewmat\n",
            "quast-5.3.0/quast_libs/genemark-es/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Class/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Class/Data/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Class/Data/Inheritable.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Devel/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Devel/StackTrace/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Devel/StackTrace/Frame.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Devel/StackTrace.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Exception/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Exception/Class/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Exception/Class/Base.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Exception/Class.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Hash/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Hash/Merge.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/LineFit.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Logger/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Logger/Simple.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Autoload.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Chained.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Cumulative.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Dump.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Dynamic.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Exception.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Foreign.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Metadata.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Overload.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Secure.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Universal.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/Util.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/attributes.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut/lvalue.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Object/InsideOut.pod\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Parallel/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/Parallel/ForkManager.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/ReadBackwards.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Any.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Dumper/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Dumper/Base.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Dumper.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Error.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Loader/\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Loader/Base.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Loader.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Marshall.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Mo.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Node.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Tag.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML/Types.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/lib/YAML.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/Gibbs3\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/INSTALL\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/LICENSE\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/README.GeneMark-ES-suite\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/bet_to_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/bp_seq_select.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/build_mod.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/gc_distr.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/get_sequence_from_GTF.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/gibbs-3.1.tar.gz\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/gmes.cfg\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/gmes_petap.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/gmhmme3\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_32.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_33.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_34.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_35.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_36.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_37.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_38.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_39.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_40.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_41.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_42.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_43.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_44.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_45.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_46.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_47.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_48.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_49.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_50.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_51.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_52.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_53.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_54.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_55.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_56.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_57.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_58.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_59.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_60.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_61.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_62.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_63.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_64.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_65.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_66.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_67.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_68.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_69.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/heu_dir/heu_05_gcode_1_gc_70.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/histogram.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/hmm_to_gtf.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/lib/\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/lib/LineFit.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/lib/ReadBackwards.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/make_nt_freq_mat.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/parse_ET.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/parse_by_introns.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/parse_gibbs.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/parse_set.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/predict_genes.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/prior.bp\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/probuild\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/reformat_fasta.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/reformat_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/rescale_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/rnaseq_introns_to_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/run_hmm_pbs.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/scan_for_bp.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/star_to_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/linux_64/verify_evidence_gmhmm.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/Gibbs3\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/LICENSE\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/README.GeneMark-ET\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/bet_to_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/bp_seq_select.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/build_mod.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/gc_distr.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/get_sequence_from_GTF.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/gibbs-3.1.tar.gz\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/gmes.cfg\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/gmes_petap.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/gmhmme3\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_32.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_33.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_34.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_35.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_36.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_37.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_38.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_39.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_40.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_41.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_42.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_43.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_44.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_45.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_46.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_47.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_48.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_49.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_50.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_51.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_52.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_53.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_54.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_55.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_56.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_57.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_58.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_59.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_60.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_61.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_62.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_63.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_64.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_65.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_66.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_67.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_68.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_69.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/heu_dir/heu_05_gcode_1_gc_70.mod\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/histogram.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/hmm_to_gtf.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/lib/\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/lib/LineFit.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/lib/ReadBackwards.pm\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/make_nt_freq_mat.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/parse_ET.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/parse_by_introns.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/parse_gibbs.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/parse_set.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/predict_genes.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/prior.bp\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/probuild\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/reformat_fasta.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/reformat_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/rescale_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/rnaseq_introns_to_gff.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/run_hmm_pbs.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/scan_for_bp.pl\n",
            "quast-5.3.0/quast_libs/genemark-es/macosx/verify_evidence_gmhmm.pl\n",
            "quast-5.3.0/quast_libs/genemark.py\n",
            "quast-5.3.0/quast_libs/genes_parser.py\n",
            "quast-5.3.0/quast_libs/genome_analyzer.py\n",
            "quast-5.3.0/quast_libs/glimmer/\n",
            "quast-5.3.0/quast_libs/glimmer/COPYRIGHT\n",
            "quast-5.3.0/quast_libs/glimmer/LICENSE\n",
            "quast-5.3.0/quast_libs/glimmer/README.first\n",
            "quast-5.3.0/quast_libs/glimmer/src/\n",
            "quast-5.3.0/quast_libs/glimmer/src/context.h\n",
            "quast-5.3.0/quast_libs/glimmer/src/delcher.h\n",
            "quast-5.3.0/quast_libs/glimmer/src/gene.h\n",
            "quast-5.3.0/quast_libs/glimmer/src/glimmerhmm.c\n",
            "quast-5.3.0/quast_libs/glimmer/src/graph.c\n",
            "quast-5.3.0/quast_libs/glimmer/src/graph.h\n",
            "quast-5.3.0/quast_libs/glimmer/src/icm.h\n",
            "quast-5.3.0/quast_libs/glimmer/src/makefile\n",
            "quast-5.3.0/quast_libs/glimmer/src/oc1.h\n",
            "quast-5.3.0/quast_libs/glimmer/src/sites.c\n",
            "quast-5.3.0/quast_libs/glimmer/src/sites.h\n",
            "quast-5.3.0/quast_libs/glimmer/src/strarray.h\n",
            "quast-5.3.0/quast_libs/glimmer/src/tree_util_prob.c\n",
            "quast-5.3.0/quast_libs/glimmer/src/util.c\n",
            "quast-5.3.0/quast_libs/glimmer/trained/\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/acc.dt\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/acc1.mar\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/acc2\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/acc4\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/acc5\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/acc6\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/all.hexfreq\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/atg.dt\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/atg.markov\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/coding_0_100.model\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/config.file\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/don.dt\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/don1.mar\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/don2\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/don4\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/don5\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/don6\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/exon.distr\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/exon.hexfreq\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/exonno.distr\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/exons.dat\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/false.acc\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/false.atg\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/false.don\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/false.stop\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/genelist\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/noncoding_0_100.model\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/outex\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/outin\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/score_ex.acc\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/score_ex.don\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/score_in.acc\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/score_in.don\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/seqs\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/stop.dt\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/stop.markov\n",
            "quast-5.3.0/quast_libs/glimmer/trained/arabidopsis/train_0_100.cfg\n",
            "quast-5.3.0/quast_libs/glimmer/version_update\n",
            "quast-5.3.0/quast_libs/glimmer.py\n",
            "quast-5.3.0/quast_libs/html_saver/\n",
            "quast-5.3.0/quast_libs/html_saver/__init__.py\n",
            "quast-5.3.0/quast_libs/html_saver/glossary.json\n",
            "quast-5.3.0/quast_libs/html_saver/html_saver.py\n",
            "quast-5.3.0/quast_libs/html_saver/icarus_menu_templ.html\n",
            "quast-5.3.0/quast_libs/html_saver/json_saver.py\n",
            "quast-5.3.0/quast_libs/html_saver/static/\n",
            "quast-5.3.0/quast_libs/html_saver/static/bootstrap/\n",
            "quast-5.3.0/quast_libs/html_saver/static/bootstrap/bootstrap-tooltip-5px-lower.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/bootstrap/bootstrap-tooltip-vlad.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/bootstrap/bootstrap.css\n",
            "quast-5.3.0/quast_libs/html_saver/static/bootstrap/bootstrap.min.css\n",
            "quast-5.3.0/quast_libs/html_saver/static/bootstrap/bootstrap.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/common.css\n",
            "quast-5.3.0/quast_libs/html_saver/static/d3.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/dragtable.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/API.txt\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/FAQ.txt\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/LICENSE.txt\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/Makefile\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/NEWS.txt\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/PLUGINS.txt\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/README.txt\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/excanvas.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/excanvas.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.colorhelpers.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.colorhelpers.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.crosshair.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.crosshair.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.dashes.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.fillbetween.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.fillbetween.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.image.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.image.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.navigate.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.navigate.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.pie.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.pie.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.resize.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.resize.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.selection.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.selection.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.stack.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.stack.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.symbol.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.symbol.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.threshold.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.threshold.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.flot.tickrotor.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/flot/jquery.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/icarus.css\n",
            "quast-5.3.0/quast_libs/html_saver/static/ie_html5.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/draggable.png\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/draggable_16p.png\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/favicon.ico\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/icon_plot.png\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/quast_logo.png\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/quast_logo_black.png\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/semi-transparent-10px.png\n",
            "quast-5.3.0/quast_libs/html_saver/static/img/semi-transparent-10px.pxm\n",
            "quast-5.3.0/quast_libs/html_saver/static/jquery-1.7.2.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/jquery-1.8.2.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/jquery-1.8.2.min.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/jquery-ui.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/jquery.css\n",
            "quast-5.3.0/quast_libs/html_saver/static/report.css\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/build_icarus.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/build_report.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/build_report_common.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/build_report_meta.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/build_total_report.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/build_total_report_common.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/build_total_report_meta.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/display_icarus.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/draw_cumulative_plot.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/draw_frc_plot.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/draw_gc_plot.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/draw_genes_plot.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/draw_meta_misassembl_plot.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/draw_metasummary_plot.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/draw_nx_plot.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/hsvToRgb.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/icarus_interface.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/icarus_utils.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/lighter_darker.js\n",
            "quast-5.3.0/quast_libs/html_saver/static/scripts/utils.js\n",
            "quast-5.3.0/quast_libs/html_saver/template.html\n",
            "quast-5.3.0/quast_libs/html_saver/viewers_template.html\n",
            "quast-5.3.0/quast_libs/icarus.py\n",
            "quast-5.3.0/quast_libs/icarus_builder.py\n",
            "quast-5.3.0/quast_libs/icarus_parser.py\n",
            "quast-5.3.0/quast_libs/icarus_utils.py\n",
            "quast-5.3.0/quast_libs/kronatools/\n",
            "quast-5.3.0/quast_libs/kronatools/LICENSE.txt\n",
            "quast-5.3.0/quast_libs/kronatools/img/\n",
            "quast-5.3.0/quast_libs/kronatools/img/favicon.ico\n",
            "quast-5.3.0/quast_libs/kronatools/img/favicon.uri\n",
            "quast-5.3.0/quast_libs/kronatools/img/hidden.png\n",
            "quast-5.3.0/quast_libs/kronatools/img/hidden.uri\n",
            "quast-5.3.0/quast_libs/kronatools/img/loading.gif\n",
            "quast-5.3.0/quast_libs/kronatools/img/loading.uri\n",
            "quast-5.3.0/quast_libs/kronatools/img/logo-small.png\n",
            "quast-5.3.0/quast_libs/kronatools/img/logo-small.uri\n",
            "quast-5.3.0/quast_libs/kronatools/lib/\n",
            "quast-5.3.0/quast_libs/kronatools/lib/KronaTools.pm\n",
            "quast-5.3.0/quast_libs/kronatools/scripts/\n",
            "quast-5.3.0/quast_libs/kronatools/scripts/ImportText.pl\n",
            "quast-5.3.0/quast_libs/kronatools/src/\n",
            "quast-5.3.0/quast_libs/kronatools/src/krona-2.0.js\n",
            "quast-5.3.0/quast_libs/log.py\n",
            "quast-5.3.0/quast_libs/metautils.py\n",
            "quast-5.3.0/quast_libs/minimap2/\n",
            "quast-5.3.0/quast_libs/minimap2/.github/\n",
            "quast-5.3.0/quast_libs/minimap2/.github/workflows/\n",
            "quast-5.3.0/quast_libs/minimap2/.github/workflows/ci.yaml\n",
            "quast-5.3.0/quast_libs/minimap2/.gitignore\n",
            "quast-5.3.0/quast_libs/minimap2/.gitmodules\n",
            "quast-5.3.0/quast_libs/minimap2/FAQ.md\n",
            "quast-5.3.0/quast_libs/minimap2/LICENSE.txt\n",
            "quast-5.3.0/quast_libs/minimap2/MANIFEST.in\n",
            "quast-5.3.0/quast_libs/minimap2/Makefile\n",
            "quast-5.3.0/quast_libs/minimap2/Makefile.simde\n",
            "quast-5.3.0/quast_libs/minimap2/NEWS.md\n",
            "quast-5.3.0/quast_libs/minimap2/README.md\n",
            "quast-5.3.0/quast_libs/minimap2/align.c\n",
            "quast-5.3.0/quast_libs/minimap2/bseq.c\n",
            "quast-5.3.0/quast_libs/minimap2/bseq.h\n",
            "quast-5.3.0/quast_libs/minimap2/code_of_conduct.md\n",
            "quast-5.3.0/quast_libs/minimap2/cookbook.md\n",
            "quast-5.3.0/quast_libs/minimap2/esterr.c\n",
            "quast-5.3.0/quast_libs/minimap2/example.c\n",
            "quast-5.3.0/quast_libs/minimap2/format.c\n",
            "quast-5.3.0/quast_libs/minimap2/hit.c\n",
            "quast-5.3.0/quast_libs/minimap2/index.c\n",
            "quast-5.3.0/quast_libs/minimap2/kalloc.c\n",
            "quast-5.3.0/quast_libs/minimap2/kalloc.h\n",
            "quast-5.3.0/quast_libs/minimap2/kdq.h\n",
            "quast-5.3.0/quast_libs/minimap2/ketopt.h\n",
            "quast-5.3.0/quast_libs/minimap2/khash.h\n",
            "quast-5.3.0/quast_libs/minimap2/krmq.h\n",
            "quast-5.3.0/quast_libs/minimap2/kseq.h\n",
            "quast-5.3.0/quast_libs/minimap2/ksort.h\n",
            "quast-5.3.0/quast_libs/minimap2/ksw2.h\n",
            "quast-5.3.0/quast_libs/minimap2/ksw2_dispatch.c\n",
            "quast-5.3.0/quast_libs/minimap2/ksw2_extd2_sse.c\n",
            "quast-5.3.0/quast_libs/minimap2/ksw2_exts2_sse.c\n",
            "quast-5.3.0/quast_libs/minimap2/ksw2_extz2_sse.c\n",
            "quast-5.3.0/quast_libs/minimap2/ksw2_ll_sse.c\n",
            "quast-5.3.0/quast_libs/minimap2/kthread.c\n",
            "quast-5.3.0/quast_libs/minimap2/kthread.h\n",
            "quast-5.3.0/quast_libs/minimap2/kvec.h\n",
            "quast-5.3.0/quast_libs/minimap2/lchain.c\n",
            "quast-5.3.0/quast_libs/minimap2/main.c\n",
            "quast-5.3.0/quast_libs/minimap2/map.c\n",
            "quast-5.3.0/quast_libs/minimap2/minimap.h\n",
            "quast-5.3.0/quast_libs/minimap2/minimap2.1\n",
            "quast-5.3.0/quast_libs/minimap2/misc/\n",
            "quast-5.3.0/quast_libs/minimap2/misc/README.md\n",
            "quast-5.3.0/quast_libs/minimap2/misc/mmphase.js\n",
            "quast-5.3.0/quast_libs/minimap2/misc/paftools.js\n",
            "quast-5.3.0/quast_libs/minimap2/misc.c\n",
            "quast-5.3.0/quast_libs/minimap2/mmpriv.h\n",
            "quast-5.3.0/quast_libs/minimap2/options.c\n",
            "quast-5.3.0/quast_libs/minimap2/pe.c\n",
            "quast-5.3.0/quast_libs/minimap2/pyproject.toml\n",
            "quast-5.3.0/quast_libs/minimap2/python/\n",
            "quast-5.3.0/quast_libs/minimap2/python/README.rst\n",
            "quast-5.3.0/quast_libs/minimap2/python/cmappy.h\n",
            "quast-5.3.0/quast_libs/minimap2/python/cmappy.pxd\n",
            "quast-5.3.0/quast_libs/minimap2/python/mappy.pyx\n",
            "quast-5.3.0/quast_libs/minimap2/python/minimap2.py\n",
            "quast-5.3.0/quast_libs/minimap2/sdust.c\n",
            "quast-5.3.0/quast_libs/minimap2/sdust.h\n",
            "quast-5.3.0/quast_libs/minimap2/seed.c\n",
            "quast-5.3.0/quast_libs/minimap2/setup.py\n",
            "quast-5.3.0/quast_libs/minimap2/sketch.c\n",
            "quast-5.3.0/quast_libs/minimap2/splitidx.c\n",
            "quast-5.3.0/quast_libs/minimap2/sse2neon/\n",
            "quast-5.3.0/quast_libs/minimap2/sse2neon/emmintrin.h\n",
            "quast-5.3.0/quast_libs/minimap2/tex/\n",
            "quast-5.3.0/quast_libs/minimap2/tex/Makefile\n",
            "quast-5.3.0/quast_libs/minimap2/tex/bioinfo.cls\n",
            "quast-5.3.0/quast_libs/minimap2/tex/blasr-mc.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/bowtie2-s3.sam.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/bwa-s3.sam.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/bwa.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/eval2roc.pl\n",
            "quast-5.3.0/quast_libs/minimap2/tex/graphmap.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/hs38-simu.sh\n",
            "quast-5.3.0/quast_libs/minimap2/tex/minialign.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/minimap2.bib\n",
            "quast-5.3.0/quast_libs/minimap2/tex/minimap2.tex\n",
            "quast-5.3.0/quast_libs/minimap2/tex/mm2-s3.sam.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/mm2-update.tex\n",
            "quast-5.3.0/quast_libs/minimap2/tex/mm2.approx.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/mm2.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/natbib.bst\n",
            "quast-5.3.0/quast_libs/minimap2/tex/natbib.sty\n",
            "quast-5.3.0/quast_libs/minimap2/tex/ngmlr.eval\n",
            "quast-5.3.0/quast_libs/minimap2/tex/roc.gp\n",
            "quast-5.3.0/quast_libs/minimap2/tex/snap-s3.sam.eval\n",
            "quast-5.3.0/quast_libs/optimal_assembly.py\n",
            "quast-5.3.0/quast_libs/options_parser.py\n",
            "quast-5.3.0/quast_libs/plotter.py\n",
            "quast-5.3.0/quast_libs/plotter_data.py\n",
            "quast-5.3.0/quast_libs/qconfig.py\n",
            "quast-5.3.0/quast_libs/qutils.py\n",
            "quast-5.3.0/quast_libs/ra_utils/\n",
            "quast-5.3.0/quast_libs/ra_utils/__init__.py\n",
            "quast-5.3.0/quast_libs/ra_utils/argparse.py\n",
            "quast-5.3.0/quast_libs/ra_utils/misc.py\n",
            "quast-5.3.0/quast_libs/ra_utils/vcfToBedpe.py\n",
            "quast-5.3.0/quast_libs/reads_analyzer.py\n",
            "quast-5.3.0/quast_libs/reporting.py\n",
            "quast-5.3.0/quast_libs/run_barrnap.py\n",
            "quast-5.3.0/quast_libs/run_busco.py\n",
            "quast-5.3.0/quast_libs/sambamba/\n",
            "quast-5.3.0/quast_libs/sambamba/sambamba_linux\n",
            "quast-5.3.0/quast_libs/sambamba/sambamba_osx\n",
            "quast-5.3.0/quast_libs/search_references_meta.py\n",
            "quast-5.3.0/quast_libs/site_packages/\n",
            "quast-5.3.0/quast_libs/site_packages/__init__.py\n",
            "quast-5.3.0/quast_libs/site_packages/_bz2.py\n",
            "quast-5.3.0/quast_libs/site_packages/_compression.py\n",
            "quast-5.3.0/quast_libs/site_packages/bz2.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/__init__.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/disk.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/format_stack.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/func_inspect.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/hashing.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/logger.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/memory.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/my_exceptions.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/numpy_pickle.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/parallel.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib2/testing.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/__init__.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/_compat.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/_memory_helpers.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/_multiprocessing_helpers.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/disk.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/format_stack.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/func_inspect.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/hashing.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/logger.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/memory.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/my_exceptions.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/numpy_pickle.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/parallel.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/pool.py\n",
            "quast-5.3.0/quast_libs/site_packages/joblib3/testing.py\n",
            "quast-5.3.0/quast_libs/site_packages/jsontemplate/\n",
            "quast-5.3.0/quast_libs/site_packages/jsontemplate/__init__.py\n",
            "quast-5.3.0/quast_libs/site_packages/jsontemplate/jsontemplate.py\n",
            "quast-5.3.0/quast_libs/site_packages/ordered_dict.py\n",
            "quast-5.3.0/quast_libs/site_packages/simplejson/\n",
            "quast-5.3.0/quast_libs/site_packages/simplejson/__init__.py\n",
            "quast-5.3.0/quast_libs/site_packages/simplejson/_speedups.so\n",
            "quast-5.3.0/quast_libs/site_packages/simplejson/decoder.py\n",
            "quast-5.3.0/quast_libs/site_packages/simplejson/encoder.py\n",
            "quast-5.3.0/quast_libs/site_packages/simplejson/ordered_dict.py\n",
            "quast-5.3.0/quast_libs/site_packages/simplejson/scanner.py\n",
            "quast-5.3.0/quast_libs/site_packages/simplejson/tool.py\n",
            "quast-5.3.0/quast_libs/unique_kmers.py\n",
            "quast-5.3.0/quast_libs/version.py\n",
            "quast-5.3.0/setup.cfg\n",
            "quast-5.3.0/setup.py\n",
            "quast-5.3.0/test_data/\n",
            "quast-5.3.0/test_data/contigs_1.fasta\n",
            "quast-5.3.0/test_data/contigs_2.fasta\n",
            "quast-5.3.0/test_data/genes.bed\n",
            "quast-5.3.0/test_data/genes.gff\n",
            "quast-5.3.0/test_data/genes.ncbi\n",
            "quast-5.3.0/test_data/genes.txt\n",
            "quast-5.3.0/test_data/meta_contigs_1.fasta\n",
            "quast-5.3.0/test_data/meta_contigs_2.fasta\n",
            "quast-5.3.0/test_data/meta_ref_1.fasta\n",
            "quast-5.3.0/test_data/meta_ref_2.fasta\n",
            "quast-5.3.0/test_data/meta_ref_3.fasta\n",
            "quast-5.3.0/test_data/operons.bed\n",
            "quast-5.3.0/test_data/operons.gff\n",
            "quast-5.3.0/test_data/operons.txt\n",
            "quast-5.3.0/test_data/reads1.fastq.gz\n",
            "quast-5.3.0/test_data/reads2.fastq.gz\n",
            "quast-5.3.0/test_data/reference.fasta.gz\n"
          ]
        }
      ],
      "source": [
        "! tar -xvzf quast-5.3.0.tar.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zSutJWE3k8mw",
        "outputId": "e9a8b540-b5b0-4837-d352-6e226592b902"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "mv: cannot stat 'quast-5.3.0': No such file or directory\n"
          ]
        }
      ],
      "source": [
        "mv quast-5.3.0 quast"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "24-xbDR6kFOo",
        "outputId": "b7d5f82a-ae08-4c4c-d54f-23cfb4b44635"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: line 1: quast/quast.py: No such file or directory\n"
          ]
        }
      ],
      "source": [
        "! quast/quast.py"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Ge4rCnLKoF2W"
      },
      "source": [
        "## 1.2 run spades"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "pa9ju3TXs6Cq",
        "outputId": "5c048fdc-d5ca-465a-f22c-6fcbbe2df581"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "SPAdes genome assembler v4.1.0\n",
            "\n",
            "Usage: spades.py [options] -o <output_dir>\n",
            "\n",
            "Basic options:\n",
            "  -o <output_dir>             directory to store all the resulting files (required)\n",
            "  --isolate                   this flag is highly recommended for high-coverage isolate and multi-cell data\n",
            "  --sc                        this flag is required for MDA (single-cell) data\n",
            "  --meta                      this flag is required for metagenomic data\n",
            "  --bio                       this flag is required for biosyntheticSPAdes mode\n",
            "  --sewage                    this flag is required for sewage mode\n",
            "  --corona                    this flag is required for coronaSPAdes mode\n",
            "  --rna                       this flag is required for RNA-Seq data\n",
            "  --plasmid                   runs plasmidSPAdes pipeline for plasmid detection\n",
            "  --metaviral                 runs metaviralSPAdes pipeline for virus detection\n",
            "  --metaplasmid               runs metaplasmidSPAdes pipeline for plasmid detection in metagenomic datasets (equivalent for --meta --plasmid)\n",
            "  --rnaviral                  this flag enables virus assembly module from RNA-Seq data\n",
            "  --iontorrent                this flag is required for IonTorrent data\n",
            "  --test                      runs SPAdes on toy dataset\n",
            "  -h, --help                  prints this usage message\n",
            "  -v, --version               prints version\n",
            "\n",
            "Input data:\n",
            "  --12 <filename>             file with interlaced forward and reverse paired-end reads\n",
            "  -1 <filename>               file with forward paired-end reads\n",
            "  -2 <filename>               file with reverse paired-end reads\n",
            "  -s <filename>               file with unpaired reads\n",
            "  --merged <filename>         file with merged forward and reverse paired-end reads\n",
            "  --pe-12 <#> <filename>      file with interlaced reads for paired-end library number <#>.\n",
            "                              Older deprecated syntax is -pe<#>-12 <filename>\n",
            "  --pe-1 <#> <filename>       file with forward reads for paired-end library number <#>.\n",
            "                              Older deprecated syntax is -pe<#>-1 <filename>\n",
            "  --pe-2 <#> <filename>       file with reverse reads for paired-end library number <#>.\n",
            "                              Older deprecated syntax is -pe<#>-2 <filename>\n",
            "  --pe-s <#> <filename>       file with unpaired reads for paired-end library number <#>.\n",
            "                              Older deprecated syntax is -pe<#>-s <filename>\n",
            "  --pe-m <#> <filename>       file with merged reads for paired-end library number <#>.\n",
            "                              Older deprecated syntax is -pe<#>-m <filename>\n",
            "  --pe-or <#> <or>            orientation of reads for paired-end library number <#> \n",
            "                              (<or> = fr, rf, ff).\n",
            "                              Older deprecated syntax is -pe<#>-<or>\n",
            "  --s <#> <filename>          file with unpaired reads for single reads library number <#>.\n",
            "                              Older deprecated syntax is --s<#> <filename>\n",
            "  --mp-12 <#> <filename>      file with interlaced reads for mate-pair library number <#>.\n",
            "                              Older deprecated syntax is -mp<#>-12 <filename>\n",
            "  --mp-1 <#> <filename>       file with forward reads for mate-pair library number <#>.\n",
            "                              Older deprecated syntax is -mp<#>-1 <filename>\n",
            "  --mp-2 <#> <filename>       file with reverse reads for mate-pair library number <#>.\n",
            "                              Older deprecated syntax is -mp<#>-2 <filename>\n",
            "  --mp-s <#> <filename>       file with unpaired reads for mate-pair library number <#>.\n",
            "                              Older deprecated syntax is -mp<#>-s <filename>\n",
            "  --mp-or <#> <or>            orientation of reads for mate-pair library number <#> \n",
            "                              (<or> = fr, rf, ff).\n",
            "                              Older deprecated syntax is -mp<#>-<or>\n",
            "  --hqmp-12 <#> <filename>    file with interlaced reads for high-quality mate-pair library number <#>.\n",
            "                              Older deprecated syntax is -hqmp<#>-12 <filename>\n",
            "  --hqmp-1 <#> <filename>     file with forward reads for high-quality mate-pair library number <#>.\n",
            "                              Older deprecated syntax is -hqmp<#>-1 <filename>\n",
            "  --hqmp-2 <#> <filename>     file with reverse reads for high-quality mate-pair library number <#>.\n",
            "                              Older deprecated syntax is -hqmp<#>-2 <filename>\n",
            "  --hqmp-s <#> <filename>     file with unpaired reads for high-quality mate-pair library number <#>.\n",
            "                              Older deprecated syntax is -hqmp<#>-s <filename>\n",
            "  --hqmp-or <#> <or>          orientation of reads for high-quality mate-pair library number <#> \n",
            "                              (<or> = fr, rf, ff).\n",
            "                              Older deprecated syntax is -hqmp<#>-<or>\n",
            "  --sanger <filename>         file with Sanger reads\n",
            "  --pacbio <filename>         file with PacBio reads\n",
            "  --nanopore <filename>       file with Nanopore reads\n",
            "  --trusted-contigs <filename>\n",
            "                              file with trusted contigs\n",
            "  --untrusted-contigs <filename>\n",
            "                              file with untrusted contigs\n",
            "\n",
            "Pipeline options:\n",
            "  --only-error-correction     runs only read error correction (without assembling)\n",
            "  --only-assembler            runs only assembling (without read error correction)\n",
            "  --careful                   tries to reduce number of mismatches and short indels\n",
            "  --checkpoints <last or all>\n",
            "                              save intermediate check-points ('last', 'all')\n",
            "  --continue                  continue run from the last available check-point (only -o should be specified)\n",
            "  --restart-from <cp>         restart run with updated options and from the specified check-point\n",
            "                              ('ec', 'as', 'k<int>', 'mc', 'last')\n",
            "  --disable-gzip-output       forces error correction not to compress the corrected reads\n",
            "  --disable-rr                disables repeat resolution stage of assembling\n",
            "\n",
            "Advanced options:\n",
            "  --dataset <filename>        file with dataset description in YAML format\n",
            "  -t <int>, --threads <int>   number of threads. [default: 16]\n",
            "  -m <int>, --memory <int>    RAM limit for SPAdes in Gb (terminates if exceeded). [default: 250]\n",
            "  --tmp-dir <dirname>         directory for temporary files. [default: <output_dir>/tmp]\n",
            "  -k <int> [<int> ...]        list of k-mer sizes (must be odd and less than 128)\n",
            "                              [default: 'auto']\n",
            "  --cov-cutoff <float>        coverage cutoff value (a positive float number, or 'auto', or 'off')\n",
            "                              [default: 'off']\n",
            "  --phred-offset <33 or 64>   PHRED quality offset in the input reads (33 or 64),\n",
            "                              [default: auto-detect]\n",
            "  --custom-hmms <dirname>     directory with custom hmms that replace default ones,\n",
            "                              [default: None]\n",
            "  --gfa11                     use GFA v1.1 format for assembly graph\n"
          ]
        }
      ],
      "source": [
        "! spades/bin/spades.py"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "DjPTj93rm7js",
        "outputId": "2020e1a5-a54d-43ad-bf2a-97092eb27e22"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "\n",
            "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
            "\n",
            "\n",
            "Command line: spades/bin/spades.py\t--careful\t-1\t/content/7_S4_L001_R1_001.fastq\t-2\t/content/7_S4_L001_R2_001.fastq\t-o\t/content/genome_assembly/spades\t\n",
            "\n",
            "System information:\n",
            "  SPAdes version: 4.1.0\n",
            "  Python version: 3.11.13\n",
            "  OS: Linux-6.1.123+-x86_64-with-glibc2.35\n",
            "\n",
            "Output dir: /content/genome_assembly/spades\n",
            "Mode: read error correction and assembling\n",
            "Debug mode is turned OFF\n",
            "\n",
            "Dataset parameters:\n",
            "  Standard mode\n",
            "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
            "  Reads:\n",
            "    Library number: 1, library type: paired-end\n",
            "      orientation: fr\n",
            "      left reads: ['/content/7_S4_L001_R1_001.fastq']\n",
            "      right reads: ['/content/7_S4_L001_R2_001.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "Read error correction parameters:\n",
            "  Iterations: 1\n",
            "  PHRED offset will be auto-detected\n",
            "  Corrected reads will be compressed\n",
            "Assembly parameters:\n",
            "  k: automatic selection based on read length\n",
            "  Repeat resolution is enabled\n",
            "  Mismatch careful mode is turned ON\n",
            "  MismatchCorrector will be used\n",
            "  Coverage cutoff is turned OFF\n",
            "  Assembly graph output will use GFA v1.2 format\n",
            "Other parameters:\n",
            "  Dir for temp files: /content/genome_assembly/spades/tmp\n",
            "  Threads: 16\n",
            "  Memory limit (in Gb): 12\n",
            "\n",
            "\n",
            "======= SPAdes pipeline started. Log can be found here: /content/genome_assembly/spades/spades.log\n",
            "\n",
            "/content/7_S4_L001_R1_001.fastq: max reads length: 151\n",
            "/content/7_S4_L001_R2_001.fastq: max reads length: 151\n",
            "\n",
            "Reads length: 151\n",
            "\n",
            "Default k-mer sizes were set to [21, 33, 55, 77] because estimated read length (151) is equal to or greater than 150\n",
            "\n",
            "===== Before start started. \n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-hammer /content/genome_assembly/spades/corrected/configs/config.info\n",
            "\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  76)   Starting BayesHammer, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  77)   Loading config from \"/content/genome_assembly/spades/corrected/configs/config.info\"\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  79)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  87)   Trying to determine PHRED offset\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  93)   Determined value is 33\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (hammer_tools.cpp          :  40)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 114)   Size of aux. kmer data 24 bytes\n",
            "     === ITERATION 0 begins ===\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 16 files using 2 threads. This might take a while.\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99998 Gb\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 4194304\n",
            "  0:00:00.001  1153M / 1153M INFO   K-mer Splitting          (kmer_data.cpp             :  98)   Processing \"/content/7_S4_L001_R1_001.fastq\"\n",
            "  0:00:00.841  1153M / 1153M INFO   K-mer Splitting          (kmer_data.cpp             : 108)   Processed 66668 reads\n",
            "  0:00:00.842  1153M / 1153M INFO   K-mer Splitting          (kmer_data.cpp             :  98)   Processing \"/content/7_S4_L001_R2_001.fastq\"\n",
            "  0:00:01.644  1153M / 1153M INFO   K-mer Splitting          (kmer_data.cpp             : 108)   Processed 133336 reads\n",
            "  0:00:01.644  1153M / 1153M INFO   K-mer Splitting          (kmer_data.cpp             : 113)   Total 133336 reads processed\n",
            "  0:00:01.651     1M / 63M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:01.675     1M / 63M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 975546 kmers in total.\n",
            "  0:00:01.675     1M / 63M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:01.748     1M / 63M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 975546 kmers, 716800 bytes occupied (5.87814 bits per kmer).\n",
            "  0:00:01.749     1M / 63M   INFO   K-mer Counting           (kmer_data.cpp             : 355)   Arranging kmers in hash map order\n",
            "  0:00:01.807    16M / 63M   INFO    General                 (main.cpp                  : 149)   Clustering Hamming graph.\n",
            "  0:00:05.613    16M / 63M   INFO    General                 (main.cpp                  : 156)   Extracting clusters:\n",
            "  0:00:05.628    16M / 63M   INFO    General                 (concurrent_dsu.cpp        :  19)   Connecting to root\n",
            "  0:00:05.649    16M / 63M   INFO    General                 (concurrent_dsu.cpp        :  35)   Calculating counts\n",
            "  0:00:05.860    33M / 63M   INFO    General                 (concurrent_dsu.cpp        :  64)   Writing down entries\n",
            "  0:00:06.257    16M / 63M   INFO    General                 (main.cpp                  : 168)   Clustering done. Total clusters: 490918\n",
            "  0:00:06.257     9M / 63M   INFO   K-mer Counting           (kmer_data.cpp             : 372)   Collecting K-mer information, this takes a while.\n",
            "  0:00:06.264    32M / 63M   INFO   K-mer Counting           (kmer_data.cpp             : 378)   Processing \"/content/7_S4_L001_R1_001.fastq\"\n",
            "  0:00:10.810    32M / 63M   INFO   K-mer Counting           (kmer_data.cpp             : 378)   Processing \"/content/7_S4_L001_R2_001.fastq\"\n",
            "  0:00:15.010    32M / 63M   INFO   K-mer Counting           (kmer_data.cpp             : 385)   Collection done, postprocessing.\n",
            "  0:00:15.017    32M / 63M   INFO   K-mer Counting           (kmer_data.cpp             : 398)   There are 975546 kmers in total. Among them 742500 (76.1112%) are singletons.\n",
            "  0:00:15.017    32M / 63M   INFO    General                 (main.cpp                  : 174)   Subclustering Hamming graph\n",
            "  0:00:15.748    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 637)   Subclustering done. Total 33 non-read kmers were generated.\n",
            "  0:00:15.749    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 638)   Subclustering statistics:\n",
            "  0:00:15.749    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 639)     Total singleton hamming clusters: 459344. Among them 40634 (8.84609%) are good\n",
            "  0:00:15.749    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 640)     Total singleton subclusters: 510. Among them 510 (100%) are good\n",
            "  0:00:15.749    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 641)     Total non-singleton subcluster centers: 34737. Among them 20880 (60.1088%) are good\n",
            "  0:00:15.749    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 642)     Average size of non-trivial subcluster: 14.8614 kmers\n",
            "  0:00:15.749    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 643)     Average number of sub-clusters per non-singleton cluster: 1.11633\n",
            "  0:00:15.749    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 644)     Total solid k-mers: 62024\n",
            "  0:00:15.749    32M / 63M   INFO   Hamming Subclustering    (kmer_cluster.cpp          : 645)     Substitution probabilities: (     0.928501    0.0297012    0.0239421    0.0178553 )\n",
            "(    0.0293103     0.941213    0.0121734    0.0173033 )\n",
            "(     0.016396    0.0119499     0.941869    0.0297848 )\n",
            "(    0.0175251    0.0231669    0.0292942     0.930014 )\n",
            "  0:00:15.750    32M / 63M   INFO    General                 (main.cpp                  : 179)   Finished clustering.\n",
            "  0:00:15.751    32M / 63M   INFO    General                 (main.cpp                  : 198)   Starting solid k-mers expansion in 2 threads.\n",
            "  0:00:17.408    32M / 63M   INFO    General                 (main.cpp                  : 219)   Solid k-mers iteration 0 produced 9549 new k-mers.\n",
            "  0:00:20.724    32M / 63M   INFO    General                 (main.cpp                  : 219)   Solid k-mers iteration 1 produced 1153 new k-mers.\n",
            "  0:00:22.347    32M / 63M   INFO    General                 (main.cpp                  : 219)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
            "  0:00:22.347    32M / 63M   INFO    General                 (main.cpp                  : 223)   Solid k-mers finalized\n",
            "  0:00:22.347    32M / 63M   INFO    General                 (hammer_tools.cpp          : 226)   Starting read correction in 2 threads.\n",
            "  0:00:22.347    32M / 63M   INFO    General                 (hammer_tools.cpp          : 239)   Correcting pair of reads: \"/content/7_S4_L001_R1_001.fastq\" and \"/content/7_S4_L001_R2_001.fastq\"\n",
            "  0:00:22.516   102M / 110M  INFO    General                 (hammer_tools.cpp          : 173)   Prepared batch 0 of 66668 reads.\n",
            "  0:00:24.850   108M / 120M  INFO    General                 (hammer_tools.cpp          : 180)   Processed batch 0\n",
            "  0:00:24.960   108M / 120M  INFO    General                 (hammer_tools.cpp          : 190)   Written batch 0\n",
            "  0:00:24.991    32M / 120M  INFO    General                 (hammer_tools.cpp          : 280)   Correction done. Changed 33238 bases in 18657 reads.\n",
            "  0:00:24.996    32M / 120M  INFO    General                 (hammer_tools.cpp          : 281)   Failed to correct 0 bases out of 8928286.\n",
            "  0:00:24.996     1M / 120M  INFO    General                 (main.cpp                  : 256)   Saving corrected dataset description to /content/genome_assembly/spades/corrected/corrected.yaml\n",
            "  0:00:24.997     1M / 120M  INFO    General                 (main.cpp                  : 263)   All done. Exiting.\n",
            "\n",
            "===== Read error correction finished. \n",
            "\n",
            "\n",
            "===== corrected reads compression started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/compress_all.py --input_file /content/genome_assembly/spades/corrected/corrected.yaml --ext_python_modules_home /content/spades/share/spades --max_threads 16 --output_dir /content/genome_assembly/spades/corrected --gzip_output\n",
            "\n",
            "== Compressing corrected reads (with pigz)\n",
            "== Files to compress: ['/content/genome_assembly/spades/corrected/7_S4_L001_R1_00100.0_0.cor.fastq', '/content/genome_assembly/spades/corrected/7_S4_L001_R2_00100.0_0.cor.fastq', '/content/genome_assembly/spades/corrected/7_S4_L001_R_unpaired00.0_0.cor.fastq']\n",
            "== Files compression is finished\n",
            "== Dataset yaml file is updated\n",
            "\n",
            "===== corrected reads compression finished. \n",
            "\n",
            "\n",
            "===== Read error correction finished. \n",
            "\n",
            "\n",
            "===== Assembling started. \n",
            "\n",
            "\n",
            "===== K21 started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-core /content/genome_assembly/spades/K21/configs/config.info /content/genome_assembly/spades/K21/configs/careful_mode.info\n",
            "\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/genome_assembly/spades/K21/configs/config.info\"\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/genome_assembly/spades/K21/configs/careful_mode.info\"\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 102)   Starting SPAdes, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 103)   Maximum k-mer length: 128\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 104)   Assembling dataset (\"/content/genome_assembly/spades/dataset.info\") with K=21\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 105)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (pipeline.cpp              : 234)   Two-step repeat resolution disabled\n",
            "  0:00:00.001     1M / 19M   INFO   GraphCore                (graph_core.hpp            : 689)   Graph created, vertex min_id: 3, edge min_id: 3\n",
            "  0:00:00.001     1M / 19M   INFO   GraphCore                (graph_core.hpp            : 690)   Vertex size: 48, edge size: 40\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.001     1M / 19M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Binary Read Conversion (id: read_conversion)\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (read_converter.cpp        :  78)   Converting reads to binary format for library #0 (takes a while)\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (read_converter.cpp        :  99)   Converting paired reads\n",
            "  0:00:00.300    81M / 81M   INFO    General                 (binary_converter.cpp      : 127)   16384 reads processed\n",
            "  0:00:00.308    81M / 81M   INFO    General                 (binary_converter.cpp      : 127)   32768 reads processed\n",
            "  0:00:00.434    81M / 81M   INFO    General                 (binary_converter.cpp      : 143)   61698 reads written\n",
            "  0:00:00.444    43M / 61M   INFO    General                 (read_converter.cpp        : 113)   Converting single reads\n",
            "  0:00:00.464    83M / 83M   INFO    General                 (binary_converter.cpp      : 143)   1540 reads written\n",
            "  0:00:00.466    69M / 69M   INFO    General                 (read_converter.cpp        : 119)   Converting merged reads\n",
            "  0:00:00.470    82M / 82M   INFO    General                 (binary_converter.cpp      : 143)   0 reads written\n",
            "  0:00:00.474     1M / 61M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == de Bruijn graph construction (id: construction)\n",
            "  0:00:00.475     1M / 61M   INFO    General                 (construction.cpp          : 150)   Max read length 151\n",
            "  0:00:00.476     1M / 61M   INFO    General                 (construction.cpp          : 156)   Average read length 70.3532\n",
            "  0:00:00.476     1M / 61M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
            "  0:00:00.476     1M / 61M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.476     1M / 61M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.476     1M / 61M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99997 Gb\n",
            "  0:00:00.476     1M / 61M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 3355443\n",
            "  0:00:00.969  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 128)   Processed 249872 reads\n",
            "  0:00:00.969     1M / 86M   INFO    General                 (kmer_splitters.hpp        : 134)   Used 249872 reads\n",
            "  0:00:00.972     1M / 86M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.981     1M / 86M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 196360 kmers in total.\n",
            "  0:00:00.981     1M / 86M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
            "  0:00:00.982     1M / 86M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 453)   Building kmer index\n",
            "  0:00:00.982     1M / 86M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.982     1M / 86M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.982     1M / 86M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99996 Gb\n",
            "  0:00:00.982     1M / 86M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 3355443\n",
            "  0:00:01.026  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 197)   Processed 196360 kmers\n",
            "  0:00:01.026  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 202)   Used 196360 kmers.\n",
            "  0:00:01.026     1M / 86M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:01.033     1M / 86M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 194040 kmers in total.\n",
            "  0:00:01.033     1M / 86M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:01.050     1M / 86M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 194040 kmers, 155248 bytes occupied (6.40066 bits per kmer).\n",
            "  0:00:01.050     1M / 86M   INFO    General                 (kmer_index_builder.hpp    : 168)   Merging final buckets.\n",
            "  0:00:01.056     1M / 86M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 101)   Building k-mer extensions from k+1-mers\n",
            "  0:00:01.090     1M / 86M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 106)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:01.090     1M / 86M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
            "  0:00:01.091     1M / 86M   INFO    General                 (construction.cpp          : 298)   Early tip clipper length bound set as (RL - K)\n",
            "  0:00:01.091     1M / 86M   INFO   Early tip clipping       (early_simplification.hpp  :  48)   Early tip clipping\n",
            "  0:00:01.148     1M / 86M   INFO   Early tip clipping       (early_simplification.hpp  :  83)   #tipped junctions: 3375\n",
            "  0:00:01.149     1M / 86M   INFO   Early tip clipping       (early_simplification.hpp  :  94)   Clipped tips: 3616\n",
            "  0:00:01.149     1M / 86M   INFO   Early tip clipping       (early_simplification.hpp  :  50)   91592 22-mers were removed by early tip clipper\n",
            "  0:00:01.151     1M / 86M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
            "  0:00:01.151     1M / 86M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 381)   Extracting unbranching paths\n",
            "  0:00:01.205     1M / 86M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 400)   Extracting unbranching paths finished. 7514 sequences extracted\n",
            "  0:00:01.224     1M / 86M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 336)   Collecting perfect loops\n",
            "  0:00:01.236     1M / 86M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 369)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:00:01.237     1M / 86M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 586)   Sorting edges...\n",
            "  0:00:01.239     1M / 86M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 588)   Edges sorted\n",
            "  0:00:01.239     1M / 86M   INFO    General                 (debruijn_graph_constructor: 516)   Total 15028 edges to create\n",
            "  0:00:01.239     2M / 86M   INFO    General                 (debruijn_graph_constructor: 519)   Collecting link records\n",
            "  0:00:01.243     2M / 86M   INFO    General                 (debruijn_graph_constructor: 521)   Ordering link records\n",
            "  0:00:01.244     2M / 86M   INFO    General                 (debruijn_graph_constructor: 524)   Sorting done\n",
            "  0:00:01.245     2M / 86M   INFO    General                 (debruijn_graph_constructor: 537)   Sorting LinkRecords...\n",
            "  0:00:01.245     2M / 86M   INFO    General                 (debruijn_graph_constructor: 540)   LinkRecords sorted\n",
            "  0:00:01.245     2M / 86M   INFO    General                 (debruijn_graph_constructor: 542)   Total 5194 vertices to create\n",
            "  0:00:01.245     3M / 86M   INFO    General                 (debruijn_graph_constructor: 545)   Connecting the graph\n",
            "  0:00:01.246     2M / 86M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
            "  0:00:01.247     2M / 86M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:01.263     3M / 86M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 196360 kmers, 156920 bytes occupied (6.39316 bits per kmer).\n",
            "  0:00:01.264     3M / 86M   INFO    General                 (coverage_hash_map_builder.:  49)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:01.795     3M / 86M   INFO    General                 (construction.cpp          : 427)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:01.816     3M / 86M   INFO    General                 (coverage_filling.hpp      :  83)   Processed 15027 edges\n",
            "  0:00:01.818     2M / 86M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
            "  0:00:01.819     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 182)   Kmer coverage valley at: 0\n",
            "  0:00:01.819     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 202)   K-mer histogram maximum: 3\n",
            "  0:00:01.819     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 238)   Estimated median coverage: 4. Coverage mad: 2.9652\n",
            "  0:00:01.820     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 260)   Fitting coverage model\n",
            "  0:00:01.824     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 2\n",
            "  0:00:01.843     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 4\n",
            "  0:00:01.907     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 8\n",
            "  0:00:01.988     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 16\n",
            "  0:00:02.075     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 310)   Fitted mean coverage: 4. Fitted coverage std. dev: 2.39566e-15\n",
            "  0:00:02.075     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 335)   Probability of erroneous kmer at valley: 1\n",
            "  0:00:02.075     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 359)   Preliminary threshold calculated as: 4\n",
            "  0:00:02.075     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 363)   Threshold adjusted to: 1\n",
            "  0:00:02.075     2M / 86M   INFO    General                 (kmer_coverage_model.cpp   : 376)   Estimated genome size (ignoring repeats): 189429\n",
            "  0:00:02.075     2M / 86M   INFO    General                 (genomic_info_filler.cpp   :  56)   Mean coverage was calculated as 4\n",
            "  0:00:02.075     2M / 86M   INFO    General                 (genomic_info_filler.cpp   :  71)   EC coverage threshold value was calculated as 1\n",
            "  0:00:02.075     2M / 86M   INFO    General                 (genomic_info_filler.cpp   :  72)   Trusted kmer low bound: 0\n",
            "  0:00:02.076     2M / 86M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Raw Simplification (id: raw_simplification)\n",
            "  0:00:02.076     2M / 86M   INFO    General                 (simplification.cpp        : 129)   PROCEDURE == Initial cleaning\n",
            "  0:00:02.076     2M / 86M   INFO    General                 (graph_simplification.hpp  : 674)   Flanking coverage based disconnection disabled\n",
            "  0:00:02.076     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Self conjugate edge remover\n",
            "  0:00:02.077     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:02.077     2M / 86M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification (id: simplification)\n",
            "  0:00:02.077     2M / 86M   INFO    General                 (simplification.cpp        : 397)   Graph simplification started\n",
            "  0:00:02.077     2M / 86M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.077     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:02.077     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.079     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 341 times\n",
            "  0:00:02.079     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.116     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 276 times\n",
            "  0:00:02.116     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.117     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.117     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:02.118     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.118     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 15 times\n",
            "  0:00:02.118     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.118     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.120     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.121     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.121     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:02.122     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.122     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.122     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.122     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.122     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.123     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.123     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.123     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:02.123     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.124     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.124     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.124     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.125     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.125     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.125     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.132     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 1063 times\n",
            "  0:00:02.133     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:02.133     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.134     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 98 times\n",
            "  0:00:02.134     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.148     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 63 times\n",
            "  0:00:02.148     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.149     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.149     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 12\n",
            "  0:00:02.149     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.149     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 3 times\n",
            "  0:00:02.149     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.149     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.149     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.150     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.150     2M / 86M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 13\n",
            "  0:00:02.150     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.150     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.150     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.150     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.150     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.150     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.150     2M / 86M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
            "  0:00:02.150     2M / 86M   INFO    General                 (simplification.cpp        : 189)   PROCEDURE == Post simplification\n",
            "  0:00:02.150     2M / 86M   INFO    General                 (graph_simplification.hpp  : 455)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:02.150     2M / 86M   INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:02.150     2M / 86M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.151     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.151     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.151     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.159     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.159     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.159     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.160     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.167     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.167     2M / 86M   INFO    General                 (simplification.cpp        : 348)   Disrupting self-conjugate edges\n",
            "  0:00:02.168     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Removing isolated edges\n",
            "  0:00:02.169     2M / 86M   INFO   Simplification           (parallel_processing.hpp   : 171)   Removing isolated edges triggered 188 times\n",
            "  0:00:02.169     2M / 86M   INFO    General                 (simplification.cpp        : 508)   After simplification:\n",
            "  0:00:02.169     2M / 86M   INFO    General                 (simplification.cpp        : 509)     Average coverage = 134.827\n",
            "  0:00:02.170     2M / 86M   INFO    General                 (simplification.cpp        : 510)     Total length = 44286\n",
            "  0:00:02.170     2M / 86M   INFO    General                 (simplification.cpp        : 511)     Median edge length: 26\n",
            "  0:00:02.170     2M / 86M   INFO    General                 (simplification.cpp        : 512)     Edges: 6215\n",
            "  0:00:02.170     2M / 86M   INFO    General                 (simplification.cpp        : 513)     Vertices: 4004\n",
            "  0:00:02.170     2M / 86M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:02.170     2M / 86M   INFO    General                 (read_converter.cpp        : 135)   Outputting contigs to \"/content/genome_assembly/spades/K21/simplified_contigs\"\n",
            "  0:00:02.175     6M / 86M   INFO    General                 (binary_converter.cpp      : 143)   3108 reads written\n",
            "  0:00:02.177     2M / 86M   INFO    General                 (pipeline.cpp              : 292)   SPAdes finished\n",
            "  0:00:02.179     1M / 86M   INFO    General                 (main.cpp                  : 131)   Assembling time: 0 hours 0 minutes 2 seconds\n",
            "\n",
            "===== K21 finished. \n",
            "\n",
            "\n",
            "===== K33 started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-core /content/genome_assembly/spades/K33/configs/config.info /content/genome_assembly/spades/K33/configs/careful_mode.info\n",
            "\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/genome_assembly/spades/K33/configs/config.info\"\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/genome_assembly/spades/K33/configs/careful_mode.info\"\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 102)   Starting SPAdes, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 103)   Maximum k-mer length: 128\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 104)   Assembling dataset (\"/content/genome_assembly/spades/dataset.info\") with K=33\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 105)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (pipeline.cpp              : 234)   Two-step repeat resolution disabled\n",
            "  0:00:00.001     1M / 19M   INFO   GraphCore                (graph_core.hpp            : 689)   Graph created, vertex min_id: 3, edge min_id: 3\n",
            "  0:00:00.001     1M / 19M   INFO   GraphCore                (graph_core.hpp            : 690)   Vertex size: 48, edge size: 40\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.001     1M / 19M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Binary Read Conversion (id: read_conversion)\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (read_converter.cpp        :  57)   Binary reads detected\n",
            "  0:00:00.001     1M / 19M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == de Bruijn graph construction (id: construction)\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (construction.cpp          : 115)   Contigs from previous K will be used: /content/genome_assembly/spades/K21/simplified_contigs\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (construction.cpp          : 150)   Max read length 151\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (construction.cpp          : 156)   Average read length 70.3532\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99997 Gb\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1677721\n",
            "  0:00:00.473  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 128)   Processed 256088 reads\n",
            "  0:00:00.473     1M / 93M   INFO    General                 (kmer_splitters.hpp        : 134)   Used 256088 reads\n",
            "  0:00:00.479     1M / 93M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.487     1M / 93M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 211928 kmers in total.\n",
            "  0:00:00.487     1M / 93M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
            "  0:00:00.488     1M / 93M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 453)   Building kmer index\n",
            "  0:00:00.488     1M / 93M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.488     1M / 93M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.488     1M / 93M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99996 Gb\n",
            "  0:00:00.488     1M / 93M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1677721\n",
            "  0:00:00.549  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 197)   Processed 211928 kmers\n",
            "  0:00:00.549  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 202)   Used 211928 kmers.\n",
            "  0:00:00.550     1M / 93M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.558     1M / 93M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 211308 kmers in total.\n",
            "  0:00:00.558     1M / 93M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.578     1M / 93M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 211308 kmers, 168048 bytes occupied (6.3622 bits per kmer).\n",
            "  0:00:00.578     1M / 93M   INFO    General                 (kmer_index_builder.hpp    : 168)   Merging final buckets.\n",
            "  0:00:00.588     1M / 93M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 101)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.627     1M / 93M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 106)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.627     1M / 93M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
            "  0:00:00.628     1M / 93M   INFO    General                 (construction.cpp          : 298)   Early tip clipper length bound set as (RL - K)\n",
            "  0:00:00.628     1M / 93M   INFO   Early tip clipping       (early_simplification.hpp  :  48)   Early tip clipping\n",
            "  0:00:00.686     1M / 93M   INFO   Early tip clipping       (early_simplification.hpp  :  83)   #tipped junctions: 3427\n",
            "  0:00:00.687     1M / 93M   INFO   Early tip clipping       (early_simplification.hpp  :  94)   Clipped tips: 3698\n",
            "  0:00:00.688     1M / 93M   INFO   Early tip clipping       (early_simplification.hpp  :  50)   109448 34-mers were removed by early tip clipper\n",
            "  0:00:00.688     1M / 93M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
            "  0:00:00.688     1M / 93M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 381)   Extracting unbranching paths\n",
            "  0:00:00.736     1M / 93M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 400)   Extracting unbranching paths finished. 4555 sequences extracted\n",
            "  0:00:00.756     1M / 93M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 336)   Collecting perfect loops\n",
            "  0:00:00.770     1M / 93M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 369)   Collecting perfect loops finished. 1 loops collected\n",
            "  0:00:00.771     1M / 93M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 586)   Sorting edges...\n",
            "  0:00:00.772     1M / 93M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 588)   Edges sorted\n",
            "  0:00:00.772     1M / 93M   INFO    General                 (debruijn_graph_constructor: 516)   Total 9112 edges to create\n",
            "  0:00:00.772     2M / 93M   INFO    General                 (debruijn_graph_constructor: 519)   Collecting link records\n",
            "  0:00:00.775     2M / 93M   INFO    General                 (debruijn_graph_constructor: 521)   Ordering link records\n",
            "  0:00:00.776     2M / 93M   INFO    General                 (debruijn_graph_constructor: 524)   Sorting done\n",
            "  0:00:00.776     2M / 93M   INFO    General                 (debruijn_graph_constructor: 537)   Sorting LinkRecords...\n",
            "  0:00:00.777     2M / 93M   INFO    General                 (debruijn_graph_constructor: 540)   LinkRecords sorted\n",
            "  0:00:00.777     2M / 93M   INFO    General                 (debruijn_graph_constructor: 542)   Total 3936 vertices to create\n",
            "  0:00:00.777     2M / 93M   INFO    General                 (debruijn_graph_constructor: 545)   Connecting the graph\n",
            "  0:00:00.778     2M / 93M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
            "  0:00:00.778     2M / 93M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.799     2M / 93M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 211928 kmers, 168496 bytes occupied (6.3605 bits per kmer).\n",
            "  0:00:00.800     3M / 93M   INFO    General                 (coverage_hash_map_builder.:  49)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:01.538     3M / 93M   INFO    General                 (construction.cpp          : 427)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:01.595     3M / 93M   INFO    General                 (coverage_filling.hpp      :  83)   Processed 9111 edges\n",
            "  0:00:01.607     2M / 93M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
            "  0:00:01.617     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 182)   Kmer coverage valley at: 0\n",
            "  0:00:01.626     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 202)   K-mer histogram maximum: 3\n",
            "  0:00:01.638     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 238)   Estimated median coverage: 4. Coverage mad: 2.9652\n",
            "  0:00:01.650     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 260)   Fitting coverage model\n",
            "  0:00:01.666     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 2\n",
            "  0:00:01.707     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 4\n",
            "  0:00:01.829     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 8\n",
            "  0:00:01.985     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 16\n",
            "  0:00:02.067     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 310)   Fitted mean coverage: 4. Fitted coverage std. dev: 2.23239e-12\n",
            "  0:00:02.074     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 335)   Probability of erroneous kmer at valley: 1\n",
            "  0:00:02.088     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 359)   Preliminary threshold calculated as: 4\n",
            "  0:00:02.100     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 363)   Threshold adjusted to: 1\n",
            "  0:00:02.114     2M / 93M   INFO    General                 (kmer_coverage_model.cpp   : 376)   Estimated genome size (ignoring repeats): 204868\n",
            "  0:00:02.127     2M / 93M   INFO    General                 (genomic_info_filler.cpp   :  56)   Mean coverage was calculated as 4\n",
            "  0:00:02.131     2M / 93M   INFO    General                 (genomic_info_filler.cpp   :  71)   EC coverage threshold value was calculated as 1\n",
            "  0:00:02.133     2M / 93M   INFO    General                 (genomic_info_filler.cpp   :  72)   Trusted kmer low bound: 0\n",
            "  0:00:02.134     2M / 93M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Raw Simplification (id: raw_simplification)\n",
            "  0:00:02.136     2M / 93M   INFO    General                 (simplification.cpp        : 129)   PROCEDURE == Initial cleaning\n",
            "  0:00:02.137     2M / 93M   INFO    General                 (graph_simplification.hpp  : 674)   Flanking coverage based disconnection disabled\n",
            "  0:00:02.138     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Self conjugate edge remover\n",
            "  0:00:02.141     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:02.142     2M / 93M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification (id: simplification)\n",
            "  0:00:02.144     2M / 93M   INFO    General                 (simplification.cpp        : 397)   Graph simplification started\n",
            "  0:00:02.145     2M / 93M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.146     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:02.148     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.164     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 459 times\n",
            "  0:00:02.168     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.185     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 107 times\n",
            "  0:00:02.187     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.188     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 4 times\n",
            "  0:00:02.188     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:02.188     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.188     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 7 times\n",
            "  0:00:02.188     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.192     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.192     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.192     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.193     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.193     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.195     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.195     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.195     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 2 times\n",
            "  0:00:02.195     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 2 times\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 10 times\n",
            "  0:00:02.196     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 8 times\n",
            "  0:00:02.196     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.197     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 1 times\n",
            "  0:00:02.201     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.201     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 2 times\n",
            "  0:00:02.201     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:02.201     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.201     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 1 times\n",
            "  0:00:02.201     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.201     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.201     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.205     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 376 times\n",
            "  0:00:02.209     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:02.209     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.210     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 87 times\n",
            "  0:00:02.210     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.221     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 43 times\n",
            "  0:00:02.224     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.224     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.225     2M / 93M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 12\n",
            "  0:00:02.226     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.226     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.226     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.226     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.226     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.226     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.226     1M / 93M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
            "  0:00:02.228     1M / 93M   INFO    General                 (simplification.cpp        : 189)   PROCEDURE == Post simplification\n",
            "  0:00:02.232     1M / 93M   INFO    General                 (graph_simplification.hpp  : 455)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:02.232     1M / 93M   INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:02.233     1M / 93M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.233     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.233     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.233     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.240     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.243     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.248     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.248     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.255     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.255     1M / 93M   INFO    General                 (simplification.cpp        : 348)   Disrupting self-conjugate edges\n",
            "  0:00:02.256     2M / 93M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Removing isolated edges\n",
            "  0:00:02.263     1M / 93M   INFO   Simplification           (parallel_processing.hpp   : 171)   Removing isolated edges triggered 457 times\n",
            "  0:00:02.268     1M / 93M   INFO    General                 (simplification.cpp        : 508)   After simplification:\n",
            "  0:00:02.268     1M / 93M   INFO    General                 (simplification.cpp        : 509)     Average coverage = 128.797\n",
            "  0:00:02.268     1M / 93M   INFO    General                 (simplification.cpp        : 510)     Total length = 34456\n",
            "  0:00:02.268     1M / 93M   INFO    General                 (simplification.cpp        : 511)     Median edge length: 34\n",
            "  0:00:02.268     1M / 93M   INFO    General                 (simplification.cpp        : 512)     Edges: 3359\n",
            "  0:00:02.268     1M / 93M   INFO    General                 (simplification.cpp        : 513)     Vertices: 2296\n",
            "  0:00:02.268     1M / 93M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:02.272     1M / 93M   INFO    General                 (read_converter.cpp        : 135)   Outputting contigs to \"/content/genome_assembly/spades/K33/simplified_contigs\"\n",
            "  0:00:02.281     5M / 93M   INFO    General                 (binary_converter.cpp      : 143)   1680 reads written\n",
            "  0:00:02.291     1M / 93M   INFO    General                 (pipeline.cpp              : 292)   SPAdes finished\n",
            "  0:00:02.299     1M / 93M   INFO    General                 (main.cpp                  : 131)   Assembling time: 0 hours 0 minutes 2 seconds\n",
            "\n",
            "===== K33 finished. \n",
            "\n",
            "\n",
            "===== K55 started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-core /content/genome_assembly/spades/K55/configs/config.info /content/genome_assembly/spades/K55/configs/careful_mode.info\n",
            "\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/genome_assembly/spades/K55/configs/config.info\"\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/genome_assembly/spades/K55/configs/careful_mode.info\"\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (main.cpp                  : 102)   Starting SPAdes, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (main.cpp                  : 103)   Maximum k-mer length: 128\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (main.cpp                  : 104)   Assembling dataset (\"/content/genome_assembly/spades/dataset.info\") with K=55\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (main.cpp                  : 105)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (pipeline.cpp              : 234)   Two-step repeat resolution disabled\n",
            "  0:00:00.002     1M / 19M   INFO   GraphCore                (graph_core.hpp            : 689)   Graph created, vertex min_id: 3, edge min_id: 3\n",
            "  0:00:00.002     1M / 19M   INFO   GraphCore                (graph_core.hpp            : 690)   Vertex size: 48, edge size: 40\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (pipeline.cpp              : 245)   Will need read mapping, kmer mapper will be attached\n",
            "  0:00:00.002     1M / 19M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Binary Read Conversion (id: read_conversion)\n",
            "  0:00:00.003     1M / 19M   INFO    General                 (read_converter.cpp        :  57)   Binary reads detected\n",
            "  0:00:00.003     1M / 19M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == de Bruijn graph construction (id: construction)\n",
            "  0:00:00.003     1M / 19M   INFO    General                 (construction.cpp          : 115)   Contigs from previous K will be used: /content/genome_assembly/spades/K33/simplified_contigs\n",
            "  0:00:00.004     1M / 19M   INFO    General                 (construction.cpp          : 150)   Max read length 151\n",
            "  0:00:00.004     1M / 19M   INFO    General                 (construction.cpp          : 156)   Average read length 70.3532\n",
            "  0:00:00.005     1M / 19M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
            "  0:00:00.015     1M / 19M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.016     1M / 19M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.016     1M / 19M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99997 Gb\n",
            "  0:00:00.016     1M / 19M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1677721\n",
            "  0:00:00.534  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 128)   Processed 253232 reads\n",
            "  0:00:00.534     1M / 53M   INFO    General                 (kmer_splitters.hpp        : 134)   Used 253232 reads\n",
            "  0:00:00.537     1M / 53M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.564     1M / 53M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 201580 kmers in total.\n",
            "  0:00:00.575     1M / 53M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
            "  0:00:00.576     1M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 453)   Building kmer index\n",
            "  0:00:00.576     1M / 53M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.576     1M / 53M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.576     1M / 53M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99996 Gb\n",
            "  0:00:00.576     1M / 53M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1677721\n",
            "  0:00:00.707  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 197)   Processed 201580 kmers\n",
            "  0:00:00.718  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 202)   Used 201580 kmers.\n",
            "  0:00:00.727     1M / 53M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.761     1M / 53M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 202641 kmers in total.\n",
            "  0:00:00.775     1M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.832     1M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 202641 kmers, 161472 bytes occupied (6.3747 bits per kmer).\n",
            "  0:00:00.833     1M / 53M   INFO    General                 (kmer_index_builder.hpp    : 168)   Merging final buckets.\n",
            "  0:00:00.857     1M / 53M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 101)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.930     1M / 53M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 106)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.933     1M / 53M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
            "  0:00:00.943     1M / 53M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 381)   Extracting unbranching paths\n",
            "  0:00:01.081     1M / 53M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 400)   Extracting unbranching paths finished. 7379 sequences extracted\n",
            "  0:00:01.175     1M / 53M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 336)   Collecting perfect loops\n",
            "  0:00:01.210     1M / 53M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 369)   Collecting perfect loops finished. 1 loops collected\n",
            "  0:00:01.213     1M / 53M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 586)   Sorting edges...\n",
            "  0:00:01.221     1M / 53M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 588)   Edges sorted\n",
            "  0:00:01.222     1M / 53M   INFO    General                 (debruijn_graph_constructor: 516)   Total 14760 edges to create\n",
            "  0:00:01.225     2M / 53M   INFO    General                 (debruijn_graph_constructor: 519)   Collecting link records\n",
            "  0:00:01.237     2M / 53M   INFO    General                 (debruijn_graph_constructor: 521)   Ordering link records\n",
            "  0:00:01.238     2M / 53M   INFO    General                 (debruijn_graph_constructor: 524)   Sorting done\n",
            "  0:00:01.239     2M / 53M   INFO    General                 (debruijn_graph_constructor: 537)   Sorting LinkRecords...\n",
            "  0:00:01.239     2M / 53M   INFO    General                 (debruijn_graph_constructor: 540)   LinkRecords sorted\n",
            "  0:00:01.240     2M / 53M   INFO    General                 (debruijn_graph_constructor: 542)   Total 8441 vertices to create\n",
            "  0:00:01.240     3M / 53M   INFO    General                 (debruijn_graph_constructor: 545)   Connecting the graph\n",
            "  0:00:01.244     3M / 53M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
            "  0:00:01.244     3M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:01.280     3M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 201580 kmers, 160600 bytes occupied (6.37365 bits per kmer).\n",
            "  0:00:01.291     4M / 53M   INFO    General                 (coverage_hash_map_builder.:  49)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:01.719     4M / 53M   INFO    General                 (construction.cpp          : 427)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:01.780     4M / 53M   INFO    General                 (coverage_filling.hpp      :  83)   Processed 14759 edges\n",
            "  0:00:01.784     2M / 53M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
            "  0:00:01.785     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 182)   Kmer coverage valley at: 0\n",
            "  0:00:01.790     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 202)   K-mer histogram maximum: 3\n",
            "  0:00:01.792     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 238)   Estimated median coverage: 4. Coverage mad: 2.9652\n",
            "  0:00:01.796     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 260)   Fitting coverage model\n",
            "  0:00:01.808     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 2\n",
            "  0:00:01.848     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 4\n",
            "  0:00:01.978     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 8\n",
            "  0:00:02.136     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 16\n",
            "  0:00:02.247     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 310)   Fitted mean coverage: 4. Fitted coverage std. dev: 1.77852e-13\n",
            "  0:00:02.254     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 335)   Probability of erroneous kmer at valley: 1\n",
            "  0:00:02.267     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 359)   Preliminary threshold calculated as: 4\n",
            "  0:00:02.277     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 363)   Threshold adjusted to: 1\n",
            "  0:00:02.294     2M / 53M   INFO    General                 (kmer_coverage_model.cpp   : 376)   Estimated genome size (ignoring repeats): 195588\n",
            "  0:00:02.303     2M / 53M   INFO    General                 (genomic_info_filler.cpp   :  56)   Mean coverage was calculated as 4\n",
            "  0:00:02.318     2M / 53M   INFO    General                 (genomic_info_filler.cpp   :  71)   EC coverage threshold value was calculated as 1\n",
            "  0:00:02.327     2M / 53M   INFO    General                 (genomic_info_filler.cpp   :  72)   Trusted kmer low bound: 0\n",
            "  0:00:02.332     2M / 53M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Gap Closer (id: early_gapcloser)\n",
            "  0:00:02.343    16M / 53M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:02.346    16M / 53M   INFO    General                 (gap_closer.cpp            : 102)   Total edges in tip neighborhood: 5869 out of 14759, length: 268111\n",
            "  0:00:02.346    16M / 53M   INFO    General                 (edge_index.hpp            : 196)   Using small index (max_id = 14836)\n",
            "  0:00:02.354    16M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:02.403    16M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 178438 kmers, 129624 bytes occupied (5.8115 bits per kmer).\n",
            "  0:00:02.405    17M / 53M   INFO    General                 (edge_index_builders.hpp   : 266)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:02.424    17M / 53M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:02.755    17M / 53M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 61698 reads processed\n",
            "  0:00:02.761    16M / 53M   INFO    General                 (gap_closer.cpp            : 491)   Initializing gap closer\n",
            "  0:00:02.762    16M / 53M   INFO   GapCloser                (gap_closer.cpp            : 406)   Collecting gap candidates\n",
            "  0:00:02.763    16M / 53M   INFO   GapCloser                (gap_closer.cpp            : 410)   Total 129 tips collected, total 158 connection candidates\n",
            "  0:00:02.764    16M / 53M   INFO   GapCloser                (gap_closer.cpp            : 431)   Closing short gaps complete: filled 2 gaps after checking 156 candidates\n",
            "  0:00:02.765    16M / 53M   INFO    General                 (gap_closer.cpp            : 495)   Gap closer done\n",
            "  0:00:02.766     2M / 53M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Raw Simplification (id: raw_simplification)\n",
            "  0:00:02.766     2M / 53M   INFO    General                 (simplification.cpp        : 129)   PROCEDURE == Initial cleaning\n",
            "  0:00:02.766     2M / 53M   INFO    General                 (graph_simplification.hpp  : 674)   Flanking coverage based disconnection disabled\n",
            "  0:00:02.766     2M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Self conjugate edge remover\n",
            "  0:00:02.767     2M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:02.767     2M / 53M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification (id: simplification)\n",
            "  0:00:02.767     2M / 53M   INFO    General                 (simplification.cpp        : 397)   Graph simplification started\n",
            "  0:00:02.767     2M / 53M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:02.767     2M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:02.768     2M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.780     2M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 2814 times\n",
            "  0:00:02.780     2M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.790     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 63 times\n",
            "  0:00:02.790     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.791     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.791     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:02.791     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.791     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 1 times\n",
            "  0:00:02.791     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.791     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.791     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.791     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.791     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.792     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.792     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:02.792     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.793     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.793     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.793     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.794     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.794     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.794     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.795     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:02.795     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.795     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.795     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.795     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.795     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.796     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 73 times\n",
            "  0:00:02.796     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:02.796     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.796     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 45 times\n",
            "  0:00:02.797     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.801     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 29 times\n",
            "  0:00:02.801     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.802     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.802     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 12\n",
            "  0:00:02.802     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.802     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 3 times\n",
            "  0:00:02.802     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.802     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.802     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.802     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.803     3M / 53M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 13\n",
            "  0:00:02.803     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:02.803     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:02.803     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:02.803     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:02.803     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:02.803     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.803     3M / 53M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Gap Closer (id: late_gapcloser)\n",
            "  0:00:02.807    16M / 53M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:02.808    16M / 53M   INFO    General                 (gap_closer.cpp            : 102)   Total edges in tip neighborhood: 2912 out of 3527, length: 193981\n",
            "  0:00:02.808    16M / 53M   INFO    General                 (edge_index.hpp            : 196)   Using small index (max_id = 14836)\n",
            "  0:00:02.812    16M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:02.836    16M / 53M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 101000 kmers, 73704 bytes occupied (5.83794 bits per kmer).\n",
            "  0:00:02.836    17M / 53M   INFO    General                 (edge_index_builders.hpp   : 266)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:02.846    17M / 53M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:03.246    17M / 53M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 61698 reads processed\n",
            "  0:00:03.253    16M / 53M   INFO    General                 (gap_closer.cpp            : 491)   Initializing gap closer\n",
            "  0:00:03.254    16M / 53M   INFO   GapCloser                (gap_closer.cpp            : 406)   Collecting gap candidates\n",
            "  0:00:03.255    16M / 53M   INFO   GapCloser                (gap_closer.cpp            : 410)   Total 38 tips collected, total 60 connection candidates\n",
            "  0:00:03.255    16M / 53M   INFO   GapCloser                (gap_closer.cpp            : 431)   Closing short gaps complete: filled 0 gaps after checking 60 candidates\n",
            "  0:00:03.255    16M / 53M   INFO    General                 (gap_closer.cpp            : 495)   Gap closer done\n",
            "  0:00:03.257     3M / 53M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
            "  0:00:03.257     3M / 53M   INFO    General                 (simplification.cpp        : 189)   PROCEDURE == Post simplification\n",
            "  0:00:03.257     3M / 53M   INFO    General                 (graph_simplification.hpp  : 455)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:03.257     3M / 53M   INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:03.257     3M / 53M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:03.257     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:03.258     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:03.258     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:03.259     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:03.259     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:03.259     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:03.260     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:03.261     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:03.261     3M / 53M   INFO    General                 (simplification.cpp        : 348)   Disrupting self-conjugate edges\n",
            "  0:00:03.261     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Removing isolated edges\n",
            "  0:00:03.263     3M / 53M   INFO   Simplification           (parallel_processing.hpp   : 171)   Removing isolated edges triggered 1325 times\n",
            "  0:00:03.263     3M / 53M   INFO    General                 (simplification.cpp        : 508)   After simplification:\n",
            "  0:00:03.263     3M / 53M   INFO    General                 (simplification.cpp        : 509)     Average coverage = 119.153\n",
            "  0:00:03.263     3M / 53M   INFO    General                 (simplification.cpp        : 510)     Total length = 18597\n",
            "  0:00:03.263     3M / 53M   INFO    General                 (simplification.cpp        : 511)     Median edge length: 72\n",
            "  0:00:03.263     3M / 53M   INFO    General                 (simplification.cpp        : 512)     Edges: 877\n",
            "  0:00:03.263     3M / 53M   INFO    General                 (simplification.cpp        : 513)     Vertices: 676\n",
            "  0:00:03.264     3M / 53M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:03.264     3M / 53M   INFO    General                 (read_converter.cpp        : 135)   Outputting contigs to \"/content/genome_assembly/spades/K55/simplified_contigs\"\n",
            "  0:00:03.267     6M / 53M   INFO    General                 (binary_converter.cpp      : 143)   439 reads written\n",
            "  0:00:03.269     3M / 53M   INFO    General                 (pipeline.cpp              : 292)   SPAdes finished\n",
            "  0:00:03.270     1M / 53M   INFO    General                 (main.cpp                  : 131)   Assembling time: 0 hours 0 minutes 3 seconds\n",
            "\n",
            "===== K55 finished. \n",
            "\n",
            "\n",
            "===== K77 started. \n",
            "\n",
            "\n",
            "== Running: /content/spades/bin/spades-core /content/genome_assembly/spades/K77/configs/config.info /content/genome_assembly/spades/K77/configs/careful_mode.info\n",
            "\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/genome_assembly/spades/K77/configs/config.info\"\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  :  94)   Loaded config from \"/content/genome_assembly/spades/K77/configs/careful_mode.info\"\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (memory_limit.cpp          :  55)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 102)   Starting SPAdes, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 103)   Maximum k-mer length: 128\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 104)   Assembling dataset (\"/content/genome_assembly/spades/dataset.info\") with K=77\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (main.cpp                  : 105)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
            "  0:00:00.000     1M / 19M   INFO    General                 (pipeline.cpp              : 234)   Two-step repeat resolution disabled\n",
            "  0:00:00.001     1M / 19M   INFO   GraphCore                (graph_core.hpp            : 689)   Graph created, vertex min_id: 3, edge min_id: 3\n",
            "  0:00:00.001     1M / 19M   INFO   GraphCore                (graph_core.hpp            : 690)   Vertex size: 48, edge size: 40\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (pipeline.cpp              : 245)   Will need read mapping, kmer mapper will be attached\n",
            "  0:00:00.001     1M / 19M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Binary Read Conversion (id: read_conversion)\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (read_converter.cpp        :  57)   Binary reads detected\n",
            "  0:00:00.001     1M / 19M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == de Bruijn graph construction (id: construction)\n",
            "  0:00:00.001     1M / 19M   INFO    General                 (construction.cpp          : 115)   Contigs from previous K will be used: /content/genome_assembly/spades/K55/simplified_contigs\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (construction.cpp          : 150)   Max read length 151\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (construction.cpp          : 156)   Average read length 70.3532\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99997 Gb\n",
            "  0:00:00.002     1M / 19M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1118481\n",
            "  0:00:00.185  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 128)   Processed 250750 reads\n",
            "  0:00:00.185     1M / 41M   INFO    General                 (kmer_splitters.hpp        : 134)   Used 250750 reads\n",
            "  0:00:00.187     1M / 41M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.197     1M / 41M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 168656 kmers in total.\n",
            "  0:00:00.197     1M / 41M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
            "  0:00:00.198     1M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 453)   Building kmer index\n",
            "  0:00:00.198     1M / 41M   INFO    General                 (kmer_index_builder.hpp    : 258)   Splitting kmer instances into 20 files using 2 threads. This might take a while.\n",
            "  0:00:00.198     1M / 41M   INFO    General                 (file_limit.hpp            :  43)   Open file limit set to 1048576\n",
            "  0:00:00.198     1M / 41M   INFO    General                 (kmer_splitter.hpp         :  94)   Memory available for splitting buffers: 1.99996 Gb\n",
            "  0:00:00.198     1M / 41M   INFO    General                 (kmer_splitter.hpp         : 102)   Using cell size of 1118481\n",
            "  0:00:00.254  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 197)   Processed 168656 kmers\n",
            "  0:00:00.255  1161M / 1161M INFO    General                 (kmer_splitters.hpp        : 202)   Used 168656 kmers.\n",
            "  0:00:00.255     1M / 41M   INFO    General                 (kmer_index_builder.hpp    : 264)   Starting k-mer counting.\n",
            "  0:00:00.270     1M / 41M   INFO    General                 (kmer_index_builder.hpp    : 275)   K-mer counting done. There are 170400 kmers in total.\n",
            "  0:00:00.270     1M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.286     1M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 170400 kmers, 138264 bytes occupied (6.49127 bits per kmer).\n",
            "  0:00:00.287     1M / 41M   INFO    General                 (kmer_index_builder.hpp    : 168)   Merging final buckets.\n",
            "  0:00:00.296     1M / 41M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 101)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.330     1M / 41M   INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 106)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.331     1M / 41M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
            "  0:00:00.332     1M / 41M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 381)   Extracting unbranching paths\n",
            "  0:00:00.385     1M / 41M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 400)   Extracting unbranching paths finished. 5151 sequences extracted\n",
            "  0:00:00.422     1M / 41M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 336)   Collecting perfect loops\n",
            "  0:00:00.434     1M / 41M   INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 369)   Collecting perfect loops finished. 1 loops collected\n",
            "  0:00:00.435     1M / 41M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 586)   Sorting edges...\n",
            "  0:00:00.436     1M / 41M   INFO   DeBruijnGraphConstructor (debruijn_graph_constructor: 588)   Edges sorted\n",
            "  0:00:00.436     1M / 41M   INFO    General                 (debruijn_graph_constructor: 516)   Total 10304 edges to create\n",
            "  0:00:00.436     2M / 41M   INFO    General                 (debruijn_graph_constructor: 519)   Collecting link records\n",
            "  0:00:00.440     2M / 41M   INFO    General                 (debruijn_graph_constructor: 521)   Ordering link records\n",
            "  0:00:00.441     2M / 41M   INFO    General                 (debruijn_graph_constructor: 524)   Sorting done\n",
            "  0:00:00.442     2M / 41M   INFO    General                 (debruijn_graph_constructor: 537)   Sorting LinkRecords...\n",
            "  0:00:00.442     2M / 41M   INFO    General                 (debruijn_graph_constructor: 540)   LinkRecords sorted\n",
            "  0:00:00.442     2M / 41M   INFO    General                 (debruijn_graph_constructor: 542)   Total 6896 vertices to create\n",
            "  0:00:00.442     3M / 41M   INFO    General                 (debruijn_graph_constructor: 545)   Connecting the graph\n",
            "  0:00:00.444     2M / 41M   INFO    General                 (stage.cpp                 : 121)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
            "  0:00:00.444     2M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:00.460     3M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 168656 kmers, 137032 bytes occupied (6.49995 bits per kmer).\n",
            "  0:00:00.461     3M / 41M   INFO    General                 (coverage_hash_map_builder.:  49)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:00.635     3M / 41M   INFO    General                 (construction.cpp          : 427)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:00.671     3M / 41M   INFO    General                 (coverage_filling.hpp      :  83)   Processed 10303 edges\n",
            "  0:00:00.675     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
            "  0:00:00.676     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 182)   Kmer coverage valley at: 4\n",
            "  0:00:00.676     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 202)   K-mer histogram maximum: 7\n",
            "  0:00:00.676     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 238)   Estimated median coverage: 8. Coverage mad: 4.4478\n",
            "  0:00:00.676     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 260)   Fitting coverage model\n",
            "  0:00:00.688     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 2\n",
            "  0:00:00.737     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 4\n",
            "  0:00:00.932     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 8\n",
            "  0:00:01.296     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 296)   ... iteration 16\n",
            "  0:00:01.417     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 310)   Fitted mean coverage: 4. Fitted coverage std. dev: 1.08934e-13\n",
            "  0:00:01.418     2M / 41M   WARN    General                 (kmer_coverage_model.cpp   : 328)   Valley value was estimated improperly, reset to 2\n",
            "  0:00:01.418     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 335)   Probability of erroneous kmer at valley: 1\n",
            "  0:00:01.418     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 359)   Preliminary threshold calculated as: 4\n",
            "  0:00:01.418     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 363)   Threshold adjusted to: 2\n",
            "  0:00:01.418     2M / 41M   INFO    General                 (kmer_coverage_model.cpp   : 376)   Estimated genome size (ignoring repeats): 31290\n",
            "  0:00:01.418     2M / 41M   INFO    General                 (genomic_info_filler.cpp   :  56)   Mean coverage was calculated as 4\n",
            "  0:00:01.419     2M / 41M   INFO    General                 (genomic_info_filler.cpp   :  71)   EC coverage threshold value was calculated as 2\n",
            "  0:00:01.419     2M / 41M   INFO    General                 (genomic_info_filler.cpp   :  72)   Trusted kmer low bound: 0\n",
            "  0:00:01.419     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Gap Closer (id: early_gapcloser)\n",
            "  0:00:01.432    15M / 41M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:01.434    15M / 41M   INFO    General                 (gap_closer.cpp            : 102)   Total edges in tip neighborhood: 5632 out of 10303, length: 264345\n",
            "  0:00:01.435    15M / 41M   INFO    General                 (edge_index.hpp            : 196)   Using small index (max_id = 10358)\n",
            "  0:00:01.442    15M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:01.484    15M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 155586 kmers, 113128 bytes occupied (5.81687 bits per kmer).\n",
            "  0:00:01.486    17M / 41M   INFO    General                 (edge_index_builders.hpp   : 266)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:01.504    17M / 41M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:01.695    17M / 41M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 61698 reads processed\n",
            "  0:00:01.704    15M / 41M   INFO    General                 (gap_closer.cpp            : 491)   Initializing gap closer\n",
            "  0:00:01.704    15M / 41M   INFO   GapCloser                (gap_closer.cpp            : 406)   Collecting gap candidates\n",
            "  0:00:01.706    15M / 41M   INFO   GapCloser                (gap_closer.cpp            : 410)   Total 139 tips collected, total 172 connection candidates\n",
            "  0:00:01.707    15M / 41M   INFO   GapCloser                (gap_closer.cpp            : 431)   Closing short gaps complete: filled 2 gaps after checking 170 candidates\n",
            "  0:00:01.708    15M / 41M   INFO    General                 (gap_closer.cpp            : 495)   Gap closer done\n",
            "  0:00:01.710     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Raw Simplification (id: raw_simplification)\n",
            "  0:00:01.710     2M / 41M   INFO    General                 (simplification.cpp        : 129)   PROCEDURE == Initial cleaning\n",
            "  0:00:01.710     2M / 41M   INFO    General                 (simplification.cpp        :  63)   Most init cleaning disabled on main iteration\n",
            "  0:00:01.710     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Self conjugate edge remover\n",
            "  0:00:01.711     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:01.711     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification (id: simplification)\n",
            "  0:00:01.711     2M / 41M   INFO    General                 (simplification.cpp        : 397)   Graph simplification started\n",
            "  0:00:01.711     2M / 41M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:01.711     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:01.711     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.720     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 1681 times\n",
            "  0:00:01.721     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.725     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 24 times\n",
            "  0:00:01.725     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.726     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:01.726     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:01.726     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.726     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 2 times\n",
            "  0:00:01.726     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 1 times\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:01.727     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:01.727     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.727     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 1 times\n",
            "  0:00:01.728     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.728     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 13 times\n",
            "  0:00:01.728     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:01.729     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.729     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 17 times\n",
            "  0:00:01.729     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.730     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 3 times\n",
            "  0:00:01.730     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.730     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 2 times\n",
            "  0:00:01.730     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:01.730     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.730     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 1 times\n",
            "  0:00:01.730     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.730     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 1 times\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:01.731     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:01.731     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.731     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.732     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:01.732     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:01.732     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.732     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.732     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.732     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.732     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.732     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 2 times\n",
            "  0:00:01.732     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:01.732     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.733     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 2 times\n",
            "  0:00:01.733     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.733     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 1 times\n",
            "  0:00:01.733     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.734     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:01.734     2M / 41M   INFO    General                 (simplification.cpp        : 402)   PROCEDURE == Simplification cycle, iteration 12\n",
            "  0:00:01.734     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.734     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.734     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.734     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.734     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Low coverage edge remover\n",
            "  0:00:01.734     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Low coverage edge remover triggered 0 times\n",
            "  0:00:01.735     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Gap Closer (id: late_gapcloser)\n",
            "  0:00:01.741    15M / 41M   INFO    General                 (edge_index.hpp            : 132)   Size of edge index entries: 12/8\n",
            "  0:00:01.742    16M / 41M   INFO    General                 (gap_closer.cpp            : 102)   Total edges in tip neighborhood: 3662 out of 3715, length: 231189\n",
            "  0:00:01.742    16M / 41M   INFO    General                 (edge_index.hpp            : 196)   Using small index (max_id = 10358)\n",
            "  0:00:01.747    16M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:01.778    16M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 117159 kmers, 85368 bytes occupied (5.82921 bits per kmer).\n",
            "  0:00:01.779    17M / 41M   INFO    General                 (edge_index_builders.hpp   : 266)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:01.792    17M / 41M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:01.960    17M / 41M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 61698 reads processed\n",
            "  0:00:01.975    16M / 41M   INFO    General                 (gap_closer.cpp            : 491)   Initializing gap closer\n",
            "  0:00:01.975    16M / 41M   INFO   GapCloser                (gap_closer.cpp            : 406)   Collecting gap candidates\n",
            "  0:00:01.978    16M / 41M   INFO   GapCloser                (gap_closer.cpp            : 410)   Total 37 tips collected, total 59 connection candidates\n",
            "  0:00:01.979    16M / 41M   INFO   GapCloser                (gap_closer.cpp            : 431)   Closing short gaps complete: filled 0 gaps after checking 59 candidates\n",
            "  0:00:01.979    16M / 41M   INFO    General                 (gap_closer.cpp            : 495)   Gap closer done\n",
            "  0:00:01.981     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
            "  0:00:01.981     2M / 41M   INFO    General                 (simplification.cpp        : 189)   PROCEDURE == Post simplification\n",
            "  0:00:01.984     2M / 41M   INFO    General                 (graph_simplification.hpp  : 455)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:01.985     2M / 41M   INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:01.985     2M / 41M   INFO    General                 (graph_simplification.hpp  : 646)   Creating parallel br instance\n",
            "  0:00:01.985     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.986     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.986     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.986     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.986     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Tip clipper\n",
            "  0:00:01.987     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Tip clipper triggered 0 times\n",
            "  0:00:01.987     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Bulge remover\n",
            "  0:00:01.987     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Bulge remover triggered 0 times\n",
            "  0:00:01.988     2M / 41M   INFO    General                 (simplification.cpp        : 348)   Disrupting self-conjugate edges\n",
            "  0:00:01.988     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 168)   Running Removing isolated edges\n",
            "  0:00:01.990     2M / 41M   INFO   Simplification           (parallel_processing.hpp   : 171)   Removing isolated edges triggered 1780 times\n",
            "  0:00:01.991     2M / 41M   INFO    General                 (simplification.cpp        : 508)   After simplification:\n",
            "  0:00:01.991     2M / 41M   INFO    General                 (simplification.cpp        : 509)     Average coverage = 90.0325\n",
            "  0:00:01.991     2M / 41M   INFO    General                 (simplification.cpp        : 510)     Total length = 11291\n",
            "  0:00:01.991     2M / 41M   INFO    General                 (simplification.cpp        : 511)     Median edge length: 246\n",
            "  0:00:01.991     2M / 41M   INFO    General                 (simplification.cpp        : 512)     Edges: 155\n",
            "  0:00:01.991     2M / 41M   INFO    General                 (simplification.cpp        : 513)     Vertices: 176\n",
            "  0:00:01.991     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
            "  0:00:01.991     2M / 41M   INFO    General                 (graph_pack_helpers.cpp    :  44)   Index refill\n",
            "  0:00:01.991     2M / 41M   INFO    General                 (edge_index.hpp            : 175)   Using small index (max_id = 10358)\n",
            "  0:00:01.992     2M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 410)   Building perfect hash indices\n",
            "  0:00:01.995     2M / 41M   INFO   K-mer Index Building     (kmer_index_builder.hpp    : 446)   Index built. Total 11258 kmers, 8896 bytes occupied (6.32155 bits per kmer).\n",
            "  0:00:01.995     2M / 41M   INFO    General                 (edge_index_builders.hpp   : 253)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:01.997     2M / 41M   INFO    General                 (graph_pack_helpers.cpp    :  54)   Normalizing k-mer map. Total 4698 kmers to process\n",
            "  0:00:01.998     2M / 41M   INFO    General                 (kmer_mapper.hpp           :  87)   Total 72 kmers with non-trivial roots\n",
            "  0:00:01.998     2M / 41M   INFO    General                 (graph_pack_helpers.cpp    :  56)   Normalizing done\n",
            "  0:00:01.998     2M / 41M   INFO    General                 (mismatch_correction.cpp   : 392)   Collect potential mismatches\n",
            "  0:00:02.004     2M / 41M   INFO    General                 (mismatch_correction.cpp   : 193)   Total 40 edges (out of 155) with 241 potential mismatch positions (6.025 positions per edge)\n",
            "  0:00:02.005     2M / 41M   INFO    General                 (mismatch_correction.cpp   : 394)   Potential mismatches collected\n",
            "  0:00:02.005     2M / 41M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:02.297     2M / 41M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 249872 reads processed\n",
            "  0:00:02.297     2M / 41M   INFO    General                 (mismatch_correction.cpp   : 387)   All edges processed\n",
            "  0:00:02.298     2M / 41M   INFO    General                 (mismatch_correction.cpp   : 442)   Corrected 0 nucleotides\n",
            "  0:00:02.298     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:02.298     2M / 41M   INFO    General                 (contig_output.hpp         :  20)   Outputting contigs to \"/content/genome_assembly/spades/K77/before_rr.fasta\"\n",
            "  0:00:02.299     2M / 41M   INFO    General                 (contig_output_stage.cpp   : 155)   Writing GFA graph to \"/content/genome_assembly/spades/K77/assembly_graph_after_simplification.gfa\"\n",
            "  0:00:02.299     2M / 41M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
            "  0:00:02.299     2M / 41M   INFO    General                 (graph_pack_helpers.cpp    :  54)   Normalizing k-mer map. Total 4698 kmers to process\n",
            "  0:00:02.299     2M / 41M   INFO    General                 (graph_pack_helpers.cpp    :  56)   Normalizing done\n",
            "  0:00:02.300     2M / 41M   INFO    General                 (pair_info_count.cpp       : 157)   Min edge length for estimation: 246\n",
            "  0:00:02.300     2M / 41M   INFO    General                 (pair_info_count.cpp       : 168)   Estimating insert size for library #0\n",
            "  0:00:02.300     2M / 41M   INFO    General                 (pair_info_count.cpp       :  41)   Selecting usual mapper\n",
            "  0:00:02.300     2M / 41M   INFO    General                 (paired_info_utils.cpp     :  87)   Estimating insert size (takes a while)\n",
            "  0:00:02.331    50M / 65M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:02.513    50M / 65M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 61698 reads processed\n",
            "  0:00:02.638    50M / 65M   INFO    General                 (paired_info_utils.cpp     : 104)   Edge pairs: 519\n",
            "  0:00:02.639    50M / 65M   INFO    General                 (paired_info_utils.cpp     : 106)   8854 paired reads (14.3505% of all) aligned to long edges\n",
            "  0:00:02.639     2M / 65M   INFO    General                 (pair_info_count.cpp       : 188)     Insert size = 108.644, deviation = 28.0569, left quantile = 81, right quantile = 148, read length = 151\n",
            "  0:00:02.639     2M / 65M   WARN    General                 (pair_info_count.cpp       : 195)   Estimated mean insert size 108.644 is very small compared to read length 151\n",
            "  0:00:02.639     2M / 65M   INFO    General                 (pair_info_count.cpp       : 203)   Filtering data for library #0\n",
            "  0:00:02.639     2M / 65M   INFO    General                 (pair_info_count.cpp       :  41)   Selecting usual mapper\n",
            "  0:00:02.640     2M / 65M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:02.765     2M / 65M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 61698 reads processed\n",
            "  0:00:02.765     2M / 65M   INFO    General                 (pair_info_count.cpp       : 207)   Mapping library #0\n",
            "  0:00:02.765     2M / 65M   INFO    General                 (pair_info_count.cpp       : 209)   Mapping paired reads (takes a while)\n",
            "  0:00:02.765     2M / 65M   INFO    General                 (pair_info_count.cpp       :  41)   Selecting usual mapper\n",
            "  0:00:02.766     2M / 65M   INFO    General                 (paired_info_utils.cpp     : 138)   Left insert size quantile 81, right insert size quantile 148, filtering threshold 2, rounding threshold 0\n",
            "  0:00:02.771    15M / 65M   INFO    General                 (sequence_mapper_notifier.h:  64)   Starting sequence mapping\n",
            "  0:00:02.912    16M / 65M   INFO    General                 (sequence_mapper_notifier.h: 103)   Total 61698 reads processed\n",
            "  0:00:02.922     2M / 65M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Distance Estimation (id: distance_estimation)\n",
            "  0:00:02.922     2M / 65M   INFO    General                 (distance_estimation.cpp   :  42)   Processing library #0\n",
            "  0:00:02.922     2M / 65M   INFO    General                 (distance_estimation_utils.:  23)   Estimating distances\n",
            "  0:00:02.922     2M / 65M   INFO   DistanceEstimator        (distance_estimation.hpp   : 132)   Using SIMPLE distance estimator\n",
            "  0:00:02.933     2M / 65M   INFO    General                 (distance_estimation_utils.: 131)   Refining clustered pair information\n",
            "  0:00:02.934     2M / 65M   INFO    General                 (distance_estimation_utils.: 133)   The refining of clustered pair information has been finished\n",
            "  0:00:02.934     2M / 65M   INFO    General                 (distance_estimation_utils.: 135)   Improving paired information\n",
            "  0:00:02.953     2M / 65M   INFO   PairInfoImprover         (pair_info_improver.hpp    :  78)   Paired info stats: missing = 2; contradictional = 0\n",
            "  0:00:02.977     2M / 65M   INFO   PairInfoImprover         (pair_info_improver.hpp    :  78)   Paired info stats: missing = 0; contradictional = 0\n",
            "  0:00:02.977     2M / 65M   INFO    General                 (distance_estimation_utils.:  80)   Filling scaffolding index\n",
            "  0:00:02.977     2M / 65M   INFO    General                 (distance_estimation_utils.:  23)   Estimating distances\n",
            "  0:00:02.978     2M / 65M   INFO   DistanceEstimator        (distance_estimation.hpp   : 132)   Using SMOOTHING distance estimator\n",
            "  0:00:02.995     2M / 65M   INFO    General                 (distance_estimation.cpp   :  51)   Clearing raw paired index\n",
            "  0:00:02.995     2M / 65M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
            "  0:00:02.996     2M / 65M   INFO    General                 (repeat_resolving.cpp      :  88)   Using Path-Extend repeat resolving\n",
            "  0:00:02.996     2M / 65M   INFO    General                 (launcher.cpp              : 603)   ExSPAnder repeat resolving tool started\n",
            "  0:00:02.996     2M / 65M   INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 33.2247 calculated of length 5680\n",
            "  0:00:02.996     2M / 65M   INFO    General                 (launcher.cpp              : 420)   Creating main extenders, unique edge length = 2000\n",
            "  0:00:02.996     2M / 65M   INFO    General                 (extenders_logic.cpp       : 341)   Estimated coverage of library #0 is 90.0325\n",
            "  0:00:02.996     2M / 65M   INFO    General                 (extenders_logic.cpp       : 352)   Creating extender; library index size: 303\n",
            "  0:00:02.997     2M / 65M   INFO    General                 (extenders_logic.cpp       : 341)   Estimated coverage of library #0 is 90.0325\n",
            "  0:00:02.997     2M / 65M   INFO    General                 (extenders_logic.cpp       : 352)   Creating extender; library index size: 303\n",
            "  0:00:02.997     2M / 66M   INFO    General                 (extenders_logic.cpp       : 550)   Using 1 paired-end library\n",
            "  0:00:02.997     2M / 66M   INFO    General                 (extenders_logic.cpp       : 551)   Using 1 paired-end scaffolding library\n",
            "  0:00:02.997     2M / 66M   INFO    General                 (extenders_logic.cpp       : 552)   Using 0 single read libraries\n",
            "  0:00:02.997     2M / 66M   INFO    General                 (launcher.cpp              : 449)   Total number of extenders is 3\n",
            "  0:00:02.997     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 0 paths from 77 (0%)\n",
            "  0:00:02.997     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 8 paths from 77 (10%)\n",
            "  0:00:02.999     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 16 paths from 77 (20%)\n",
            "  0:00:02.999     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 24 paths from 77 (31%)\n",
            "  0:00:03.005     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 32 paths from 77 (41%)\n",
            "  0:00:03.006     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 40 paths from 77 (51%)\n",
            "  0:00:03.006     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 48 paths from 77 (62%)\n",
            "  0:00:03.006     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 56 paths from 77 (72%)\n",
            "  0:00:03.007     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 64 paths from 77 (83%)\n",
            "  0:00:03.007     2M / 66M   INFO    General                 (path_extenders.cpp        :  36)   Processed 72 paths from 77 (93%)\n",
            "  0:00:03.007     2M / 66M   INFO    General                 (launcher.cpp              : 252)   Finalizing paths\n",
            "  0:00:03.007     2M / 66M   INFO    General                 (launcher.cpp              : 254)   Deduplicating paths\n",
            "  0:00:03.007     2M / 66M   INFO    General                 (launcher.cpp              : 258)   Paths deduplicated\n",
            "  0:00:03.007     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
            "  0:00:03.007     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
            "  0:00:03.007     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
            "  0:00:03.008     2M / 66M   INFO   OverlapRemover           (overlap_remover.hpp       : 117)   Marking start/end overlaps\n",
            "  0:00:03.008     2M / 66M   INFO   OverlapRemover           (overlap_remover.hpp       : 120)   Marking remaining overlaps\n",
            "  0:00:03.008     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
            "  0:00:03.008     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
            "  0:00:03.008     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
            "  0:00:03.008     2M / 66M   INFO    General                 (launcher.cpp              : 275)   Paths finalized\n",
            "  0:00:03.008     2M / 66M   INFO    General                 (launcher.cpp              : 456)   Closing gaps in paths\n",
            "  0:00:03.009     2M / 66M   INFO    General                 (launcher.cpp              : 486)   Gap closing completed\n",
            "  0:00:03.015     2M / 66M   INFO    General                 (launcher.cpp              : 304)   Traversing tandem repeats\n",
            "  0:00:03.016     2M / 66M   INFO    General                 (launcher.cpp              : 314)   Traversed 0 loops\n",
            "  0:00:03.016     2M / 66M   INFO    General                 (launcher.cpp              : 252)   Finalizing paths\n",
            "  0:00:03.016     2M / 66M   INFO    General                 (launcher.cpp              : 254)   Deduplicating paths\n",
            "  0:00:03.016     2M / 66M   INFO    General                 (launcher.cpp              : 258)   Paths deduplicated\n",
            "  0:00:03.017     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
            "  0:00:03.017     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
            "  0:00:03.017     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
            "  0:00:03.017     2M / 66M   INFO   OverlapRemover           (overlap_remover.hpp       : 117)   Marking start/end overlaps\n",
            "  0:00:03.017     2M / 66M   INFO   OverlapRemover           (overlap_remover.hpp       : 120)   Marking remaining overlaps\n",
            "  0:00:03.017     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
            "  0:00:03.017     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
            "  0:00:03.017     2M / 66M   INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
            "  0:00:03.017     2M / 66M   INFO    General                 (launcher.cpp              : 275)   Paths finalized\n",
            "  0:00:03.017     2M / 66M   INFO    General                 (launcher.cpp              : 666)   ExSPAnder repeat resolving tool finished\n",
            "  0:00:03.018     2M / 66M   INFO   StageManager             (stage.cpp                 : 189)   STAGE == Contig Output (id: contig_output)\n",
            "  0:00:03.018     2M / 66M   INFO    General                 (contig_output.hpp         :  20)   Outputting contigs to \"/content/genome_assembly/spades/K77/before_rr.fasta\"\n",
            "  0:00:03.019     2M / 66M   INFO    General                 (contig_output_stage.cpp   : 155)   Writing GFA graph to \"/content/genome_assembly/spades/K77/assembly_graph_with_scaffolds.gfa\"\n",
            "  0:00:03.019     2M / 66M   INFO    General                 (contig_output_stage.cpp   : 169)   Outputting FastG graph to \"/content/genome_assembly/spades/K77/assembly_graph.fastg\"\n",
            "  0:00:03.021     2M / 66M   INFO    General                 (contig_output_stage.cpp   : 200)   Breaking scaffolds\n",
            "  0:00:03.021     2M / 66M   INFO    General                 (contig_output_stage.cpp   : 101)   Outputting contigs to /content/genome_assembly/spades/K77/final_contigs.fasta\n",
            "  0:00:03.021     2M / 66M   INFO    General                 (contig_output_stage.cpp   : 107)   Outputting FastG paths to /content/genome_assembly/spades/K77/final_contigs.paths\n",
            "  0:00:03.022     2M / 66M   INFO    General                 (contig_output_stage.cpp   : 101)   Outputting contigs to /content/genome_assembly/spades/K77/scaffolds.fasta\n",
            "  0:00:03.022     2M / 66M   INFO    General                 (contig_output_stage.cpp   : 107)   Outputting FastG paths to /content/genome_assembly/spades/K77/scaffolds.paths\n",
            "  0:00:03.023     2M / 66M   INFO    General                 (contig_output_stage.cpp   : 114)   Populating GFA with scaffold paths\n",
            "  0:00:03.025     2M / 66M   INFO    General                 (pipeline.cpp              : 292)   SPAdes finished\n",
            "  0:00:03.026     1M / 66M   INFO    General                 (main.cpp                  : 131)   Assembling time: 0 hours 0 minutes 3 seconds\n",
            "\n",
            "===== K77 finished. \n",
            "\n",
            "\n",
            "===== Copy files started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/copy_files.py /content/genome_assembly/spades/K77/before_rr.fasta /content/genome_assembly/spades/before_rr.fasta /content/genome_assembly/spades/K77/assembly_graph_after_simplification.gfa /content/genome_assembly/spades/assembly_graph_after_simplification.gfa /content/genome_assembly/spades/K77/final_contigs.fasta /content/genome_assembly/spades/contigs.fasta /content/genome_assembly/spades/K77/first_pe_contigs.fasta /content/genome_assembly/spades/first_pe_contigs.fasta /content/genome_assembly/spades/K77/strain_graph.gfa /content/genome_assembly/spades/strain_graph.gfa /content/genome_assembly/spades/K77/scaffolds.fasta /content/genome_assembly/spades/scaffolds.fasta /content/genome_assembly/spades/K77/scaffolds.paths /content/genome_assembly/spades/scaffolds.paths /content/genome_assembly/spades/K77/assembly_graph_with_scaffolds.gfa /content/genome_assembly/spades/assembly_graph_with_scaffolds.gfa /content/genome_assembly/spades/K77/assembly_graph.fastg /content/genome_assembly/spades/assembly_graph.fastg /content/genome_assembly/spades/K77/final_contigs.paths /content/genome_assembly/spades/contigs.paths\n",
            "\n",
            "\n",
            "===== Copy files finished. \n",
            "\n",
            "\n",
            "===== Assembling finished. \n",
            "\n",
            "\n",
            "===== Mismatch correction started. \n",
            "\n",
            "\n",
            "===== Mismatch correction contigs started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/correction_iteration_script.py --corrected /content/genome_assembly/spades/contigs.fasta --assembled /content/genome_assembly/spades/misc/assembled_contigs.fasta --assembly_type contigs --output_dir /content/genome_assembly/spades --bin_home /content/spades/bin\n",
            "\n",
            "== Processing of contigs\n",
            "== Running contig polishing tool: /content/spades/bin/spades-corrector-core /content/genome_assembly/spades/mismatch_corrector/contigs/configs/corrector.info /content/genome_assembly/spades/misc/assembled_contigs.fasta\n",
            "== Dataset description file was created: /content/genome_assembly/spades/mismatch_corrector/contigs/configs/corrector.info\n",
            "Run: /content/spades/bin/spades-corrector-core /content/genome_assembly/spades/mismatch_corrector/contigs/configs/corrector.info /content/genome_assembly/spades/misc/assembled_contigs.fasta\n",
            "\"/content/genome_assembly/spades/mismatch_corrector/contigs/configs/log.properties\"  0:00:00.000     1M / 15M   INFO    General                 (main.cpp                  :  57)   Starting mismatch corrector, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.002     1M / 15M   INFO    General                 (main.cpp                  :  58)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.002     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 173)   Splitting assembly...\n",
            "  0:00:00.002     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 174)   Assembly file: \"/content/genome_assembly/spades/misc/assembled_contigs.fasta\"\n",
            "  0:00:00.006     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 126)   Running bwa index ...: /content/spades/bin/spades-bwa index /content/genome_assembly/spades/misc/assembled_contigs.fasta\n",
            "[bwa_index] Pack FASTA... 0.00 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[bwa_index] 0.00 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.00 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.00 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 0.00 sec\n",
            "[main] Version: 0.7.16a-r1185-dirty\n",
            "[main] CMD: /content/spades/bin/spades-bwa index /content/genome_assembly/spades/misc/assembled_contigs.fasta\n",
            "[main] Real time: 0.041 sec; CPU: 0.009 sec\n",
            "  0:00:00.050     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 188)   Processing paired sublib of number 0\n",
            "  0:00:00.050     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 189)   /content/7_S4_L001_R1_001.fastq /content/7_S4_L001_R2_001.fastq\n",
            "  0:00:00.051     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 151)   Running bwa mem ...:/content/spades/bin/spades-bwa mem  -v 1 -t 2  /content/genome_assembly/spades/misc/assembled_contigs.fasta /content/7_S4_L001_R1_001.fastq /content/7_S4_L001_R2_001.fastq   > /content/genome_assembly/spades/tmp/corrector_akont457/lib0_c8G1Ak/tmp.sam\n",
            "[main] Version: 0.7.16a-r1185-dirty\n",
            "[main] CMD: /content/spades/bin/spades-bwa mem -v 1 -t 2 /content/genome_assembly/spades/misc/assembled_contigs.fasta /content/7_S4_L001_R1_001.fastq /content/7_S4_L001_R2_001.fastq\n",
            "[main] Real time: 3.561 sec; CPU: 6.441 sec\n",
            "  0:00:03.621     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 197)   Adding samfile \"/content/genome_assembly/spades/tmp/corrector_akont457/lib0_c8G1Ak/tmp.sam\"\n",
            "  0:00:04.086     5M / 41M   INFO   DatasetProcessor         (dataset_processor.cpp     : 225)   Processing contigs\n",
            "  0:00:06.153     5M / 41M   INFO   DatasetProcessor         (dataset_processor.cpp     : 245)   Gluing processed contigs\n",
            "  0:00:06.172     1M / 41M   INFO    General                 (main.cpp                  :  71)   Correcting time: 0 hours 0 minutes 6 seconds\n",
            "\n",
            "===== Mismatch correction contigs finished. \n",
            "\n",
            "\n",
            "===== Mismatch correction scaffolds started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/correction_iteration_script.py --corrected /content/genome_assembly/spades/scaffolds.fasta --assembled /content/genome_assembly/spades/misc/assembled_scaffolds.fasta --assembly_type scaffolds --output_dir /content/genome_assembly/spades --bin_home /content/spades/bin\n",
            "\n",
            "== Processing of scaffolds\n",
            "== Running contig polishing tool: /content/spades/bin/spades-corrector-core /content/genome_assembly/spades/mismatch_corrector/scaffolds/configs/corrector.info /content/genome_assembly/spades/misc/assembled_scaffolds.fasta\n",
            "== Dataset description file was created: /content/genome_assembly/spades/mismatch_corrector/scaffolds/configs/corrector.info\n",
            "Run: /content/spades/bin/spades-corrector-core /content/genome_assembly/spades/mismatch_corrector/scaffolds/configs/corrector.info /content/genome_assembly/spades/misc/assembled_scaffolds.fasta\n",
            "\"/content/genome_assembly/spades/mismatch_corrector/scaffolds/configs/log.properties\"  0:00:00.000     1M / 15M   INFO    General                 (main.cpp                  :  57)   Starting mismatch corrector, built from N/A, git revision 29e9c9b9f6689a8ba3ff4247e5195e68a81e4f11\n",
            "  0:00:00.007     1M / 15M   INFO    General                 (main.cpp                  :  58)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.011     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 173)   Splitting assembly...\n",
            "  0:00:00.011     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 174)   Assembly file: \"/content/genome_assembly/spades/misc/assembled_scaffolds.fasta\"\n",
            "  0:00:00.015     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 126)   Running bwa index ...: /content/spades/bin/spades-bwa index /content/genome_assembly/spades/misc/assembled_scaffolds.fasta\n",
            "[bwa_index] Pack FASTA... 0.00 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[bwa_index] 0.00 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.00 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.00 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 0.00 sec\n",
            "[main] Version: 0.7.16a-r1185-dirty\n",
            "[main] CMD: /content/spades/bin/spades-bwa index /content/genome_assembly/spades/misc/assembled_scaffolds.fasta\n",
            "[main] Real time: 0.054 sec; CPU: 0.010 sec\n",
            "  0:00:00.073     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 188)   Processing paired sublib of number 0\n",
            "  0:00:00.084     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 189)   /content/7_S4_L001_R1_001.fastq /content/7_S4_L001_R2_001.fastq\n",
            "  0:00:00.098     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 151)   Running bwa mem ...:/content/spades/bin/spades-bwa mem  -v 1 -t 2  /content/genome_assembly/spades/misc/assembled_scaffolds.fasta /content/7_S4_L001_R1_001.fastq /content/7_S4_L001_R2_001.fastq   > /content/genome_assembly/spades/tmp/corrector_d2qoqsqo/lib0_ERm4l4/tmp.sam\n",
            "[main] Version: 0.7.16a-r1185-dirty\n",
            "[main] CMD: /content/spades/bin/spades-bwa mem -v 1 -t 2 /content/genome_assembly/spades/misc/assembled_scaffolds.fasta /content/7_S4_L001_R1_001.fastq /content/7_S4_L001_R2_001.fastq\n",
            "[main] Real time: 4.707 sec; CPU: 6.571 sec\n",
            "  0:00:04.830     1M / 15M   INFO   DatasetProcessor         (dataset_processor.cpp     : 197)   Adding samfile \"/content/genome_assembly/spades/tmp/corrector_d2qoqsqo/lib0_ERm4l4/tmp.sam\"\n",
            "  0:00:05.346     5M / 41M   INFO   DatasetProcessor         (dataset_processor.cpp     : 225)   Processing contigs\n",
            "  0:00:07.368     5M / 41M   INFO   DatasetProcessor         (dataset_processor.cpp     : 245)   Gluing processed contigs\n",
            "  0:00:07.370     1M / 41M   INFO    General                 (main.cpp                  :  71)   Correcting time: 0 hours 0 minutes 7 seconds\n",
            "\n",
            "===== Mismatch correction scaffolds finished. \n",
            "\n",
            "\n",
            "===== Mismatch correction finished. \n",
            "\n",
            "\n",
            "===== Breaking scaffolds started. \n",
            "\n",
            "\n",
            "== Running: /usr/bin/python3 /content/spades/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /content/genome_assembly/spades/scaffolds.fasta --misc_dir /content/genome_assembly/spades/misc --threshold_for_breaking_scaffolds 3\n",
            "\n",
            "\n",
            "===== Breaking scaffolds finished. \n",
            "\n",
            "\n",
            "===== Terminate started. \n",
            "\n",
            "\n",
            "===== Terminate finished. \n",
            "\n",
            " * Corrected reads are in /content/genome_assembly/spades/corrected/\n",
            " * Assembled contigs are in /content/genome_assembly/spades/contigs.fasta\n",
            " * Assembled scaffolds are in /content/genome_assembly/spades/scaffolds.fasta\n",
            " * Paths in the assembly graph corresponding to the contigs are in /content/genome_assembly/spades/contigs.paths\n",
            " * Paths in the assembly graph corresponding to the scaffolds are in /content/genome_assembly/spades/scaffolds.paths\n",
            " * Assembly graph is in /content/genome_assembly/spades/assembly_graph.fastg\n",
            " * Assembly graph in GFA format is in /content/genome_assembly/spades/assembly_graph_with_scaffolds.gfa\n",
            "\n",
            "======= SPAdes pipeline finished WITH WARNINGS!\n",
            "\n",
            "=== Error correction and assembling warnings:\n",
            " * 0:00:01.418     2M / 41M   WARN    General                 (kmer_coverage_model.cpp   : 328)   Valley value was estimated improperly, reset to 2\n",
            " * 0:00:02.639     2M / 65M   WARN    General                 (pair_info_count.cpp       : 195)   Estimated mean insert size 108.644 is very small compared to read length 151\n",
            "======= Warnings saved to /content/genome_assembly/spades/warnings.log\n",
            "\n",
            "SPAdes log can be found here: /content/genome_assembly/spades/spades.log\n",
            "\n",
            "Thank you for using SPAdes! If you use it in your research, please cite:\n",
            "\n",
            "  Prjibelski, A., Antipov, D., Meleshko, D., Lapidus, A. and Korobeynikov, A., 2020. Using SPAdes de novo assembler. Current protocols in bioinformatics, 70(1), p.e102.\n",
            "  doi.org/10.1002/cpbi.102\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! spades/bin/spades.py --careful -1 7_S4_L001_R1_001.fastq -2 7_S4_L001_R2_001.fastq -o genome_assembly/spades"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5B_kQQmZrg7s"
      },
      "source": [
        "# 2. genome assembly quality control"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3OARjN7crjYg"
      },
      "source": [
        "## 2.1 prepare environment for assembly quality control"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "HTRptUwLruS2"
      },
      "outputs": [],
      "source": [
        "! mkdir genome_assembly/assembly_analysis"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "doL5W0VprxxI"
      },
      "source": [
        "## 2.2 run quast"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lwxmbRP4sB0Y",
        "outputId": "80da1153-05b6-4c93-dc2a-ddb975f61387"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: line 1: quast/quast.py: No such file or directory\n"
          ]
        }
      ],
      "source": [
        "! quast/quast.py -o genome_assembly/assembly_analysis/quast_spades -m 0 --threads 1 genome_assembly/spades/K77/scaffolds.fasta genome_assembly/spades/scaffolds.fasta genome_assembly/k11/platanus_k11_contig.fa"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "m8CszQ2EsCqz"
      },
      "outputs": [],
      "source": [
        "! mkdir prokka"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "C0s-IkkdtDS3",
        "outputId": "d0c4f667-1bfa-4f38-d2ed-f5a401939cdd"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "PREFIX=/usr/local\n",
            "Unpacking payload ...\n",
            "Extracting _libgcc_mutex-0.1-conda_forge.tar.bz2\n",
            "Extracting ca-certificates-2025.4.26-hbd8a1cb_0.conda\n",
            "Extracting ld_impl_linux-64-2.43-h712a8e2_4.conda\n",
            "Extracting libgomp-15.1.0-h767d61c_2.conda\n",
            "Extracting pybind11-abi-4-hd8ed1ab_3.tar.bz2\n",
            "Extracting python_abi-3.12-7_cp312.conda\n",
            "Extracting tzdata-2025b-h78e105d_0.conda\n",
            "Extracting _openmp_mutex-4.5-2_gnu.tar.bz2\n",
            "Extracting libgcc-15.1.0-h767d61c_2.conda\n",
            "Extracting c-ares-1.34.5-hb9d3cd8_0.conda\n",
            "Extracting libexpat-2.7.0-h5888daf_0.conda\n",
            "Extracting libffi-3.4.6-h2dba641_1.conda\n",
            "Extracting libgcc-ng-15.1.0-h69a702a_2.conda\n",
            "Extracting libiconv-1.18-h4ce23a2_1.conda\n",
            "Extracting liblzma-5.8.1-hb9d3cd8_1.conda\n",
            "Extracting libstdcxx-15.1.0-h8f9b012_2.conda\n",
            "Extracting libzlib-1.3.1-hb9d3cd8_2.conda\n",
            "Extracting ncurses-6.5-h2d0b736_3.conda\n",
            "Extracting openssl-3.5.0-h7b32b05_1.conda\n",
            "Extracting reproc-14.2.5.post0-hb9d3cd8_0.conda\n",
            "Extracting bzip2-1.0.8-h4bc722e_7.conda\n",
            "Extracting cpp-expected-1.1.0-hff21bea_1.conda\n",
            "Extracting fmt-11.1.4-h07f6e7f_1.conda\n",
            "Extracting keyutils-1.6.1-h166bdaf_0.tar.bz2\n",
            "Extracting libedit-3.1.20250104-pl5321h7949ede_0.conda\n",
            "Extracting libev-4.33-hd590300_2.conda\n",
            "Extracting libnsl-2.0.1-hd590300_0.conda\n",
            "Extracting libsolv-0.7.32-h7955e40_2.conda\n",
            "Extracting libsqlite-3.49.2-hee588c1_0.conda\n",
            "Extracting libssh2-1.11.1-hcf80075_0.conda\n",
            "Extracting libstdcxx-ng-15.1.0-h4852527_2.conda\n",
            "Extracting libuuid-2.38.1-h0b41bf4_0.conda\n",
            "Extracting libxcrypt-4.4.36-hd590300_1.conda\n",
            "Extracting lz4-c-1.10.0-h5888daf_1.conda\n",
            "Extracting lzo-2.10-hd590300_1001.conda\n",
            "Extracting readline-8.2-h8c095d6_2.conda\n",
            "Extracting reproc-cpp-14.2.5.post0-h5888daf_0.conda\n",
            "Extracting simdjson-3.12.3-h84d6215_0.conda\n",
            "Extracting tk-8.6.13-noxft_h4845f30_101.conda\n",
            "Extracting yaml-cpp-0.8.0-h3f2d84a_0.conda\n",
            "Extracting zstd-1.5.7-hb8e6e7a_2.conda\n",
            "Extracting icu-75.1-he02047a_0.conda\n",
            "Extracting krb5-1.21.3-h659f571_0.conda\n",
            "Extracting libnghttp2-1.64.0-h161d5f1_0.conda\n",
            "Extracting nlohmann_json-3.11.3-he02047a_1.conda\n",
            "Extracting python-3.12.10-h9e4cc4f_0_cpython.conda\n",
            "Extracting libcurl-8.13.0-h332b0f4_0.conda\n",
            "Extracting libxml2-2.13.8-h4bc477f_0.conda\n",
            "Extracting menuinst-2.2.0-py312h7900ff3_0.conda\n",
            "Extracting archspec-0.2.5-pyhd8ed1ab_0.conda\n",
            "Extracting boltons-24.0.0-pyhd8ed1ab_1.conda\n",
            "Extracting brotli-python-1.1.0-py312h2ec8cdc_2.conda\n",
            "Extracting certifi-2025.4.26-pyhd8ed1ab_0.conda\n",
            "Extracting charset-normalizer-3.4.2-pyhd8ed1ab_0.conda\n",
            "Extracting colorama-0.4.6-pyhd8ed1ab_1.conda\n",
            "Extracting distro-1.9.0-pyhd8ed1ab_1.conda\n",
            "Extracting frozendict-2.4.6-py312h66e93f0_0.conda\n",
            "Extracting hpack-4.1.0-pyhd8ed1ab_0.conda\n",
            "Extracting hyperframe-6.1.0-pyhd8ed1ab_0.conda\n",
            "Extracting idna-3.10-pyhd8ed1ab_1.conda\n",
            "Extracting jsonpointer-3.0.0-py312h7900ff3_1.conda\n",
            "Extracting libarchive-3.7.7-h75ea233_4.conda\n",
            "Extracting packaging-25.0-pyh29332c3_1.conda\n",
            "Extracting platformdirs-4.3.8-pyhe01879c_0.conda\n",
            "Extracting pluggy-1.5.0-pyhd8ed1ab_1.conda\n",
            "Extracting pycosat-0.6.6-py312h66e93f0_2.conda\n",
            "Extracting pycparser-2.22-pyh29332c3_1.conda\n",
            "Extracting pysocks-1.7.1-pyha55dd90_7.conda\n",
            "Extracting ruamel.yaml.clib-0.2.8-py312h66e93f0_1.conda\n",
            "Extracting setuptools-80.1.0-pyhff2d567_0.conda\n",
            "Extracting truststore-0.10.1-pyh29332c3_0.conda\n",
            "Extracting wheel-0.45.1-pyhd8ed1ab_1.conda\n",
            "Extracting cffi-1.17.1-py312h06ac9bb_0.conda\n",
            "Extracting h2-4.2.0-pyhd8ed1ab_0.conda\n",
            "Extracting jsonpatch-1.33-pyhd8ed1ab_1.conda\n",
            "Extracting libmamba-2.1.1-h430c389_0.conda\n",
            "Extracting pip-25.1.1-pyh8b19718_0.conda\n",
            "Extracting ruamel.yaml-0.18.10-py312h66e93f0_0.conda\n",
            "Extracting tqdm-4.67.1-pyhd8ed1ab_1.conda\n",
            "Extracting libmambapy-2.1.1-py312h07448e0_0.conda\n",
            "Extracting mamba-2.1.1-had4a41a_0.conda\n",
            "Extracting zstandard-0.23.0-py312h66e93f0_2.conda\n",
            "Extracting conda-package-streaming-0.11.0-pyhd8ed1ab_1.conda\n",
            "Extracting urllib3-2.4.0-pyhd8ed1ab_0.conda\n",
            "Extracting requests-2.32.3-pyhd8ed1ab_1.conda\n",
            "Extracting conda-package-handling-2.4.0-pyh7900ff3_2.conda\n",
            "Extracting conda-25.3.0-py312h7900ff3_0.conda\n",
            "Extracting conda-libmamba-solver-25.3.0-pyhd8ed1ab_0.conda\n",
            "\n",
            "Installing base environment...\n",
            "\n",
            "Transaction\n",
            "\n",
            "  Prefix: /usr/local\n",
            "\n",
            "  Updating specs:\n",
            "\n",
            "   - _libgcc_mutex==0.1=conda_forge\n",
            "   - ca-certificates==2025.4.26=hbd8a1cb_0\n",
            "   - ld_impl_linux-64==2.43=h712a8e2_4\n",
            "   - libgomp==15.1.0=h767d61c_2\n",
            "   - pybind11-abi==4=hd8ed1ab_3\n",
            "   - python_abi==3.12=7_cp312\n",
            "   - tzdata==2025b=h78e105d_0\n",
            "   - _openmp_mutex==4.5=2_gnu\n",
            "   - libgcc==15.1.0=h767d61c_2\n",
            "   - c-ares==1.34.5=hb9d3cd8_0\n",
            "   - libexpat==2.7.0=h5888daf_0\n",
            "   - libffi==3.4.6=h2dba641_1\n",
            "   - libgcc-ng==15.1.0=h69a702a_2\n",
            "   - libiconv==1.18=h4ce23a2_1\n",
            "   - liblzma==5.8.1=hb9d3cd8_1\n",
            "   - libstdcxx==15.1.0=h8f9b012_2\n",
            "   - libzlib==1.3.1=hb9d3cd8_2\n",
            "   - ncurses==6.5=h2d0b736_3\n",
            "   - openssl==3.5.0=h7b32b05_1\n",
            "   - reproc==14.2.5.0post0=hb9d3cd8_0\n",
            "   - bzip2==1.0.8=h4bc722e_7\n",
            "   - cpp-expected==1.1.0=hff21bea_1\n",
            "   - fmt==11.1.4=h07f6e7f_1\n",
            "   - keyutils==1.6.1=h166bdaf_0\n",
            "   - libedit==3.1.20250104=pl5321h7949ede_0\n",
            "   - libev==4.33=hd590300_2\n",
            "   - libnsl==2.0.1=hd590300_0\n",
            "   - libsolv==0.7.32=h7955e40_2\n",
            "   - libsqlite==3.49.2=hee588c1_0\n",
            "   - libssh2==1.11.1=hcf80075_0\n",
            "   - libstdcxx-ng==15.1.0=h4852527_2\n",
            "   - libuuid==2.38.1=h0b41bf4_0\n",
            "   - libxcrypt==4.4.36=hd590300_1\n",
            "   - lz4-c==1.10.0=h5888daf_1\n",
            "   - lzo==2.10=hd590300_1001\n",
            "   - readline==8.2=h8c095d6_2\n",
            "   - reproc-cpp==14.2.5.0post0=h5888daf_0\n",
            "   - simdjson==3.12.3=h84d6215_0\n",
            "   - tk==8.6.13=noxft_h4845f30_101\n",
            "   - yaml-cpp==0.8.0=h3f2d84a_0\n",
            "   - zstd==1.5.7=hb8e6e7a_2\n",
            "   - icu==75.1=he02047a_0\n",
            "   - krb5==1.21.3=h659f571_0\n",
            "   - libnghttp2==1.64.0=h161d5f1_0\n",
            "   - nlohmann_json==3.11.3=he02047a_1\n",
            "   - python==3.12.10=h9e4cc4f_0_cpython\n",
            "   - libcurl==8.13.0=h332b0f4_0\n",
            "   - libxml2==2.13.8=h4bc477f_0\n",
            "   - menuinst==2.2.0=py312h7900ff3_0\n",
            "   - archspec==0.2.5=pyhd8ed1ab_0\n",
            "   - boltons==24.0.0=pyhd8ed1ab_1\n",
            "   - brotli-python==1.1.0=py312h2ec8cdc_2\n",
            "   - certifi==2025.4.26=pyhd8ed1ab_0\n",
            "   - charset-normalizer==3.4.2=pyhd8ed1ab_0\n",
            "   - colorama==0.4.6=pyhd8ed1ab_1\n",
            "   - distro==1.9.0=pyhd8ed1ab_1\n",
            "   - frozendict==2.4.6=py312h66e93f0_0\n",
            "   - hpack==4.1.0=pyhd8ed1ab_0\n",
            "   - hyperframe==6.1.0=pyhd8ed1ab_0\n",
            "   - idna==3.10=pyhd8ed1ab_1\n",
            "   - jsonpointer==3.0.0=py312h7900ff3_1\n",
            "   - libarchive==3.7.7=h75ea233_4\n",
            "   - packaging==25.0=pyh29332c3_1\n",
            "   - platformdirs==4.3.8=pyhe01879c_0\n",
            "   - pluggy==1.5.0=pyhd8ed1ab_1\n",
            "   - pycosat==0.6.6=py312h66e93f0_2\n",
            "   - pycparser==2.22=pyh29332c3_1\n",
            "   - pysocks==1.7.1=pyha55dd90_7\n",
            "   - ruamel.yaml.clib==0.2.8=py312h66e93f0_1\n",
            "   - setuptools==80.1.0=pyhff2d567_0\n",
            "   - truststore==0.10.1=pyh29332c3_0\n",
            "   - wheel==0.45.1=pyhd8ed1ab_1\n",
            "   - cffi==1.17.1=py312h06ac9bb_0\n",
            "   - h2==4.2.0=pyhd8ed1ab_0\n",
            "   - jsonpatch==1.33=pyhd8ed1ab_1\n",
            "   - libmamba==2.1.1=h430c389_0\n",
            "   - pip==25.1.1=pyh8b19718_0\n",
            "   - ruamel.yaml==0.18.10=py312h66e93f0_0\n",
            "   - tqdm==4.67.1=pyhd8ed1ab_1\n",
            "   - libmambapy==2.1.1=py312h07448e0_0\n",
            "   - mamba==2.1.1=had4a41a_0\n",
            "   - zstandard==0.23.0=py312h66e93f0_2\n",
            "   - conda-package-streaming==0.11.0=pyhd8ed1ab_1\n",
            "   - urllib3==2.4.0=pyhd8ed1ab_0\n",
            "   - requests==2.32.3=pyhd8ed1ab_1\n",
            "   - conda-package-handling==2.4.0=pyh7900ff3_2\n",
            "   - conda==25.3.0=py312h7900ff3_0\n",
            "   - conda-libmamba-solver==25.3.0=pyhd8ed1ab_0\n",
            "\n",
            "\n",
            "  Package                         Version  Build               Channel         Size\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "  Install:\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  \u001b[32m+ _libgcc_mutex          \u001b[0m           0.1  conda_forge         conda-forge         \n",
            "  \u001b[32m+ _openmp_mutex          \u001b[0m           4.5  2_gnu               conda-forge         \n",
            "  \u001b[32m+ archspec               \u001b[0m         0.2.5  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ boltons                \u001b[0m        24.0.0  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ brotli-python          \u001b[0m         1.1.0  py312h2ec8cdc_2     conda-forge         \n",
            "  \u001b[32m+ bzip2                  \u001b[0m         1.0.8  h4bc722e_7          conda-forge         \n",
            "  \u001b[32m+ c-ares                 \u001b[0m        1.34.5  hb9d3cd8_0          conda-forge         \n",
            "  \u001b[32m+ ca-certificates        \u001b[0m     2025.4.26  hbd8a1cb_0          conda-forge         \n",
            "  \u001b[32m+ certifi                \u001b[0m     2025.4.26  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ cffi                   \u001b[0m        1.17.1  py312h06ac9bb_0     conda-forge         \n",
            "  \u001b[32m+ charset-normalizer     \u001b[0m         3.4.2  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ colorama               \u001b[0m         0.4.6  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ conda                  \u001b[0m        25.3.0  py312h7900ff3_0     conda-forge         \n",
            "  \u001b[32m+ conda-libmamba-solver  \u001b[0m        25.3.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ conda-package-handling \u001b[0m         2.4.0  pyh7900ff3_2        conda-forge         \n",
            "  \u001b[32m+ conda-package-streaming\u001b[0m        0.11.0  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ cpp-expected           \u001b[0m         1.1.0  hff21bea_1          conda-forge         \n",
            "  \u001b[32m+ distro                 \u001b[0m         1.9.0  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ fmt                    \u001b[0m        11.1.4  h07f6e7f_1          conda-forge         \n",
            "  \u001b[32m+ frozendict             \u001b[0m         2.4.6  py312h66e93f0_0     conda-forge         \n",
            "  \u001b[32m+ h2                     \u001b[0m         4.2.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ hpack                  \u001b[0m         4.1.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ hyperframe             \u001b[0m         6.1.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ icu                    \u001b[0m          75.1  he02047a_0          conda-forge         \n",
            "  \u001b[32m+ idna                   \u001b[0m          3.10  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ jsonpatch              \u001b[0m          1.33  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ jsonpointer            \u001b[0m         3.0.0  py312h7900ff3_1     conda-forge         \n",
            "  \u001b[32m+ keyutils               \u001b[0m         1.6.1  h166bdaf_0          conda-forge         \n",
            "  \u001b[32m+ krb5                   \u001b[0m        1.21.3  h659f571_0          conda-forge         \n",
            "  \u001b[32m+ ld_impl_linux-64       \u001b[0m          2.43  h712a8e2_4          conda-forge         \n",
            "  \u001b[32m+ libarchive             \u001b[0m         3.7.7  h75ea233_4          conda-forge         \n",
            "  \u001b[32m+ libcurl                \u001b[0m        8.13.0  h332b0f4_0          conda-forge         \n",
            "  \u001b[32m+ libedit                \u001b[0m  3.1.20250104  pl5321h7949ede_0    conda-forge         \n",
            "  \u001b[32m+ libev                  \u001b[0m          4.33  hd590300_2          conda-forge         \n",
            "  \u001b[32m+ libexpat               \u001b[0m         2.7.0  h5888daf_0          conda-forge         \n",
            "  \u001b[32m+ libffi                 \u001b[0m         3.4.6  h2dba641_1          conda-forge         \n",
            "  \u001b[32m+ libgcc                 \u001b[0m        15.1.0  h767d61c_2          conda-forge         \n",
            "  \u001b[32m+ libgcc-ng              \u001b[0m        15.1.0  h69a702a_2          conda-forge         \n",
            "  \u001b[32m+ libgomp                \u001b[0m        15.1.0  h767d61c_2          conda-forge         \n",
            "  \u001b[32m+ libiconv               \u001b[0m          1.18  h4ce23a2_1          conda-forge         \n",
            "  \u001b[32m+ liblzma                \u001b[0m         5.8.1  hb9d3cd8_1          conda-forge         \n",
            "  \u001b[32m+ libmamba               \u001b[0m         2.1.1  h430c389_0          conda-forge         \n",
            "  \u001b[32m+ libmambapy             \u001b[0m         2.1.1  py312h07448e0_0     conda-forge         \n",
            "  \u001b[32m+ libnghttp2             \u001b[0m        1.64.0  h161d5f1_0          conda-forge         \n",
            "  \u001b[32m+ libnsl                 \u001b[0m         2.0.1  hd590300_0          conda-forge         \n",
            "  \u001b[32m+ libsolv                \u001b[0m        0.7.32  h7955e40_2          conda-forge         \n",
            "  \u001b[32m+ libsqlite              \u001b[0m        3.49.2  hee588c1_0          conda-forge         \n",
            "  \u001b[32m+ libssh2                \u001b[0m        1.11.1  hcf80075_0          conda-forge         \n",
            "  \u001b[32m+ libstdcxx              \u001b[0m        15.1.0  h8f9b012_2          conda-forge         \n",
            "  \u001b[32m+ libstdcxx-ng           \u001b[0m        15.1.0  h4852527_2          conda-forge         \n",
            "  \u001b[32m+ libuuid                \u001b[0m        2.38.1  h0b41bf4_0          conda-forge         \n",
            "  \u001b[32m+ libxcrypt              \u001b[0m        4.4.36  hd590300_1          conda-forge         \n",
            "  \u001b[32m+ libxml2                \u001b[0m        2.13.8  h4bc477f_0          conda-forge         \n",
            "  \u001b[32m+ libzlib                \u001b[0m         1.3.1  hb9d3cd8_2          conda-forge         \n",
            "  \u001b[32m+ lz4-c                  \u001b[0m        1.10.0  h5888daf_1          conda-forge         \n",
            "  \u001b[32m+ lzo                    \u001b[0m          2.10  hd590300_1001       conda-forge         \n",
            "  \u001b[32m+ mamba                  \u001b[0m         2.1.1  had4a41a_0          conda-forge         \n",
            "  \u001b[32m+ menuinst               \u001b[0m         2.2.0  py312h7900ff3_0     conda-forge         \n",
            "  \u001b[32m+ ncurses                \u001b[0m           6.5  h2d0b736_3          conda-forge         \n",
            "  \u001b[32m+ nlohmann_json          \u001b[0m        3.11.3  he02047a_1          conda-forge         \n",
            "  \u001b[32m+ openssl                \u001b[0m         3.5.0  h7b32b05_1          conda-forge         \n",
            "  \u001b[32m+ packaging              \u001b[0m          25.0  pyh29332c3_1        conda-forge         \n",
            "  \u001b[32m+ pip                    \u001b[0m        25.1.1  pyh8b19718_0        conda-forge         \n",
            "  \u001b[32m+ platformdirs           \u001b[0m         4.3.8  pyhe01879c_0        conda-forge         \n",
            "  \u001b[32m+ pluggy                 \u001b[0m         1.5.0  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ pybind11-abi           \u001b[0m             4  hd8ed1ab_3          conda-forge         \n",
            "  \u001b[32m+ pycosat                \u001b[0m         0.6.6  py312h66e93f0_2     conda-forge         \n",
            "  \u001b[32m+ pycparser              \u001b[0m          2.22  pyh29332c3_1        conda-forge         \n",
            "  \u001b[32m+ pysocks                \u001b[0m         1.7.1  pyha55dd90_7        conda-forge         \n",
            "  \u001b[32m+ python                 \u001b[0m       3.12.10  h9e4cc4f_0_cpython  conda-forge         \n",
            "  \u001b[32m+ python_abi             \u001b[0m          3.12  7_cp312             conda-forge         \n",
            "  \u001b[32m+ readline               \u001b[0m           8.2  h8c095d6_2          conda-forge         \n",
            "  \u001b[32m+ reproc                 \u001b[0m  14.2.5.post0  hb9d3cd8_0          conda-forge         \n",
            "  \u001b[32m+ reproc-cpp             \u001b[0m  14.2.5.post0  h5888daf_0          conda-forge         \n",
            "  \u001b[32m+ requests               \u001b[0m        2.32.3  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ ruamel.yaml            \u001b[0m       0.18.10  py312h66e93f0_0     conda-forge         \n",
            "  \u001b[32m+ ruamel.yaml.clib       \u001b[0m         0.2.8  py312h66e93f0_1     conda-forge         \n",
            "  \u001b[32m+ setuptools             \u001b[0m        80.1.0  pyhff2d567_0        conda-forge         \n",
            "  \u001b[32m+ simdjson               \u001b[0m        3.12.3  h84d6215_0          conda-forge         \n",
            "  \u001b[32m+ tk                     \u001b[0m        8.6.13  noxft_h4845f30_101  conda-forge         \n",
            "  \u001b[32m+ tqdm                   \u001b[0m        4.67.1  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ truststore             \u001b[0m        0.10.1  pyh29332c3_0        conda-forge         \n",
            "  \u001b[32m+ tzdata                 \u001b[0m         2025b  h78e105d_0          conda-forge         \n",
            "  \u001b[32m+ urllib3                \u001b[0m         2.4.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ wheel                  \u001b[0m        0.45.1  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ yaml-cpp               \u001b[0m         0.8.0  h3f2d84a_0          conda-forge         \n",
            "  \u001b[32m+ zstandard              \u001b[0m        0.23.0  py312h66e93f0_2     conda-forge         \n",
            "  \u001b[32m+ zstd                   \u001b[0m         1.5.7  hb8e6e7a_2          conda-forge         \n",
            "\n",
            "  Summary:\n",
            "\n",
            "  Install: 88 packages\n",
            "\n",
            "  Total download: 0 B\n",
            "\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "\n",
            "\n",
            "Transaction starting\n",
            "\u001b[?25l\u001b[2K\u001b[0G\u001b[?25hLinking _libgcc_mutex-0.1-conda_forge\n",
            "Linking ca-certificates-2025.4.26-hbd8a1cb_0\n",
            "Linking ld_impl_linux-64-2.43-h712a8e2_4\n",
            "Linking libgomp-15.1.0-h767d61c_2\n",
            "Linking pybind11-abi-4-hd8ed1ab_3\n",
            "Linking python_abi-3.12-7_cp312\n",
            "Linking tzdata-2025b-h78e105d_0\n",
            "Linking _openmp_mutex-4.5-2_gnu\n",
            "Linking libgcc-15.1.0-h767d61c_2\n",
            "Linking c-ares-1.34.5-hb9d3cd8_0\n",
            "Linking libexpat-2.7.0-h5888daf_0\n",
            "Linking libffi-3.4.6-h2dba641_1\n",
            "Linking libgcc-ng-15.1.0-h69a702a_2\n",
            "Linking libiconv-1.18-h4ce23a2_1\n",
            "Linking liblzma-5.8.1-hb9d3cd8_1\n",
            "Linking libstdcxx-15.1.0-h8f9b012_2\n",
            "Linking libzlib-1.3.1-hb9d3cd8_2\n",
            "Linking ncurses-6.5-h2d0b736_3\n",
            "Linking openssl-3.5.0-h7b32b05_1\n",
            "Linking reproc-14.2.5.post0-hb9d3cd8_0\n",
            "Linking bzip2-1.0.8-h4bc722e_7\n",
            "Linking cpp-expected-1.1.0-hff21bea_1\n",
            "Linking fmt-11.1.4-h07f6e7f_1\n",
            "Linking keyutils-1.6.1-h166bdaf_0\n",
            "Linking libedit-3.1.20250104-pl5321h7949ede_0\n",
            "Linking libev-4.33-hd590300_2\n",
            "Linking libnsl-2.0.1-hd590300_0\n",
            "Linking libsolv-0.7.32-h7955e40_2\n",
            "Linking libsqlite-3.49.2-hee588c1_0\n",
            "Linking libssh2-1.11.1-hcf80075_0\n",
            "Linking libstdcxx-ng-15.1.0-h4852527_2\n",
            "Linking libuuid-2.38.1-h0b41bf4_0\n",
            "Linking libxcrypt-4.4.36-hd590300_1\n",
            "Linking lz4-c-1.10.0-h5888daf_1\n",
            "Linking lzo-2.10-hd590300_1001\n",
            "Linking readline-8.2-h8c095d6_2\n",
            "Linking reproc-cpp-14.2.5.post0-h5888daf_0\n",
            "Linking simdjson-3.12.3-h84d6215_0\n",
            "Linking tk-8.6.13-noxft_h4845f30_101\n",
            "Linking yaml-cpp-0.8.0-h3f2d84a_0\n",
            "Linking zstd-1.5.7-hb8e6e7a_2\n",
            "Linking icu-75.1-he02047a_0\n",
            "Linking krb5-1.21.3-h659f571_0\n",
            "Linking libnghttp2-1.64.0-h161d5f1_0\n",
            "Linking nlohmann_json-3.11.3-he02047a_1\n",
            "Linking python-3.12.10-h9e4cc4f_0_cpython\n",
            "\u001b[33m\u001b[1mwarning  libmamba\u001b[m [python-3.12.10-h9e4cc4f_0_cpython] The following files were already present in the environment:\n",
            "    - bin/python\n",
            "Linking libcurl-8.13.0-h332b0f4_0\n",
            "Linking libxml2-2.13.8-h4bc477f_0\n",
            "Linking menuinst-2.2.0-py312h7900ff3_0\n",
            "Linking archspec-0.2.5-pyhd8ed1ab_0\n",
            "Linking boltons-24.0.0-pyhd8ed1ab_1\n",
            "Linking brotli-python-1.1.0-py312h2ec8cdc_2\n",
            "Linking certifi-2025.4.26-pyhd8ed1ab_0\n",
            "Linking charset-normalizer-3.4.2-pyhd8ed1ab_0\n",
            "\u001b[33m\u001b[1mwarning  libmamba\u001b[m [charset-normalizer-3.4.2-pyhd8ed1ab_0] The following files were already present in the environment:\n",
            "    - bin/normalizer\n",
            "Linking colorama-0.4.6-pyhd8ed1ab_1\n",
            "Linking distro-1.9.0-pyhd8ed1ab_1\n",
            "\u001b[33m\u001b[1mwarning  libmamba\u001b[m [distro-1.9.0-pyhd8ed1ab_1] The following files were already present in the environment:\n",
            "    - bin/distro\n",
            "Linking frozendict-2.4.6-py312h66e93f0_0\n",
            "Linking hpack-4.1.0-pyhd8ed1ab_0\n",
            "Linking hyperframe-6.1.0-pyhd8ed1ab_0\n",
            "Linking idna-3.10-pyhd8ed1ab_1\n",
            "Linking jsonpointer-3.0.0-py312h7900ff3_1\n",
            "\u001b[33m\u001b[1mwarning  libmamba\u001b[m [jsonpointer-3.0.0-py312h7900ff3_1] The following files were already present in the environment:\n",
            "    - bin/jsonpointer\n",
            "Linking libarchive-3.7.7-h75ea233_4\n",
            "Linking packaging-25.0-pyh29332c3_1\n",
            "Linking platformdirs-4.3.8-pyhe01879c_0\n",
            "Linking pluggy-1.5.0-pyhd8ed1ab_1\n",
            "Linking pycosat-0.6.6-py312h66e93f0_2\n",
            "Linking pycparser-2.22-pyh29332c3_1\n",
            "Linking pysocks-1.7.1-pyha55dd90_7\n",
            "Linking ruamel.yaml.clib-0.2.8-py312h66e93f0_1\n",
            "Linking setuptools-80.1.0-pyhff2d567_0\n",
            "Linking truststore-0.10.1-pyh29332c3_0\n",
            "Linking wheel-0.45.1-pyhd8ed1ab_1\n",
            "\u001b[33m\u001b[1mwarning  libmamba\u001b[m [wheel-0.45.1-pyhd8ed1ab_1] The following files were already present in the environment:\n",
            "    - bin/wheel\n",
            "Linking cffi-1.17.1-py312h06ac9bb_0\n",
            "Linking h2-4.2.0-pyhd8ed1ab_0\n",
            "Linking jsonpatch-1.33-pyhd8ed1ab_1\n",
            "\u001b[33m\u001b[1mwarning  libmamba\u001b[m [jsonpatch-1.33-pyhd8ed1ab_1] The following files were already present in the environment:\n",
            "    - bin/jsondiff\n",
            "    - bin/jsonpatch\n",
            "Linking libmamba-2.1.1-h430c389_0\n",
            "Linking pip-25.1.1-pyh8b19718_0\n",
            "\u001b[33m\u001b[1mwarning  libmamba\u001b[m [pip-25.1.1-pyh8b19718_0] The following files were already present in the environment:\n",
            "    - bin/pip\n",
            "    - bin/pip3\n",
            "Linking ruamel.yaml-0.18.10-py312h66e93f0_0\n",
            "Linking tqdm-4.67.1-pyhd8ed1ab_1\n",
            "\u001b[33m\u001b[1mwarning  libmamba\u001b[m [tqdm-4.67.1-pyhd8ed1ab_1] The following files were already present in the environment:\n",
            "    - bin/tqdm\n",
            "Linking libmambapy-2.1.1-py312h07448e0_0\n",
            "Linking mamba-2.1.1-had4a41a_0\n",
            "Linking zstandard-0.23.0-py312h66e93f0_2\n",
            "Linking conda-package-streaming-0.11.0-pyhd8ed1ab_1\n",
            "Linking urllib3-2.4.0-pyhd8ed1ab_0\n",
            "Linking requests-2.32.3-pyhd8ed1ab_1\n",
            "Linking conda-package-handling-2.4.0-pyh7900ff3_2\n",
            "Linking conda-25.3.0-py312h7900ff3_0\n",
            "Linking conda-libmamba-solver-25.3.0-pyhd8ed1ab_0\n",
            "\n",
            "Transaction finished\n",
            "\n",
            "installation finished.\n",
            "WARNING:\n",
            "    You currently have a PYTHONPATH environment variable set. This may cause\n",
            "    unexpected behavior when running the Python interpreter in Miniforge3.\n",
            "    For best results, please verify that your PYTHONPATH only points to\n",
            "    directories of packages that are compatible with the Python interpreter\n",
            "    in Miniforge3: /usr/local\n",
            "Channels:\n",
            " - conda-forge\n",
            " - bioconda\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "\n",
            "==> WARNING: A newer version of conda exists. <==\n",
            "    current version: 25.3.0\n",
            "    latest version: 25.5.1\n",
            "\n",
            "Please update conda by running\n",
            "\n",
            "    $ conda update -n base -c conda-forge conda\n",
            "\n",
            "\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - prokka\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    alsa-lib-1.2.14            |       hb9d3cd8_0         553 KB  conda-forge\n",
            "    aragorn-1.2.41             |       h7b50bb2_5         144 KB  bioconda\n",
            "    barrnap-0.9                |       hdfd78af_4         557 KB  bioconda\n",
            "    bedtools-2.31.1            |       h13024bc_3         1.5 MB  bioconda\n",
            "    blast-2.16.0               |       h66d330f_5       141.4 MB  bioconda\n",
            "    ca-certificates-2025.7.9   |       hbd8a1cb_0         149 KB  conda-forge\n",
            "    cairo-1.18.4               |       h3394656_0         955 KB  conda-forge\n",
            "    certifi-2025.7.9           |     pyhd8ed1ab_0         153 KB  conda-forge\n",
            "    conda-25.5.1               |  py312h7900ff3_0         1.1 MB  conda-forge\n",
            "    curl-8.13.0                |       h332b0f4_0         178 KB  conda-forge\n",
            "    entrez-direct-22.4         |       he881be0_0        14.0 MB  bioconda\n",
            "    expat-2.7.0                |       h5888daf_0         137 KB  conda-forge\n",
            "    font-ttf-dejavu-sans-mono-2.37|       hab24e00_0         388 KB  conda-forge\n",
            "    font-ttf-inconsolata-3.000 |       h77eed37_0          94 KB  conda-forge\n",
            "    font-ttf-source-code-pro-2.038|       h77eed37_0         684 KB  conda-forge\n",
            "    font-ttf-ubuntu-0.83       |       h77eed37_3         1.5 MB  conda-forge\n",
            "    fontconfig-2.15.0          |       h7e30c49_1         259 KB  conda-forge\n",
            "    fonts-conda-ecosystem-1    |                0           4 KB  conda-forge\n",
            "    fonts-conda-forge-1        |                0           4 KB  conda-forge\n",
            "    freetype-2.13.3            |       ha770c72_1         168 KB  conda-forge\n",
            "    giflib-5.2.2               |       hd590300_0          75 KB  conda-forge\n",
            "    graphite2-1.3.14           |       h5888daf_0          96 KB  conda-forge\n",
            "    gsl-2.7                    |       he838d99_0         3.2 MB  conda-forge\n",
            "    harfbuzz-11.2.1            |       h3beb420_0         1.7 MB  conda-forge\n",
            "    hmmer-3.4                  |       hb6cb901_4        11.4 MB  bioconda\n",
            "    infernal-1.1.4             |       h779adbc_0         3.4 MB  bioconda\n",
            "    lcms2-2.17                 |       h717163a_0         242 KB  conda-forge\n",
            "    lerc-4.0.0                 |       h0aef613_1         258 KB  conda-forge\n",
            "    libasprintf-0.25.1         |       h8e693c7_0          52 KB  conda-forge\n",
            "    libblas-3.9.0              |32_h59b9bed_openblas          17 KB  conda-forge\n",
            "    libcblas-3.9.0             |32_he106b2a_openblas          17 KB  conda-forge\n",
            "    libcups-2.3.3              |       hb8b1518_5         4.3 MB  conda-forge\n",
            "    libdeflate-1.24            |       h86f0d12_0          71 KB  conda-forge\n",
            "    libfreetype-2.13.3         |       ha770c72_1           8 KB  conda-forge\n",
            "    libfreetype6-2.13.3        |       h48d6fc4_1         371 KB  conda-forge\n",
            "    libgettextpo-0.25.1        |       h5888daf_0         174 KB  conda-forge\n",
            "    libgfortran-15.1.0         |       h69a702a_3          28 KB  conda-forge\n",
            "    libgfortran-ng-15.1.0      |       h69a702a_3          28 KB  conda-forge\n",
            "    libgfortran5-15.1.0        |       hcea5267_3         1.5 MB  conda-forge\n",
            "    libglib-2.84.2             |       h3618099_0         3.8 MB  conda-forge\n",
            "    libidn11-1.34              |       h1cef754_0         227 KB  conda-forge\n",
            "    libidn2-2.3.8              |       ha4ef2c3_0         135 KB  conda-forge\n",
            "    libjpeg-turbo-3.1.0        |       hb9d3cd8_0         614 KB  conda-forge\n",
            "    libopenblas-0.3.30         |pthreads_h94d23a6_0         5.6 MB  conda-forge\n",
            "    libpng-1.6.50              |       h943b412_0         282 KB  conda-forge\n",
            "    libtiff-4.7.0              |       hf01ce69_5         420 KB  conda-forge\n",
            "    libunistring-0.9.10        |       h7f98852_0         1.4 MB  conda-forge\n",
            "    libwebp-base-1.6.0         |       hd42ef1d_0         419 KB  conda-forge\n",
            "    libxcb-1.17.0              |       h8a09558_0         387 KB  conda-forge\n",
            "    minced-0.4.2               |       hdfd78af_1          47 KB  bioconda\n",
            "    mpi-1.0                    |          openmpi           4 KB  conda-forge\n",
            "    ncbi-vdb-3.2.1             |       h9948957_0        10.6 MB  bioconda\n",
            "    openjdk-23.0.2             |       h53dfc1b_2       181.4 MB  conda-forge\n",
            "    openmpi-4.1.6              |     hc5af2df_101         3.9 MB  conda-forge\n",
            "    openssl-3.5.1              |       h7b32b05_0         3.0 MB  conda-forge\n",
            "    parallel-20250622          |       ha770c72_0         1.9 MB  conda-forge\n",
            "    pcre2-10.45                |       hc749103_0         1.1 MB  conda-forge\n",
            "    perl-5.22.2.1              |                0        15.0 MB  conda-forge\n",
            "    perl-archive-tar-2.18      |                1          27 KB  bioconda\n",
            "    perl-bioperl-1.6.924       |                4         2.2 MB  bioconda\n",
            "    perl-common-sense-3.74     |                0           8 KB  bioconda\n",
            "    perl-exporter-tiny-0.042   |                1          10 KB  bioconda\n",
            "    perl-json-2.90             |                1          35 KB  bioconda\n",
            "    perl-json-xs-2.34          |                0          47 KB  bioconda\n",
            "    perl-list-moreutils-0.413  |                1          55 KB  bioconda\n",
            "    perl-threaded-5.32.1       |       hdfd78af_1           5 KB  bioconda\n",
            "    perl-xml-namespacesupport-1.11|                0           6 KB  bioconda\n",
            "    perl-xml-parser-2.44       |                3         150 KB  bioconda\n",
            "    perl-xml-sax-0.99          |                0          24 KB  bioconda\n",
            "    perl-xml-sax-base-1.08     |                0          13 KB  bioconda\n",
            "    perl-xml-sax-expat-0.51    |                0           5 KB  bioconda\n",
            "    perl-xml-simple-2.22       |                0          31 KB  bioconda\n",
            "    perl-yaml-1.24             |          pl522_1          39 KB  bioconda\n",
            "    pixman-0.46.2              |       h29eaf8c_0         393 KB  conda-forge\n",
            "    prodigal-2.6.3             |      h7b50bb2_10         568 KB  bioconda\n",
            "    prokka-1.13                |                2       188.4 MB  bioconda\n",
            "    pthread-stubs-0.4          |    hb9d3cd8_1002           8 KB  conda-forge\n",
            "    rpsbproc-0.5.0             |       hd6d6fdc_3         6.3 MB  bioconda\n",
            "    tbl2asn-25.7               |       h9ee0642_1         4.3 MB  bioconda\n",
            "    wget-1.21.4                |       hda4d442_0         752 KB  conda-forge\n",
            "    xorg-libice-1.1.2          |       hb9d3cd8_0          57 KB  conda-forge\n",
            "    xorg-libsm-1.2.6           |       he73a12e_0          27 KB  conda-forge\n",
            "    xorg-libx11-1.8.12         |       h4f16b4b_0         816 KB  conda-forge\n",
            "    xorg-libxau-1.0.12         |       hb9d3cd8_0          14 KB  conda-forge\n",
            "    xorg-libxdmcp-1.1.5        |       hb9d3cd8_0          19 KB  conda-forge\n",
            "    xorg-libxext-1.3.6         |       hb9d3cd8_0          49 KB  conda-forge\n",
            "    xorg-libxfixes-6.0.1       |       hb9d3cd8_0          19 KB  conda-forge\n",
            "    xorg-libxi-1.8.2           |       hb9d3cd8_0          46 KB  conda-forge\n",
            "    xorg-libxrandr-1.5.4       |       hb9d3cd8_0          29 KB  conda-forge\n",
            "    xorg-libxrender-0.9.12     |       hb9d3cd8_0          32 KB  conda-forge\n",
            "    xorg-libxt-1.3.1           |       hb9d3cd8_0         371 KB  conda-forge\n",
            "    xorg-libxtst-1.2.5         |       hb9d3cd8_3          32 KB  conda-forge\n",
            "    zlib-1.3.1                 |       hb9d3cd8_2          90 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:       625.7 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  alsa-lib           conda-forge/linux-64::alsa-lib-1.2.14-hb9d3cd8_0 \n",
            "  aragorn            bioconda/linux-64::aragorn-1.2.41-h7b50bb2_5 \n",
            "  barrnap            bioconda/noarch::barrnap-0.9-hdfd78af_4 \n",
            "  bedtools           bioconda/linux-64::bedtools-2.31.1-h13024bc_3 \n",
            "  blast              bioconda/linux-64::blast-2.16.0-h66d330f_5 \n",
            "  cairo              conda-forge/linux-64::cairo-1.18.4-h3394656_0 \n",
            "  curl               conda-forge/linux-64::curl-8.13.0-h332b0f4_0 \n",
            "  entrez-direct      bioconda/linux-64::entrez-direct-22.4-he881be0_0 \n",
            "  expat              conda-forge/linux-64::expat-2.7.0-h5888daf_0 \n",
            "  font-ttf-dejavu-s~ conda-forge/noarch::font-ttf-dejavu-sans-mono-2.37-hab24e00_0 \n",
            "  font-ttf-inconsol~ conda-forge/noarch::font-ttf-inconsolata-3.000-h77eed37_0 \n",
            "  font-ttf-source-c~ conda-forge/noarch::font-ttf-source-code-pro-2.038-h77eed37_0 \n",
            "  font-ttf-ubuntu    conda-forge/noarch::font-ttf-ubuntu-0.83-h77eed37_3 \n",
            "  fontconfig         conda-forge/linux-64::fontconfig-2.15.0-h7e30c49_1 \n",
            "  fonts-conda-ecosy~ conda-forge/noarch::fonts-conda-ecosystem-1-0 \n",
            "  fonts-conda-forge  conda-forge/noarch::fonts-conda-forge-1-0 \n",
            "  freetype           conda-forge/linux-64::freetype-2.13.3-ha770c72_1 \n",
            "  giflib             conda-forge/linux-64::giflib-5.2.2-hd590300_0 \n",
            "  graphite2          conda-forge/linux-64::graphite2-1.3.14-h5888daf_0 \n",
            "  gsl                conda-forge/linux-64::gsl-2.7-he838d99_0 \n",
            "  harfbuzz           conda-forge/linux-64::harfbuzz-11.2.1-h3beb420_0 \n",
            "  hmmer              bioconda/linux-64::hmmer-3.4-hb6cb901_4 \n",
            "  infernal           bioconda/linux-64::infernal-1.1.4-h779adbc_0 \n",
            "  lcms2              conda-forge/linux-64::lcms2-2.17-h717163a_0 \n",
            "  lerc               conda-forge/linux-64::lerc-4.0.0-h0aef613_1 \n",
            "  libasprintf        conda-forge/linux-64::libasprintf-0.25.1-h8e693c7_0 \n",
            "  libblas            conda-forge/linux-64::libblas-3.9.0-32_h59b9bed_openblas \n",
            "  libcblas           conda-forge/linux-64::libcblas-3.9.0-32_he106b2a_openblas \n",
            "  libcups            conda-forge/linux-64::libcups-2.3.3-hb8b1518_5 \n",
            "  libdeflate         conda-forge/linux-64::libdeflate-1.24-h86f0d12_0 \n",
            "  libfreetype        conda-forge/linux-64::libfreetype-2.13.3-ha770c72_1 \n",
            "  libfreetype6       conda-forge/linux-64::libfreetype6-2.13.3-h48d6fc4_1 \n",
            "  libgettextpo       conda-forge/linux-64::libgettextpo-0.25.1-h5888daf_0 \n",
            "  libgfortran        conda-forge/linux-64::libgfortran-15.1.0-h69a702a_3 \n",
            "  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-15.1.0-h69a702a_3 \n",
            "  libgfortran5       conda-forge/linux-64::libgfortran5-15.1.0-hcea5267_3 \n",
            "  libglib            conda-forge/linux-64::libglib-2.84.2-h3618099_0 \n",
            "  libidn11           conda-forge/linux-64::libidn11-1.34-h1cef754_0 \n",
            "  libidn2            conda-forge/linux-64::libidn2-2.3.8-ha4ef2c3_0 \n",
            "  libjpeg-turbo      conda-forge/linux-64::libjpeg-turbo-3.1.0-hb9d3cd8_0 \n",
            "  libopenblas        conda-forge/linux-64::libopenblas-0.3.30-pthreads_h94d23a6_0 \n",
            "  libpng             conda-forge/linux-64::libpng-1.6.50-h943b412_0 \n",
            "  libtiff            conda-forge/linux-64::libtiff-4.7.0-hf01ce69_5 \n",
            "  libunistring       conda-forge/linux-64::libunistring-0.9.10-h7f98852_0 \n",
            "  libwebp-base       conda-forge/linux-64::libwebp-base-1.6.0-hd42ef1d_0 \n",
            "  libxcb             conda-forge/linux-64::libxcb-1.17.0-h8a09558_0 \n",
            "  minced             bioconda/noarch::minced-0.4.2-hdfd78af_1 \n",
            "  mpi                conda-forge/linux-64::mpi-1.0-openmpi \n",
            "  ncbi-vdb           bioconda/linux-64::ncbi-vdb-3.2.1-h9948957_0 \n",
            "  openjdk            conda-forge/linux-64::openjdk-23.0.2-h53dfc1b_2 \n",
            "  openmpi            conda-forge/linux-64::openmpi-4.1.6-hc5af2df_101 \n",
            "  parallel           conda-forge/linux-64::parallel-20250622-ha770c72_0 \n",
            "  pcre2              conda-forge/linux-64::pcre2-10.45-hc749103_0 \n",
            "  perl               conda-forge/linux-64::perl-5.22.2.1-0 \n",
            "  perl-archive-tar   bioconda/linux-64::perl-archive-tar-2.18-1 \n",
            "  perl-bioperl       bioconda/linux-64::perl-bioperl-1.6.924-4 \n",
            "  perl-common-sense  bioconda/linux-64::perl-common-sense-3.74-0 \n",
            "  perl-exporter-tiny bioconda/linux-64::perl-exporter-tiny-0.042-1 \n",
            "  perl-json          bioconda/linux-64::perl-json-2.90-1 \n",
            "  perl-json-xs       bioconda/linux-64::perl-json-xs-2.34-0 \n",
            "  perl-list-moreuti~ bioconda/linux-64::perl-list-moreutils-0.413-1 \n",
            "  perl-threaded      bioconda/noarch::perl-threaded-5.32.1-hdfd78af_1 \n",
            "  perl-xml-namespac~ bioconda/linux-64::perl-xml-namespacesupport-1.11-0 \n",
            "  perl-xml-parser    bioconda/linux-64::perl-xml-parser-2.44-3 \n",
            "  perl-xml-sax       bioconda/linux-64::perl-xml-sax-0.99-0 \n",
            "  perl-xml-sax-base  bioconda/linux-64::perl-xml-sax-base-1.08-0 \n",
            "  perl-xml-sax-expat bioconda/linux-64::perl-xml-sax-expat-0.51-0 \n",
            "  perl-xml-simple    bioconda/linux-64::perl-xml-simple-2.22-0 \n",
            "  perl-yaml          bioconda/linux-64::perl-yaml-1.24-pl522_1 \n",
            "  pixman             conda-forge/linux-64::pixman-0.46.2-h29eaf8c_0 \n",
            "  prodigal           bioconda/linux-64::prodigal-2.6.3-h7b50bb2_10 \n",
            "  prokka             bioconda/linux-64::prokka-1.13-2 \n",
            "  pthread-stubs      conda-forge/linux-64::pthread-stubs-0.4-hb9d3cd8_1002 \n",
            "  rpsbproc           bioconda/linux-64::rpsbproc-0.5.0-hd6d6fdc_3 \n",
            "  tbl2asn            bioconda/linux-64::tbl2asn-25.7-h9ee0642_1 \n",
            "  wget               conda-forge/linux-64::wget-1.21.4-hda4d442_0 \n",
            "  xorg-libice        conda-forge/linux-64::xorg-libice-1.1.2-hb9d3cd8_0 \n",
            "  xorg-libsm         conda-forge/linux-64::xorg-libsm-1.2.6-he73a12e_0 \n",
            "  xorg-libx11        conda-forge/linux-64::xorg-libx11-1.8.12-h4f16b4b_0 \n",
            "  xorg-libxau        conda-forge/linux-64::xorg-libxau-1.0.12-hb9d3cd8_0 \n",
            "  xorg-libxdmcp      conda-forge/linux-64::xorg-libxdmcp-1.1.5-hb9d3cd8_0 \n",
            "  xorg-libxext       conda-forge/linux-64::xorg-libxext-1.3.6-hb9d3cd8_0 \n",
            "  xorg-libxfixes     conda-forge/linux-64::xorg-libxfixes-6.0.1-hb9d3cd8_0 \n",
            "  xorg-libxi         conda-forge/linux-64::xorg-libxi-1.8.2-hb9d3cd8_0 \n",
            "  xorg-libxrandr     conda-forge/linux-64::xorg-libxrandr-1.5.4-hb9d3cd8_0 \n",
            "  xorg-libxrender    conda-forge/linux-64::xorg-libxrender-0.9.12-hb9d3cd8_0 \n",
            "  xorg-libxt         conda-forge/linux-64::xorg-libxt-1.3.1-hb9d3cd8_0 \n",
            "  xorg-libxtst       conda-forge/linux-64::xorg-libxtst-1.2.5-hb9d3cd8_3 \n",
            "  zlib               conda-forge/linux-64::zlib-1.3.1-hb9d3cd8_2 \n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2025.4.26-hbd8a1cb_0 --> 2025.7.9-hbd8a1cb_0 \n",
            "  certifi                            2025.4.26-pyhd8ed1ab_0 --> 2025.7.9-pyhd8ed1ab_0 \n",
            "  conda                              25.3.0-py312h7900ff3_0 --> 25.5.1-py312h7900ff3_0 \n",
            "  openssl                                  3.5.0-h7b32b05_1 --> 3.5.1-h7b32b05_0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "prokka-1.13          | 188.4 MB  | :   0% 0/1 [00:00<?, ?it/s]\n",
            "openjdk-23.0.2       | 181.4 MB  | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.22.2.1        | 15.0 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hmmer-3.4            | 11.4 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ncbi-vdb-3.2.1       | 10.6 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "rpsbproc-0.5.0       | 6.3 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libopenblas-0.3.30   | 5.6 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbl2asn-25.7         | 4.3 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openmpi-4.1.6        | 3.9 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.2       | 3.8 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "infernal-1.1.4       | 3.4 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gsl-2.7              | 3.2 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.5.1        | 3.0 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "perl-bioperl-1.6.924 | 2.2 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "parallel-20250622    | 1.9 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.2.1      | 1.7 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :   0% 0.00011050096527152409/1 [00:00<16:00, 960.71s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | :   0% 0.001115570336015178/1 [00:00<02:08, 128.17s/it]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.22.2.1        | 15.0 MB   | :   0% 0.0010450381195896362/1 [00:00<02:22, 142.22s/it]\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :   0% 8.613167481774731e-05/1 [00:00<31:02, 1862.23s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :   0% 8.291985465562517e-05/1 [00:00<47:08, 2828.43s/it]\n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | :   9% 0.08924562688121424/1 [00:00<00:02,  2.26s/it]  \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.22.2.1        | 15.0 MB   | :  11% 0.11181907879609108/1 [00:00<00:01,  1.83s/it]   \u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :   1% 0.009129957530681215/1 [00:00<00:23, 23.38s/it]   \u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :   1% 0.009038264157463144/1 [00:00<00:29, 29.38s/it]   \n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | :  29% 0.2878171466919159/1 [00:00<00:00,  1.08it/s] \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.22.2.1        | 15.0 MB   | :  29% 0.2852954066479707/1 [00:00<00:00,  1.03it/s] \u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :   2% 0.022566498802249795/1 [00:00<00:12, 12.63s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :   2% 0.023300479158230674/1 [00:00<00:13, 13.87s/it]\n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | :  46% 0.464077259782314/1 [00:00<00:00,  1.33it/s] \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.22.2.1        | 15.0 MB   | :  45% 0.4525015057823125/1 [00:00<00:00,  1.26it/s]\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :   4% 0.03901764869243953/1 [00:00<00:08,  9.08s/it] \u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :   4% 0.035240938228640696/1 [00:00<00:10, 11.39s/it]\n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | :  65% 0.6481463652248184/1 [00:00<00:00,  1.51it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :   5% 0.05279871666327911/1 [00:00<00:07,  8.36s/it]\u001b[A\n",
            "\n",
            "\n",
            "perl-5.22.2.1        | 15.0 MB   | :  66% 0.6573289772218812/1 [00:00<00:00,  1.50it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :   5% 0.04568883991524947/1 [00:00<00:10, 11.14s/it] \n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | :  81% 0.8054417826029584/1 [00:00<00:00,  1.50it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.22.2.1        | 15.0 MB   | :  86% 0.8642465249006293/1 [00:00<00:00,  1.65it/s]\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :   7% 0.0706279733505528/1 [00:00<00:07,  7.69s/it] \u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :   6% 0.06252157041034138/1 [00:00<00:08,  8.85s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :   9% 0.08888788841191524/1 [00:00<00:06,  6.88s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :   8% 0.07620334642851953/1 [00:00<00:07,  8.40s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  11% 0.10585582835101146/1 [00:00<00:06,  6.73s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  10% 0.09726498951104832/1 [00:00<00:06,  6.93s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  12% 0.12273763661528993/1 [00:00<00:05,  6.49s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  12% 0.12214094590773589/1 [00:01<00:05,  5.71s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  14% 0.14237565847373632/1 [00:01<00:05,  6.04s/it]\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  22% 0.2161398880711011/1 [00:01<00:03,  4.14s/it] \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | : 100% 1.0/1 [00:01<00:00,  1.35s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | : 100% 1.0/1 [00:01<00:00,  1.35s/it]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  14% 0.1422904705890528/1 [00:01<00:04,  5.59s/it] \n",
            "openjdk-23.0.2       | 181.4 MB  | :  16% 0.16192754865736494/1 [00:01<00:04,  5.78s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hmmer-3.4            | 11.4 MB   | :   0% 0.0013752217507301458/1 [00:01<14:30, 871.85s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ncbi-vdb-3.2.1       | 10.6 MB   | :   0% 0.0014804186086016804/1 [00:01<13:32, 814.14s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  16% 0.16053283861329035/1 [00:01<00:04,  5.62s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hmmer-3.4            | 11.4 MB   | :  27% 0.2722939066445689/1 [00:01<00:02,  3.46s/it]    \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ncbi-vdb-3.2.1       | 10.6 MB   | :  28% 0.2842403728515226/1 [00:01<00:02,  3.33s/it]    \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  18% 0.17941227864536766/1 [00:01<00:05,  6.15s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hmmer-3.4            | 11.4 MB   | :  46% 0.46069928649459885/1 [00:01<00:01,  2.00s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  18% 0.17860936692821663/1 [00:01<00:05,  6.65s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  26% 0.2633238002420419/1 [00:01<00:04,  5.44s/it] \u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  20% 0.1958634285355574/1 [00:01<00:05,  7.06s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hmmer-3.4            | 11.4 MB   | :  66% 0.6559807750982796/1 [00:01<00:00,  1.37s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ncbi-vdb-3.2.1       | 10.6 MB   | :  70% 0.7017184204771965/1 [00:01<00:00,  1.28s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  28% 0.2831034730256447/1 [00:01<00:04,  5.65s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  19% 0.19452997902209665/1 [00:01<00:06,  7.60s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ncbi-vdb-3.2.1       | 10.6 MB   | :  92% 0.9223007931588468/1 [00:01<00:00,  1.04it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hmmer-3.4            | 11.4 MB   | :  84% 0.8375100461946589/1 [00:01<00:00,  1.13s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  30% 0.30177813615653226/1 [00:01<00:04,  5.95s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  21% 0.20862635431355295/1 [00:01<00:06,  7.85s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  25% 0.24513074653130887/1 [00:01<00:04,  6.53s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  22% 0.22197645091310858/1 [00:01<00:06,  7.95s/it]\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  23% 0.23499486809404174/1 [00:01<00:06,  8.08s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  26% 0.2608928430229566/1 [00:01<00:05,  7.46s/it] \u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  35% 0.3541555936952347/1 [00:02<00:04,  6.29s/it] \u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  25% 0.2476816058563524/1 [00:02<00:06,  8.44s/it] \n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  37% 0.37327226068720837/1 [00:02<00:03,  5.98s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  26% 0.2627730194036762/1 [00:02<00:05,  7.93s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ncbi-vdb-3.2.1       | 10.6 MB   | : 100% 1.0/1 [00:02<00:00,  1.04it/s]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hmmer-3.4            | 11.4 MB   | : 100% 1.0/1 [00:02<00:00,  1.13s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  39% 0.3930519334708112/1 [00:02<00:03,  5.77s/it] \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "rpsbproc-0.5.0       | 6.3 MB    | :   0% 0.002499396660535131/1 [00:02<14:55, 898.07s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  28% 0.2756255968752981/1 [00:02<00:05,  8.03s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libopenblas-0.3.30   | 5.6 MB    | :   0% 0.002769055467135297/1 [00:02<13:37, 819.62s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  41% 0.4106215869489835/1 [00:02<00:03,  6.09s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  29% 0.2882294147829531/1 [00:02<00:05,  8.36s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  33% 0.3278171543563463/1 [00:02<00:04,  6.90s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libopenblas-0.3.30   | 5.6 MB    | :  39% 0.38766776539894154/1 [00:02<00:02,  4.41s/it]  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  43% 0.4309537645589439/1 [00:02<00:03,  5.72s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  30% 0.30033571356267436/1 [00:02<00:06,  8.72s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libopenblas-0.3.30   | 5.6 MB    | :  84% 0.8417928620091303/1 [00:02<00:00,  1.79s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  34% 0.34271793409981655/1 [00:02<00:05,  7.86s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  32% 0.31841224187760064/1 [00:02<00:05,  7.65s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  47% 0.4652090637931164/1 [00:02<00:03,  6.37s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  33% 0.3316794186225007/1 [00:02<00:05,  8.11s/it] \n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libopenblas-0.3.30   | 5.6 MB    | : 100% 1.0/1 [00:02<00:00,  1.79s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "rpsbproc-0.5.0       | 6.3 MB    | : 100% 1.0/1 [00:02<00:00,  1.72s/it]              \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "rpsbproc-0.5.0       | 6.3 MB    | : 100% 1.0/1 [00:02<00:00,  1.72s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  48% 0.483883726924004/1 [00:02<00:03,  6.07s/it] \u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  37% 0.37312241531048135/1 [00:02<00:05,  8.07s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  35% 0.34677083216982446/1 [00:02<00:05,  7.92s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbl2asn-25.7         | 4.3 MB    | :   0% 0.0036596400713611943/1 [00:02<13:01, 784.24s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  50% 0.5006798736452757/1 [00:02<00:03,  6.05s/it]\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  39% 0.3864728249072322/1 [00:02<00:05,  8.17s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  36% 0.36136472658921454/1 [00:02<00:04,  7.67s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbl2asn-25.7         | 4.3 MB    | :  56% 0.5562652908469016/1 [00:02<00:01,  3.80s/it]    \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  52% 0.5174760203665473/1 [00:03<00:02,  6.19s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  37% 0.3745489834794589/1 [00:03<00:05,  8.64s/it] \n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  54% 0.5375871960459647/1 [00:03<00:02,  5.90s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  39% 0.3869040418231471/1 [00:03<00:05,  8.50s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  56% 0.559355886204455/1 [00:03<00:02,  5.46s/it] \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbl2asn-25.7         | 4.3 MB    | : 100% 1.0/1 [00:03<00:00,  2.09s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbl2asn-25.7         | 4.3 MB    | : 100% 1.0/1 [00:03<00:00,  2.09s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | : 100% 1.0/1 [00:03<00:00,  2.15s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | : 100% 1.0/1 [00:03<00:00,  2.15s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  40% 0.40290757377168274/1 [00:03<00:04,  7.80s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openmpi-4.1.6        | 3.9 MB    | :   0% 0.004025916839655281/1 [00:03<13:31, 814.84s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  58% 0.5784725531964285/1 [00:03<00:02,  5.40s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.2       | 3.8 MB    | :   0% 0.004142535168818928/1 [00:03<13:13, 796.73s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  44% 0.43952993659496453/1 [00:03<00:04,  7.91s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  42% 0.4160089108072715/1 [00:03<00:04,  7.83s/it] \n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.2       | 3.8 MB    | :  70% 0.7000884435303988/1 [00:03<00:01,  3.46s/it]   \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  45% 0.45477524303770583/1 [00:03<00:04,  7.68s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  43% 0.42902732798820464/1 [00:03<00:04,  8.15s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  62% 0.6207944228954223/1 [00:03<00:02,  5.56s/it]\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  47% 0.4687285743581809/1 [00:03<00:04,  7.61s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  44% 0.44146530618654845/1 [00:03<00:04,  8.44s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openmpi-4.1.6        | 3.9 MB    | : 100% 1.0/1 [00:03<00:00,  2.29s/it]              \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openmpi-4.1.6        | 3.9 MB    | : 100% 1.0/1 [00:03<00:00,  2.29s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  48% 0.48483519754909965/1 [00:03<00:03,  7.15s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  45% 0.45348868511161405/1 [00:03<00:04,  8.71s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  64% 0.6395795869915815/1 [00:03<00:02,  6.61s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "infernal-1.1.4       | 3.4 MB    | :   0% 0.0045368125900979855/1 [00:03<13:47, 831.63s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  50% 0.4989607922192102/1 [00:03<00:03,  7.96s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  47% 0.4650974647634016/1 [00:03<00:04,  9.09s/it] \n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "infernal-1.1.4       | 3.4 MB    | :  51% 0.5081230100909744/1 [00:03<00:02,  5.42s/it]    \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  66% 0.655933729851767/1 [00:03<00:02,  7.46s/it] \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gsl-2.7              | 3.2 MB    | :  90% 0.902559898448743/1 [00:03<00:00,  2.64s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "infernal-1.1.4       | 3.4 MB    | :  88% 0.8756048298889112/1 [00:03<00:00,  2.81s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  48% 0.4762087252872554/1 [00:04<00:05, 11.19s/it]\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  49% 0.486242027700586/1 [00:04<00:05, 10.95s/it] \n",
            "openjdk-23.0.2       | 181.4 MB  | :  52% 0.5229915294933617/1 [00:04<00:05, 10.89s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gsl-2.7              | 3.2 MB    | : 100% 1.0/1 [00:04<00:00,  2.64s/it]              \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  68% 0.6834484702043765/1 [00:04<00:02,  8.37s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  50% 0.49586073084063853/1 [00:04<00:05, 11.46s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "infernal-1.1.4       | 3.4 MB    | : 100% 1.0/1 [00:04<00:00,  2.81s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  53% 0.5328966720974027/1 [00:04<00:05, 11.59s/it]\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  70% 0.6962665821758732/1 [00:04<00:02,  8.57s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.5.1        | 3.0 MB    | :  70% 0.7011991688283814/1 [00:04<00:01,  4.38s/it]  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  50% 0.5049819148527573/1 [00:04<00:05, 11.34s/it] \n",
            "openjdk-23.0.2       | 181.4 MB  | :  55% 0.545041238246705/1 [00:04<00:04, 10.73s/it] \u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  71% 0.7085321893210125/1 [00:04<00:02,  8.80s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  51% 0.514103098864876/1 [00:04<00:06, 12.38s/it] \n",
            "openjdk-23.0.2       | 181.4 MB  | :  55% 0.5548602491759282/1 [00:04<00:05, 11.49s/it]\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  72% 0.720245291639794/1 [00:04<00:02,  9.28s/it] \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  52% 0.5224780041850943/1 [00:04<00:06, 12.81s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  73% 0.7312953881669464/1 [00:04<00:02,  9.28s/it]\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  56% 0.5639902067066094/1 [00:04<00:05, 12.02s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "perl-bioperl-1.6.924 | 2.2 MB    | : 100% 1.0/1 [00:04<00:00,  2.84s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  53% 0.5315162683425574/1 [00:04<00:05, 12.43s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  74% 0.7423454846940988/1 [00:04<00:02,  9.53s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.2.1      | 1.7 MB    | :   1% 0.009469283203465906/1 [00:04<08:16, 500.95s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "parallel-20250622    | 1.9 MB    | :   1% 0.008169594886025077/1 [00:04<09:37, 581.88s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  54% 0.5411349714826099/1 [00:04<00:05, 11.81s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  75% 0.7529535773601651/1 [00:04<00:02, 10.03s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "parallel-20250622    | 1.9 MB    | :  81% 0.8087898937164826/1 [00:04<00:00,  4.26s/it]   \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  55% 0.5497586363667949/1 [00:04<00:05, 12.39s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  77% 0.7669871999496487/1 [00:04<00:02,  9.07s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.2.1      | 1.7 MB    | : 100% 1.0/1 [00:04<00:00,  3.55s/it]                  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.2.1      | 1.7 MB    | : 100% 1.0/1 [00:04<00:00,  3.55s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "parallel-20250622    | 1.9 MB    | : 100% 1.0/1 [00:04<00:00,  4.26s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  56% 0.560372377762715/1 [00:05<00:05, 11.58s/it] \n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  78% 0.7782582984073442/1 [00:05<00:02,  9.11s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  57% 0.5691618823562112/1 [00:05<00:05, 11.98s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  79% 0.7894188958997681/1 [00:05<00:02, 10.03s/it]\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  62% 0.6160998699713466/1 [00:05<00:04, 12.02s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  58% 0.5776197075310849/1 [00:05<00:05, 13.24s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  80% 0.7999164876005629/1 [00:05<00:02, 10.01s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  59% 0.5888138879095943/1 [00:05<00:04, 11.96s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  81% 0.8127345995720596/1 [00:05<00:01,  9.29s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  60% 0.5974375527937794/1 [00:05<00:04, 12.41s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  82% 0.823784696099212/1 [00:05<00:01,  9.69s/it] \u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | :  64% 0.6438342692626612/1 [00:05<00:04, 12.03s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  61% 0.6067245765152094/1 [00:05<00:04, 12.69s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  61% 0.6147678024168051/1 [00:05<00:05, 15.13s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  84% 0.8447798795008017/1 [00:05<00:01, 12.29s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  63% 0.6254644636673807/1 [00:05<00:05, 13.40s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  86% 0.8575979914722984/1 [00:05<00:01, 11.34s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  64% 0.6353319263714001/1 [00:06<00:04, 12.74s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  87% 0.8710791092354244/1 [00:06<00:01, 10.11s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  64% 0.6442872706742077/1 [00:06<00:04, 12.30s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  88% 0.8844497260332788/1 [00:06<00:01,  9.27s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  65% 0.6527450958490814/1 [00:06<00:04, 12.23s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  90% 0.8975993409005901/1 [00:06<00:00,  9.15s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  67% 0.6656805931753589/1 [00:06<00:03, 11.04s/it]\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  68% 0.6759626551526564/1 [00:06<00:03, 10.94s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  72% 0.7153235593613915/1 [00:06<00:03, 11.09s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  69% 0.6852496788740864/1 [00:06<00:03, 10.90s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  73% 0.7256593603395212/1 [00:06<00:02, 10.68s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  70% 0.6964438592525959/1 [00:06<00:03, 10.26s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  74% 0.7352199762442911/1 [00:06<00:02, 10.70s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | :  71% 0.7108719139626746/1 [00:06<00:02,  9.27s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  75% 0.7487426491906775/1 [00:06<00:02,  9.69s/it]\u001b[A\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  98% 0.9755025214170147/1 [00:06<00:00,  7.18s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  72% 0.7217344149225615/1 [00:06<00:02,  9.92s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | :  99% 0.9895361440064981/1 [00:06<00:00,  7.55s/it]\u001b[A\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  74% 0.7437081764063022/1 [00:07<00:02,  9.56s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  75% 0.7542389979475665/1 [00:07<00:02,  9.75s/it]\n",
            "openjdk-23.0.2       | 181.4 MB  | :  79% 0.7937895151203592/1 [00:07<00:01,  9.60s/it]\u001b[A\n",
            "prokka-1.13          | 188.4 MB  | :  76% 0.7646039797795198/1 [00:07<00:02, 10.50s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  78% 0.7798612330361547/1 [00:07<00:02,  9.82s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  79% 0.7942892877462335/1 [00:07<00:01,  9.07s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  81% 0.8058980673980211/1 [00:07<00:01,  8.95s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  82% 0.8195798434161993/1 [00:07<00:01,  8.88s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  83% 0.8309398635040198/1 [00:07<00:01,  9.98s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  84% 0.8412219254813174/1 [00:08<00:01, 10.72s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  85% 0.8544891022262174/1 [00:08<00:01,  9.70s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  87% 0.8676733591164618/1 [00:08<00:01,  9.25s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  88% 0.8824330932451631/1 [00:08<00:00,  8.45s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  90% 0.8988512244669769/1 [00:08<00:00,  7.98s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  91% 0.9128646799037775/1 [00:08<00:00,  7.96s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  93% 0.9278731735964457/1 [00:08<00:00,  7.59s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  94% 0.9412232701960014/1 [00:08<00:00,  8.14s/it]\n",
            "prokka-1.13          | 188.4 MB  | :  99% 0.9889021866229858/1 [00:09<00:00,  8.19s/it]\n",
            "\n",
            "blast-2.16.0         | 141.4 MB  | : 100% 1.0/1 [00:11<00:00,  7.55s/it]               \u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | : 100% 1.0/1 [00:12<00:00, 84.66s/it]               \u001b[A\n",
            "prokka-1.13          | 188.4 MB  | : 100% 1.0/1 [00:12<00:00,  8.19s/it]               \n",
            "\n",
            "\n",
            "\n",
            "entrez-direct-22.4   | 14.0 MB   | : 100% 1.0/1 [00:13<00:00,  1.35s/it]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ncbi-vdb-3.2.1       | 10.6 MB   | : 100% 1.0/1 [00:18<00:00,  1.04it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.22.2.1        | 15.0 MB   | : 100% 1.0/1 [00:18<00:00,  1.65it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libopenblas-0.3.30   | 5.6 MB    | : 100% 1.0/1 [00:19<00:00,  1.79s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "rpsbproc-0.5.0       | 6.3 MB    | : 100% 1.0/1 [00:23<00:00,  1.72s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hmmer-3.4            | 11.4 MB   | : 100% 1.0/1 [00:24<00:00,  1.13s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | : 100% 1.0/1 [00:24<00:00,  2.15s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.2       | 3.8 MB    | : 100% 1.0/1 [00:25<00:00,  3.46s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbl2asn-25.7         | 4.3 MB    | : 100% 1.0/1 [00:25<00:00,  2.09s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openmpi-4.1.6        | 3.9 MB    | : 100% 1.0/1 [00:25<00:00,  2.29s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "infernal-1.1.4       | 3.4 MB    | : 100% 1.0/1 [00:27<00:00,  2.81s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.5.1        | 3.0 MB    | : 100% 1.0/1 [00:27<00:00,  4.38s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gsl-2.7              | 3.2 MB    | : 100% 1.0/1 [00:27<00:00,  2.64s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.2.1      | 1.7 MB    | : 100% 1.0/1 [00:27<00:00,  3.55s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "parallel-20250622    | 1.9 MB    | : 100% 1.0/1 [00:27<00:00,  4.26s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "perl-bioperl-1.6.924 | 2.2 MB    | : 100% 1.0/1 [00:29<00:00,  2.84s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-23.0.2       | 181.4 MB  | : 100% 1.0/1 [00:36<00:00, 84.66s/it]\u001b[A\n",
            "\n",
            "prokka-1.13          | 188.4 MB  | : 100% 1.0/1 [01:43<00:00,  8.19s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\u001b[A\n",
            "\n",
            "\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\u001b[A\n",
            "\n",
            "\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\u001b[A\n",
            "\n",
            "\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\u001b[A\n",
            "\n",
            "\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: - \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Verifying transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\  \n",
            "For Linux 64, Open MPI is built with CUDA awareness but this support is disabled by default.\n",
            "To enable it, please set the environment variable OMPI_MCA_opal_cuda_support=true before\n",
            "launching your MPI processes. Equivalently, you can set the MCA parameter in the command line:\n",
            "mpiexec --mca opal_cuda_support 1 ...\n",
            " \n",
            "In addition, the UCX support is also built but disabled by default.\n",
            "To enable it, first install UCX (conda install -c conda-forge ucx). Then, set the environment\n",
            "variables OMPI_MCA_pml=\"ucx\" OMPI_MCA_osc=\"ucx\" before launching your MPI processes.\n",
            "Equivalently, you can set the MCA parameters in the command line:\n",
            "mpiexec --mca pml ucx --mca osc ucx ...\n",
            "Note that you might also need to set UCX_MEMTYPE_CACHE=n for CUDA awareness via UCX.\n",
            "Please consult UCX's documentation for detail.\n",
            " \n",
            "\n",
            "\b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n"
          ]
        }
      ],
      "source": [
        "# Переустанавливаем Miniforge поверх существующей\n",
        "!bash miniforge.sh -b -u -p /usr/local\n",
        "import sys\n",
        "sys.path.append('/usr/local/lib/python3.10/site-packages')\n",
        "!conda install -y -c conda-forge -c bioconda prokka\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8Hw5DYQvutfX",
        "outputId": "a6154156-ff3e-43bd-f6e3-9034fe631586"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Can't locate XML/Simple.pm in @INC (you may need to install the XML::Simple module) (@INC contains: /usr/local/bin/../perl5 /usr/local/lib/perl5/site_perl/5.22.2/x86_64-linux-thread-multi /usr/local/lib/perl5/site_perl/5.22.2 /usr/local/lib/perl5/5.22.2/x86_64-linux-thread-multi /usr/local/lib/perl5/5.22.2 .) at /usr/local/bin/prokka line 27.\n",
            "BEGIN failed--compilation aborted at /usr/local/bin/prokka line 27.\n"
          ]
        }
      ],
      "source": [
        "!prokka --version\n",
        "\n",
        "\n"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}